{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DATA CONDENSTAION ##\n",
    "\n",
    "### OBJECTIVES ###\n",
    "\n",
    "1. What genes are most represented? Need to get above 5000 genes\n",
    "    - Do all GPLs hvae the same gene symbols?\n",
    "    - create a definitive gene symbol list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pickle\n",
    "import ntpath\n",
    "import mysql.connector #maybe unnecessary\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = create_engine('mysql+mysqlconnector://Simon:Bane@localhost/microarray_data_v2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sample meta import\n",
    "sampleMeta = pd.read_sql_table('master', con=engine)\n",
    "gse_training = pd.read_excel('../Training/geo-expression-array-immune-cells-studies.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [],
   "source": [
    "sampleMeta['cell.type']=sampleMeta['cell.type'].apply(lambda x: x.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gse', 'cell.types', 'n.cell.types', 'title'], dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_training.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>platform_id</th>\n",
       "      <th>gse</th>\n",
       "      <th>file</th>\n",
       "      <th>id</th>\n",
       "      <th>col</th>\n",
       "      <th>pattern</th>\n",
       "      <th>hit</th>\n",
       "      <th>cell.type</th>\n",
       "      <th>val</th>\n",
       "      <th>technology</th>\n",
       "      <th>title</th>\n",
       "      <th>normalization</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10021</td>\n",
       "      <td>GSE10021_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253209</td>\n",
       "      <td>characteristics_ch1_1</td>\n",
       "      <td>\\bT.?cell(?!.*lymphoma)(?!.*leukemia)</td>\n",
       "      <td>from T cell luke</td>\n",
       "      <td>t_cells</td>\n",
       "      <td>human cultured cell line derived from T cell l...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253634</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253635</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253636</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253637</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  platform_id       gse                                    file         id  \\\n",
       "0      GPL570  GSE10021  GSE10021_series_matrix.txt.gz.anno.tsv  GSM253209   \n",
       "1      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253634   \n",
       "2      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253635   \n",
       "3      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253636   \n",
       "4      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253637   \n",
       "\n",
       "                     col                                pattern  \\\n",
       "0  characteristics_ch1_1  \\bT.?cell(?!.*lymphoma)(?!.*leukemia)   \n",
       "1        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "2        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "3        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "4        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "\n",
       "                    hit   cell.type  \\\n",
       "0      from T cell luke     t_cells   \n",
       "1  olar macrophages obt  macrophage   \n",
       "2  olar macrophages obt  macrophage   \n",
       "3  olar macrophages obt  macrophage   \n",
       "4  olar macrophages obt  macrophage   \n",
       "\n",
       "                                                 val               technology  \\\n",
       "0  human cultured cell line derived from T cell l...  in situ oligonucleotide   \n",
       "1  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "2  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "3  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "4  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "\n",
       "                                               title normalization  \n",
       "0  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "1  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "2  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "3  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "4  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  "
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleMeta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GPL570', 'GPL571', 'GPL6244', 'GPL96', 'GPL97'], dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleMeta['platform_id'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gse10021',\n",
       " 'gse10038',\n",
       " 'gse49954',\n",
       " 'gse10046',\n",
       " 'gse100786',\n",
       " 'gse10147',\n",
       " 'gse101587',\n",
       " 'gse10178',\n",
       " 'gse101855',\n",
       " 'gse13576',\n",
       " 'gse102459',\n",
       " 'gse51382',\n",
       " 'gse10315',\n",
       " 'gse10316',\n",
       " 'gse35340',\n",
       " 'gse103381',\n",
       " 'gse65856',\n",
       " 'gse13670',\n",
       " 'gse10358',\n",
       " 'gse10437',\n",
       " 'gse36767',\n",
       " 'gse10463',\n",
       " 'gse10479',\n",
       " 'gse10524',\n",
       " 'gse105437',\n",
       " 'gse10586',\n",
       " 'gse10595',\n",
       " 'gse106151',\n",
       " 'gse22045',\n",
       " 'gse106382',\n",
       " 'gse22167',\n",
       " 'gse106532',\n",
       " 'gse10767',\n",
       " 'gse22373',\n",
       " 'gse107683',\n",
       " 'gse107818',\n",
       " 'gse10821',\n",
       " 'gse10831',\n",
       " 'gse108384',\n",
       " 'gse10846',\n",
       " 'gse10880',\n",
       " 'gse10896',\n",
       " 'gse109178',\n",
       " 'gse10927',\n",
       " 'gse109348',\n",
       " 'gse109662',\n",
       " 'gse109836',\n",
       " 'gse110023',\n",
       " 'gse110064',\n",
       " 'gse110085',\n",
       " 'gse110376',\n",
       " 'gse110545',\n",
       " 'gse11057',\n",
       " 'gse11083',\n",
       " 'gse11118',\n",
       " 'gse11151',\n",
       " 'gse11190',\n",
       " 'gse112209',\n",
       " 'gse112210',\n",
       " 'gse112809',\n",
       " 'gse11281',\n",
       " 'gse11292',\n",
       " 'gse112953',\n",
       " 'gse11327',\n",
       " 'gse113645',\n",
       " 'gse11373',\n",
       " 'gse11418',\n",
       " 'gse114181',\n",
       " 'gse11430',\n",
       " 'gse23371',\n",
       " 'gse116487',\n",
       " 'gse116660',\n",
       " 'gse116912',\n",
       " 'gse117139',\n",
       " 'gse11755',\n",
       " 'gse11761',\n",
       " 'gse117676',\n",
       " 'gse11854',\n",
       " 'gse11864',\n",
       " 'gse11881',\n",
       " 'gse11886',\n",
       " 'gse118958',\n",
       " 'gse11919',\n",
       " 'gse24223',\n",
       " 'gse119416',\n",
       " 'gse12079',\n",
       " 'gse12108',\n",
       " 'gse12131',\n",
       " 'gse12187',\n",
       " 'gse12195',\n",
       " 'gse12198',\n",
       " 'gse12293',\n",
       " 'gse12366',\n",
       " 'gse12488',\n",
       " 'gse12507',\n",
       " 'gse12583',\n",
       " 'gse72642',\n",
       " 'gse25087',\n",
       " 'gse12662',\n",
       " 'gse12679',\n",
       " 'gse12734',\n",
       " 'gse12773',\n",
       " 'gse12806',\n",
       " 'gse12838',\n",
       " 'gse16385',\n",
       " 'gse12846',\n",
       " 'gse12875',\n",
       " 'gse12891',\n",
       " 'gse25638',\n",
       " 'gse12963',\n",
       " 'gse13054',\n",
       " 'gse13139',\n",
       " 'gse13210',\n",
       " 'gse13234',\n",
       " 'gse13296',\n",
       " 'gse13300',\n",
       " 'gse13330',\n",
       " 'gse26347',\n",
       " 'gse13501',\n",
       " 'gse26928',\n",
       " 'gse13606',\n",
       " 'gse13637',\n",
       " 'gse13712',\n",
       " 'gse13727',\n",
       " 'gse13732',\n",
       " 'gse13736',\n",
       " 'gse13738',\n",
       " 'gse13762',\n",
       " 'gse13791',\n",
       " 'gse13828',\n",
       " 'gse13837',\n",
       " 'gse13849',\n",
       " 'gse13887',\n",
       " 'gse13896',\n",
       " 'gse13899',\n",
       " 'gse13906',\n",
       " 'gse13909',\n",
       " 'gse13917',\n",
       " 'gse13931',\n",
       " 'gse13987',\n",
       " 'gse14000',\n",
       " 'gse14062',\n",
       " 'gse27928',\n",
       " 'gse28490',\n",
       " 'gse28491',\n",
       " 'gse14596',\n",
       " 'gse14924',\n",
       " 'gse14926',\n",
       " 'gse19246',\n",
       " 'gse15659',\n",
       " 'gse42203',\n",
       " 'gse56767',\n",
       " 'gse42253',\n",
       " 'gse16130',\n",
       " 'gse16461',\n",
       " 'gse65135',\n",
       " 'gse17186',\n",
       " 'gse17269',\n",
       " 'gse17549',\n",
       " 'gse19067',\n",
       " 'gse19069',\n",
       " 'gse31773',\n",
       " 'gse95298',\n",
       " 'gse47855',\n",
       " 'gse47920',\n",
       " 'gse9801',\n",
       " 'gse19599',\n",
       " 'gse49910',\n",
       " 'gse64555',\n",
       " 'gse51288',\n",
       " 'gse51997',\n",
       " 'gse67321',\n",
       " 'gse4588',\n",
       " 'gse67596',\n",
       " 'gse36907',\n",
       " 'gse56314',\n",
       " 'gse55673',\n",
       " 'gse33536',\n",
       " 'gse7307',\n",
       " 'gse26050',\n",
       " 'gse26156',\n",
       " 'gse26495',\n",
       " 'gse44126',\n",
       " 'gse59412',\n",
       " 'gse94510',\n",
       " 'gse28726',\n",
       " 'gse56313',\n",
       " 'gse41831',\n",
       " 'gse41889',\n",
       " 'gse29301',\n",
       " 'gse93984',\n",
       " 'gse60920',\n",
       " 'gse84331',\n",
       " 'gse85074',\n",
       " 'gse93683',\n",
       " 'gse42733',\n",
       " 'gse45535',\n",
       " 'gse76492',\n",
       " 'gse4527',\n",
       " 'gse59786',\n",
       " 'gse94801',\n",
       " 'gse61697',\n",
       " 'gse6743',\n",
       " 'gse83441',\n",
       " 'gse93776',\n",
       " 'gse74266',\n",
       " 'gse65010',\n",
       " 'gse47796',\n",
       " 'gse42672',\n",
       " 'gse71566',\n",
       " 'gse64028',\n",
       " 'gse98640',\n",
       " 'gse9891',\n",
       " 'gse50175',\n",
       " 'gse82159',\n",
       " 'gse58173',\n",
       " 'gse43769',\n",
       " 'gse79706',\n",
       " 'gse7753',\n",
       " 'gse59051',\n",
       " 'gse43346',\n",
       " 'gse73927',\n",
       " 'gse71730']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(map(lambda x: x.lower(), sampleMeta[sampleMeta['platform_id']=='GPL570']['gse'].unique()))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " ### GENERAL ATTRIBUTES ###\n",
    " \n",
    " __GPL570__\n",
    " - lots of GSEs\n",
    " - 22899 genes\n",
    " \n",
    " __GPL571__\n",
    " - fewer GSEs\n",
    " - 13277 genes\n",
    " \n",
    " __GPL6244__\n",
    " - 1 GSE\n",
    " \n",
    " __GPL96__\n",
    " - decent amount of GSEs\n",
    " - same gene # as GPL571, 13277\n",
    "     -- though some are in the 22000 range\n",
    " \n",
    " __GPL97__\n",
    " - handful of GSEs\n",
    " - also 13277\n",
    " \n",
    " ### COMBINATIONS ###\n",
    " \n",
    " - GPL97 and GPL96 *seem* to combine well\n",
    " - GPL96 and GPL571 *seem* to combine well\n",
    " - it appears that most--if not all--of the '13000' gene symbol sets are included in the 22000 gene symbol sets\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gse10021',\n",
       " 'gse10038',\n",
       " 'gse10040',\n",
       " 'gse10046',\n",
       " 'gse100786',\n",
       " 'gse10125',\n",
       " 'gse10147',\n",
       " 'gse101587',\n",
       " 'gse10172',\n",
       " 'gse10178',\n",
       " 'gse101855',\n",
       " 'gse102459',\n",
       " 'gse10315',\n",
       " 'gse10316',\n",
       " 'gse10325',\n",
       " 'gse103381',\n",
       " 'gse10358',\n",
       " 'gse10437',\n",
       " 'gse10463',\n",
       " 'gse10479',\n",
       " 'gse10524',\n",
       " 'gse105437',\n",
       " 'gse10586',\n",
       " 'gse10595',\n",
       " 'gse10612',\n",
       " 'gse106151',\n",
       " 'gse10631',\n",
       " 'gse106382',\n",
       " 'gse106532',\n",
       " 'gse10767',\n",
       " 'gse107683',\n",
       " 'gse10778',\n",
       " 'gse107818',\n",
       " 'gse10804',\n",
       " 'gse10821',\n",
       " 'gse10831',\n",
       " 'gse108384',\n",
       " 'gse10846',\n",
       " 'gse10880',\n",
       " 'gse10896',\n",
       " 'gse109178',\n",
       " 'gse10927',\n",
       " 'gse109348',\n",
       " 'gse10957',\n",
       " 'gse109662',\n",
       " 'gse109836',\n",
       " 'gse110023',\n",
       " 'gse110064',\n",
       " 'gse110085',\n",
       " 'gse110376',\n",
       " 'gse110545',\n",
       " 'gse11057',\n",
       " 'gse11083',\n",
       " 'gse11118',\n",
       " 'gse11151',\n",
       " 'gse11188',\n",
       " 'gse11190',\n",
       " 'gse112209',\n",
       " 'gse112210',\n",
       " 'gse112809',\n",
       " 'gse11281',\n",
       " 'gse11292',\n",
       " 'gse112953',\n",
       " 'gse11327',\n",
       " 'gse1133',\n",
       " 'gse11341',\n",
       " 'gse11342',\n",
       " 'gse113645',\n",
       " 'gse11373',\n",
       " 'gse11418',\n",
       " 'gse114181',\n",
       " 'gse11430',\n",
       " 'gse11450',\n",
       " 'gse116487',\n",
       " 'gse116660',\n",
       " 'gse116912',\n",
       " 'gse117139',\n",
       " 'gse11755',\n",
       " 'gse11761',\n",
       " 'gse11767',\n",
       " 'gse117676',\n",
       " 'gse11854',\n",
       " 'gse11864',\n",
       " 'gse11881',\n",
       " 'gse11886',\n",
       " 'gse118958',\n",
       " 'gse11909',\n",
       " 'gse11919',\n",
       " 'gse119416',\n",
       " 'gse11943',\n",
       " 'gse12079',\n",
       " 'gse12108',\n",
       " 'gse12131',\n",
       " 'gse12187',\n",
       " 'gse12195',\n",
       " 'gse12198',\n",
       " 'gse12246',\n",
       " 'gse12293',\n",
       " 'gse12366',\n",
       " 'gse12484',\n",
       " 'gse12488',\n",
       " 'gse12507',\n",
       " 'gse12583',\n",
       " 'gse12585',\n",
       " 'gse12586',\n",
       " 'gse12662',\n",
       " 'gse12668',\n",
       " 'gse12679',\n",
       " 'gse12734',\n",
       " 'gse12773',\n",
       " 'gse12803',\n",
       " 'gse12806',\n",
       " 'gse12838',\n",
       " 'gse12839',\n",
       " 'gse12845',\n",
       " 'gse12846',\n",
       " 'gse12875',\n",
       " 'gse12891',\n",
       " 'gse12907',\n",
       " 'gse12963',\n",
       " 'gse13017',\n",
       " 'gse13054',\n",
       " 'gse13139',\n",
       " 'gse13210',\n",
       " 'gse13234',\n",
       " 'gse13296',\n",
       " 'gse13300',\n",
       " 'gse13330',\n",
       " 'gse13411',\n",
       " 'gse13496',\n",
       " 'gse13501',\n",
       " 'gse13576',\n",
       " 'gse13591',\n",
       " 'gse13606',\n",
       " 'gse13637',\n",
       " 'gse13670',\n",
       " 'gse13712',\n",
       " 'gse13727',\n",
       " 'gse13732',\n",
       " 'gse13736',\n",
       " 'gse13738',\n",
       " 'gse13762',\n",
       " 'gse13791',\n",
       " 'gse13828',\n",
       " 'gse13837',\n",
       " 'gse13849',\n",
       " 'gse13850',\n",
       " 'gse13887',\n",
       " 'gse13896',\n",
       " 'gse13899',\n",
       " 'gse13906',\n",
       " 'gse13909',\n",
       " 'gse13917',\n",
       " 'gse13931',\n",
       " 'gse13987',\n",
       " 'gse13996',\n",
       " 'gse14000',\n",
       " 'gse14062',\n",
       " 'gse14230',\n",
       " 'gse14256',\n",
       " 'gse14330',\n",
       " 'gse14334',\n",
       " 'gse14335',\n",
       " 'gse14340',\n",
       " 'gse14386',\n",
       " 'gse14390',\n",
       " 'gse14405',\n",
       " 'gse14419',\n",
       " 'gse14465',\n",
       " 'gse14479',\n",
       " 'gse14577',\n",
       " 'gse14596',\n",
       " 'gse14630',\n",
       " 'gse14642',\n",
       " 'gse14711',\n",
       " 'gse14801',\n",
       " 'gse14816',\n",
       " 'gse14827',\n",
       " 'gse14895',\n",
       " 'gse14897',\n",
       " 'gse14924',\n",
       " 'gse14926',\n",
       " 'gse15072',\n",
       " 'gse15174',\n",
       " 'gse15215',\n",
       " 'gse15245',\n",
       " 'gse15322',\n",
       " 'gse15389',\n",
       " 'gse15400',\n",
       " 'gse15464',\n",
       " 'gse15490',\n",
       " 'gse15495',\n",
       " 'gse15499',\n",
       " 'gse15537',\n",
       " 'gse15615',\n",
       " 'gse15645',\n",
       " 'gse15659',\n",
       " 'gse15743',\n",
       " 'gse15777',\n",
       " 'gse15928',\n",
       " 'gse15949',\n",
       " 'gse16032',\n",
       " 'gse16051',\n",
       " 'gse16093',\n",
       " 'gse16098',\n",
       " 'gse16111',\n",
       " 'gse16129',\n",
       " 'gse16130',\n",
       " 'gse16214',\n",
       " 'gse16226',\n",
       " 'gse16250',\n",
       " 'gse16354',\n",
       " 'gse16355',\n",
       " 'gse16356',\n",
       " 'gse16382',\n",
       " 'gse16385',\n",
       " 'gse16386',\n",
       " 'gse16416',\n",
       " 'gse1644',\n",
       " 'gse16455',\n",
       " 'gse16461',\n",
       " 'gse16524',\n",
       " 'gse16547',\n",
       " 'gse16654',\n",
       " 'gse16683',\n",
       " 'gse16712',\n",
       " 'gse16715',\n",
       " 'gse16728',\n",
       " 'gse16746',\n",
       " 'gse16755',\n",
       " 'gse16791',\n",
       " 'gse16836',\n",
       " 'gse16837',\n",
       " 'gse16879',\n",
       " 'gse16962',\n",
       " 'gse16972',\n",
       " 'gse17046',\n",
       " 'gse17061',\n",
       " 'gse17078',\n",
       " 'gse17114',\n",
       " 'gse17119',\n",
       " 'gse17186',\n",
       " 'gse17189',\n",
       " 'gse17269',\n",
       " 'gse17299',\n",
       " 'gse17301',\n",
       " 'gse17306',\n",
       " 'gse17354',\n",
       " 'gse17356',\n",
       " 'gse17372',\n",
       " 'gse17407',\n",
       " 'gse17459',\n",
       " 'gse17477',\n",
       " 'gse17549',\n",
       " 'gse17718',\n",
       " 'gse17777',\n",
       " 'gse17938',\n",
       " 'gse17939',\n",
       " 'gse17974',\n",
       " 'gse18026',\n",
       " 'gse18090',\n",
       " 'gse18113',\n",
       " 'gse18147',\n",
       " 'gse18198',\n",
       " 'gse18226',\n",
       " 'gse18291',\n",
       " 'gse18309',\n",
       " 'gse18313',\n",
       " 'gse18323',\n",
       " 'gse18373',\n",
       " 'gse18497',\n",
       " 'gse18565',\n",
       " 'gse18593',\n",
       " 'gse18723',\n",
       " 'gse18791',\n",
       " 'gse18810',\n",
       " 'gse18864',\n",
       " 'gse18877',\n",
       " 'gse18913',\n",
       " 'gse19067',\n",
       " 'gse19069',\n",
       " 'gse19098',\n",
       " 'gse19128',\n",
       " 'gse19143',\n",
       " 'gse19147',\n",
       " 'gse19240',\n",
       " 'gse19243',\n",
       " 'gse19246',\n",
       " 'gse19301',\n",
       " 'gse19315',\n",
       " 'gse19475',\n",
       " 'gse19510',\n",
       " 'gse19556',\n",
       " 'gse19599',\n",
       " 'gse19615',\n",
       " 'gse19625',\n",
       " 'gse19627',\n",
       " 'gse19735',\n",
       " 'gse19784',\n",
       " 'gse19816',\n",
       " 'gse19882',\n",
       " 'gse19888',\n",
       " 'gse20086',\n",
       " 'gse20124',\n",
       " 'gse20125',\n",
       " 'gse20151',\n",
       " 'gse20198',\n",
       " 'gse20307',\n",
       " 'gse20484',\n",
       " 'gse20499',\n",
       " 'gse20504',\n",
       " 'gse20508',\n",
       " 'gse20538',\n",
       " 'gse20641',\n",
       " 'gse20666',\n",
       " 'gse20685',\n",
       " 'gse20745',\n",
       " 'gse20847',\n",
       " 'gse20978',\n",
       " 'gse20986',\n",
       " 'gse21039',\n",
       " 'gse21050',\n",
       " 'gse21132',\n",
       " 'gse21243',\n",
       " 'gse2125',\n",
       " 'gse21251',\n",
       " 'gse21252',\n",
       " 'gse21351',\n",
       " 'gse21521',\n",
       " 'gse21545',\n",
       " 'gse21547',\n",
       " 'gse21648',\n",
       " 'gse21723',\n",
       " 'gse21774',\n",
       " 'gse21837',\n",
       " 'gse21891',\n",
       " 'gse21899',\n",
       " 'gse21943',\n",
       " 'gse21989',\n",
       " 'gse22025',\n",
       " 'gse22045',\n",
       " 'gse22103',\n",
       " 'gse22166',\n",
       " 'gse22167',\n",
       " 'gse22183',\n",
       " 'gse22224',\n",
       " 'gse22246',\n",
       " 'gse22250',\n",
       " 'gse22255',\n",
       " 'gse22282',\n",
       " 'gse22325',\n",
       " 'gse22373',\n",
       " 'gse22392',\n",
       " 'gse22469',\n",
       " 'gse22470',\n",
       " 'gse22501',\n",
       " 'gse22522',\n",
       " 'gse22528',\n",
       " 'gse22529',\n",
       " 'gse22552',\n",
       " 'gse22589',\n",
       " 'gse22632',\n",
       " 'gse22667',\n",
       " 'gse22668',\n",
       " 'gse22688',\n",
       " 'gse22695',\n",
       " 'gse22759',\n",
       " 'gse22886',\n",
       " 'gse23371',\n",
       " 'gse24223',\n",
       " 'gse24634',\n",
       " 'gse25087',\n",
       " 'gse25638',\n",
       " 'gse26050',\n",
       " 'gse26156',\n",
       " 'gse26294',\n",
       " 'gse26347',\n",
       " 'gse26495',\n",
       " 'gse26928',\n",
       " 'gse27928',\n",
       " 'gse28079',\n",
       " 'gse28490',\n",
       " 'gse28491',\n",
       " 'gse28726',\n",
       " 'gse29301',\n",
       " 'gse31773',\n",
       " 'gse33536',\n",
       " 'gse35340',\n",
       " 'gse36767',\n",
       " 'gse36907',\n",
       " 'gse3920',\n",
       " 'gse39262',\n",
       " 'gse3982',\n",
       " 'gse41831',\n",
       " 'gse41889',\n",
       " 'gse42203',\n",
       " 'gse42253',\n",
       " 'gse4251',\n",
       " 'gse42672',\n",
       " 'gse42733',\n",
       " 'gse43346',\n",
       " 'gse43677',\n",
       " 'gse43769',\n",
       " 'gse44126',\n",
       " 'gse44164',\n",
       " 'gse4475',\n",
       " 'gse4527',\n",
       " 'gse45535',\n",
       " 'gse4571',\n",
       " 'gse4588',\n",
       " 'gse46913',\n",
       " 'gse46917',\n",
       " 'gse47796',\n",
       " 'gse47855',\n",
       " 'gse47920',\n",
       " 'gse49910',\n",
       " 'gse49954',\n",
       " 'gse50175',\n",
       " 'gse5099',\n",
       " 'gse51288',\n",
       " 'gse51382',\n",
       " 'gse51997',\n",
       " 'gse54017',\n",
       " 'gse55673',\n",
       " 'gse5580',\n",
       " 'gse56313',\n",
       " 'gse56314',\n",
       " 'gse56767',\n",
       " 'gse57611',\n",
       " 'gse58173',\n",
       " 'gse59051',\n",
       " 'gse59412',\n",
       " 'gse59786',\n",
       " 'gse6053',\n",
       " 'gse60920',\n",
       " 'gse61697',\n",
       " 'gse62584',\n",
       " 'gse64028',\n",
       " 'gse64555',\n",
       " 'gse65010',\n",
       " 'gse65135',\n",
       " 'gse65856',\n",
       " 'gse67321',\n",
       " 'gse6740',\n",
       " 'gse6743',\n",
       " 'gse67596',\n",
       " 'gse7138',\n",
       " 'gse71566',\n",
       " 'gse71730',\n",
       " 'gse72642',\n",
       " 'gse7307',\n",
       " 'gse73927',\n",
       " 'gse74266',\n",
       " 'gse76492',\n",
       " 'gse7753',\n",
       " 'gse79706',\n",
       " 'gse82159',\n",
       " 'gse8286',\n",
       " 'gse83441',\n",
       " 'gse84331',\n",
       " 'gse85074',\n",
       " 'gse8835',\n",
       " 'gse8921',\n",
       " 'gse93683',\n",
       " 'gse93776',\n",
       " 'gse9378',\n",
       " 'gse93984',\n",
       " 'gse94510',\n",
       " 'gse94801',\n",
       " 'gse95298',\n",
       " 'gse9801',\n",
       " 'gse98640',\n",
       " 'gse9891',\n",
       " 'gse9946',\n",
       " 'master',\n",
       " 'master_test']"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "engine.table_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22899"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len(gse_df['gene_symbol_sql'])\n",
    "gse_df = pd.read_sql_table('gse10021', con=engine)\n",
    "geneSymbolList = set(gse_df['gene_symbol_sql'])\n",
    "len(geneSymbolList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gse10021\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10038\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10040\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10046\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse100786\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse101587\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10172\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10178\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse101855\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse102459\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10315\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10316\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10325\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse103381\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10358\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10437\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10463\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10479\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10524\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse105437\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10586\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10595\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10612\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10631\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106532\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse107683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10778\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse107818\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10804\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10821\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10831\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse108384\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10846\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10880\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10896\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109178\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10927\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109348\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10957\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109662\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109836\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110023\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110064\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110085\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110376\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110545\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11057\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11083\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11118\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11188\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11190\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112209\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112210\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112809\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11281\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11292\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112953\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11327\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse1133\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11341\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11342\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse113645\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11418\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse114181\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11430\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11450\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116487\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116660\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116912\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse117139\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11755\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11761\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse117676\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11854\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11864\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11881\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11886\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse118958\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11909\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11919\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse119416\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11943\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12079\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12108\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12131\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12187\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12195\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12293\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12366\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12484\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12488\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12507\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12583\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12585\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12586\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12662\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12668\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12679\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12734\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12773\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12803\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12806\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12838\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12839\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12845\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12846\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12875\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12907\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12963\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13017\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13054\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13139\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13210\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13234\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13296\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13300\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13330\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13411\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13496\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13501\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13576\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13591\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13606\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13637\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13670\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13712\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13727\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13732\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13736\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13738\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13762\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13828\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13849\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13850\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13887\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13896\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13899\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13906\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13909\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13917\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13931\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13987\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13996\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14000\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14062\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14230\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14256\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14330\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14334\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14335\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14340\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14386\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14390\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14405\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14419\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14465\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14479\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14577\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14596\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14630\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14642\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14711\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14816\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14827\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14895\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14897\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14924\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14926\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15072\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15174\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15215\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15245\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15322\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15389\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15400\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15464\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15490\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15495\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15499\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15537\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15615\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15645\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15659\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15743\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15777\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15949\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16032\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16051\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16093\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16098\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16111\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16129\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16130\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16214\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16226\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16250\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16354\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16355\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16356\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16385\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16386\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16416\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse1644\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16455\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16461\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16524\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16547\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16654\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16712\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16715\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16728\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16746\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16755\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16836\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16879\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16962\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16972\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17046\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17061\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17078\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17114\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17119\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17186\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17189\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17269\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17299\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17301\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17306\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17354\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17356\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17372\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17407\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17459\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17477\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17549\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17718\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17777\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17938\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17939\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17974\n",
      "not in genesymbollist:\n",
      "3707\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18026\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18090\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18113\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18226\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18291\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18309\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18313\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18323\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18497\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18565\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18593\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18723\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18810\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18864\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18877\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18913\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19067\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19069\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19098\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19128\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19143\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19240\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19243\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19301\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19315\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19475\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19510\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19556\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19599\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19615\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19625\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19627\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19735\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19784\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19816\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19882\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19888\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20086\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20124\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20307\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20484\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20499\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20504\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20508\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20538\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20641\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20666\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20685\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20745\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20847\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20978\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20986\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21039\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21050\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21132\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21243\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse2125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21251\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21252\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21351\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21521\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21545\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21547\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21648\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21723\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21774\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21899\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21943\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21989\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22025\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22045\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22103\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22166\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22167\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22183\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22224\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22250\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22255\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22282\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22325\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22392\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22469\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22470\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22501\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22522\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22528\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22529\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22552\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22589\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22632\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22667\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22668\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22688\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22695\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22759\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22886\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse23371\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse24223\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse24634\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse25087\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse25638\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26050\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26156\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26294\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26347\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26495\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse27928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28079\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28490\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28491\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28726\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse29301\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse31773\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse33536\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse35340\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse36767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse36907\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse3920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse39262\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse3982\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse41831\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse41889\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42203\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42253\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4251\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42672\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42733\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43346\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43677\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43769\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse44126\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse44164\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4475\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4527\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse45535\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4571\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4588\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse46913\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse46917\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47796\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47855\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse49910\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse49954\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse50175\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse5099\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51288\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51997\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse54017\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse55673\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse5580\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56313\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56314\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse57611\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse58173\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59051\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59412\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59786\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6053\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse60920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse61697\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse62584\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse64028\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse64555\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65010\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65135\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65856\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse67321\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6740\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6743\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse67596\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7138\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse71566\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse71730\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse72642\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7307\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse73927\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse74266\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse76492\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7753\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse79706\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse82159\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8286\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse83441\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse84331\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse85074\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8835\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8921\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93776\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9378\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93984\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse94510\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse94801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse95298\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse98640\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9946\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n"
     ]
    }
   ],
   "source": [
    "gselist = [x for x in engine.table_names() if 'gse' in x]\n",
    "# geneSymbolList was initialized above\n",
    "for gse in gselist:\n",
    "    print(gse)\n",
    "    gse_df = pd.read_sql_table(gse, con=engine)\n",
    "    gse_gs = set(gse_df['gene_symbol_sql'])\n",
    "    # how many gene symbols are not in geneSymbolList?\n",
    "    new_geneSymbols = gse_gs.difference(geneSymbolList)\n",
    "    print(\"not in genesymbollist:\")\n",
    "    print(len(new_geneSymbols))\n",
    "    # how much bigger does geneSymbolList get after union?\n",
    "    geneSymbolList = geneSymbolList.union(gse_gs)\n",
    "    print(\"new geneSymbolList length:\")\n",
    "    print(len(geneSymbolList))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/allGeneSymbols.pckl', 'wb') as f:\n",
    "    pickle.dump(geneSymbolList, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. LOAD THE GENE SYMBOL PICKLE FILE\n",
    "    - sampleMeta\n",
    "        - dammit, cell type has weird capitalization\n",
    "    - gse_training\n",
    "\n",
    "2. MAKE A CELL TYPE DF\n",
    "    - try with 'neutrophils'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/allGeneSymbols.pckl', 'rb') as f:\n",
    "    geneSymbolList = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/ctDict.pckl', 'rb') as f:\n",
    "    ctDict = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'CD4.T.cells': ['memory_cd4_t_cells',\n",
       "  'th17_cells',\n",
       "  'naive_cd4_t_cells',\n",
       "  'th1_cells',\n",
       "  'th2_cells',\n",
       "  'cd4_t_cells',\n",
       "  'tregs',\n",
       "  'central_memory_cd4_t_cells',\n",
       "  'effector_memory_cd4_t_cells',\n",
       "  'follicular_helper_t_cells',\n",
       "  'th1th17_cells',\n",
       "  'effector_cd4_t_cells'],\n",
       " 'CD8.T.cells': ['cd8_t_cells',\n",
       "  'naive_cd8_t_cells',\n",
       "  'effector_memory_cd8_t_cells',\n",
       "  'memory_cd8_t_cells',\n",
       "  'central_memory_cd8_t_cells'],\n",
       " 'NK.cells': ['nk_cells'],\n",
       " 'B.cells': ['b_cells',\n",
       "  'naive_b_cells',\n",
       "  'gc_b_cells',\n",
       "  'memory_b_cells',\n",
       "  'immature_b_cells',\n",
       "  'activated_b_cells',\n",
       "  'plasma_b_cells',\n",
       "  'resting_b_cells',\n",
       "  'pre-gc_b_cells'],\n",
       " 'monocytic.lineage': ['monocytes',\n",
       "  'dendritic_cells',\n",
       "  'macrophage',\n",
       "  'immature_dendritic_cells',\n",
       "  'myeloid_dendritic_cells',\n",
       "  'activated_macrophages'],\n",
       " 'neutrophils': ['neutrophils', 'granulocytes'],\n",
       " 'endothelial.cells': ['endothelial'],\n",
       " 'fibroblasts': ['fibroblast']}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ctDict['Coarse']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>platform_id</th>\n",
       "      <th>gse</th>\n",
       "      <th>file</th>\n",
       "      <th>id</th>\n",
       "      <th>col</th>\n",
       "      <th>pattern</th>\n",
       "      <th>hit</th>\n",
       "      <th>cell.type</th>\n",
       "      <th>val</th>\n",
       "      <th>technology</th>\n",
       "      <th>title</th>\n",
       "      <th>normalization</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3529</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025152</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3530</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025153</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3531</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025154</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3532</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025155</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3533</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025156</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3534</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025157</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3535</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025158</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3536</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025159</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3537</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025160</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3538</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025161</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025162</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3540</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025163</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3791</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE45535</td>\n",
       "      <td>GSE45535_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1108140</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3792</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE45535</td>\n",
       "      <td>GSE45535_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1108146</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4437</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025135</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4438</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025142</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4439</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025151</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4488</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025134</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4489</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025136</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4490</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025137</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4563</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025138</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4564</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025139</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4565</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025146</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4611</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025140</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4612</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025143</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4613</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025144</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4614</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025145</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4615</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025147</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4616</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025148</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4667</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025129</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4668</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025130</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4669</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025133</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4670</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025141</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4748</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025131</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4749</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025132</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4753</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025150</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4833</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025149</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     platform_id       gse                                    file  \\\n",
       "3529      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3530      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3531      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3532      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3533      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3534      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3535      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3536      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3537      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3538      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3539      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3540      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3791      GPL570  GSE45535  GSE45535_series_matrix.txt.gz.anno.tsv   \n",
       "3792      GPL570  GSE45535  GSE45535_series_matrix.txt.gz.anno.tsv   \n",
       "4437      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4438      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4439      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4488      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4489      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4490      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4563      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4564      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4565      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4611      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4612      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4613      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4614      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4615      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4616      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4667      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4668      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4669      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4670      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4748      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4749      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4753      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4833      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "\n",
       "              id                            col                      pattern  \\\n",
       "3529  GSM1025152  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3530  GSM1025153  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3531  GSM1025154  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3532  GSM1025155  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3533  GSM1025156  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3534  GSM1025157  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3535  GSM1025158  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3536  GSM1025159  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3537  GSM1025160  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3538  GSM1025161  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3539  GSM1025162  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3540  GSM1025163  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3791  GSM1108140  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3792  GSM1108146  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4437  GSM1025135  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4438  GSM1025142  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4439  GSM1025151  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4488  GSM1025134  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4489  GSM1025136  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4490  GSM1025137  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4563  GSM1025138  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4564  GSM1025139  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4565  GSM1025146  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4611  GSM1025140  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4612  GSM1025143  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4613  GSM1025144  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4614  GSM1025145  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4615  GSM1025147  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4616  GSM1025148  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4667  GSM1025129  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4668  GSM1025130  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4669  GSM1025133  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4670  GSM1025141  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4748  GSM1025131  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4749  GSM1025132  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4753  GSM1025150  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4833  GSM1025149  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "\n",
       "               hit     cell.type           val               technology  \\\n",
       "3529  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3530  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3531  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3532  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3533  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3534  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3535  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3536  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3537  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3538  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3539  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3540  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3791  Granulocytes  Granulocytes  Granulocytes  in situ oligonucleotide   \n",
       "3792  Granulocytes  Granulocytes  Granulocytes  in situ oligonucleotide   \n",
       "4437  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4438  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4439  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4488  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4489  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4490  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4563  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4564  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4565  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4611  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4612  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4613  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4614  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4615  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4616  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4667  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4668  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4669  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4670  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4748  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4749  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4753  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4833  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "\n",
       "                                                  title normalization  \n",
       "3529  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3530  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3531  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3532  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3533  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3534  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3535  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3536  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3537  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3538  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3539  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3540  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3791  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...       unknown  \n",
       "3792  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...       unknown  \n",
       "4437  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4438  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4439  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4488  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4489  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4490  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4563  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4564  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4565  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4611  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4612  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4613  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4614  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4615  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4616  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4667  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4668  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4669  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4670  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4748  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4749  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4753  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4833  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get neutrophil GSEs\n",
    "sampleMeta[sampleMeta['cell.type']=='Granulocytes']\n",
    "# GSE11083\n",
    "# - get relevant GSMs from GSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_df(gse, sub_ct):\n",
    "    print(gse)\n",
    "    df = pd.read_sql_table(gse.lower(), con=engine)\n",
    "    df.set_index('gene_symbol_sql', inplace=True)\n",
    "    gsm_idx = sampleMeta[sampleMeta['cell.type']==sub_ct][sampleMeta['gse']==gse]['id']\n",
    "    #apparently GSMs in sampleMeta are not present in some GSE dfs\n",
    "    gsm_idx = [g for g in gsm_idx if g in df.columns]\n",
    "    df = df[gsm_idx].T\n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CD4.T.cells\n",
      "GSE36767\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  \"\"\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSE26928\n",
      "GSE15659\n",
      "GSE44126\n",
      "GSE60920\n",
      "GSE22886\n",
      "6\n",
      "GSE110545\n",
      "1\n",
      "GSE22045\n",
      "GSE109348\n",
      "GSE25087\n",
      "GSE26928\n",
      "GSE13732\n",
      "GSE94510\n",
      "GSE71566\n",
      "GSE93776\n",
      "GSE13017\n",
      "9\n",
      "GSE50175\n",
      "GSE43769\n",
      "GSE22886\n",
      "GSE3982\n",
      "4\n",
      "GSE109348\n",
      "GSE43769\n",
      "GSE22886\n",
      "GSE3982\n",
      "4\n",
      "GSE10437\n",
      "GSE36767\n",
      "GSE11292\n",
      "GSE117139\n",
      "GSE119416\n",
      "GSE12079\n",
      "GSE12488\n",
      "GSE72642\n",
      "GSE12875\n",
      "GSE12963\n",
      "GSE27928\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE14924\n",
      "GSE14926\n",
      "GSE15659\n",
      "GSE42253\n",
      "GSE16130\n",
      "GSE16461\n",
      "GSE19069\n",
      "GSE31773\n",
      "GSE47920\n",
      "GSE49954\n",
      "GSE51288\n",
      "GSE67321\n",
      "GSE4588\n",
      "GSE7307\n",
      "GSE49910\n",
      "GSE26156\n",
      "GSE28726\n",
      "GSE60920\n",
      "GSE4527\n",
      "GSE59786\n",
      "GSE71566\n",
      "GSE51997\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE73927\n",
      "GSE14330\n",
      "GSE62584\n",
      "GSE22886\n",
      "GSE8835\n",
      "GSE10325\n",
      "GSE1133\n",
      "GSE4571\n",
      "GSE6740\n",
      "GSE17354\n",
      "GSE46917\n",
      "48\n",
      "GSE10586\n",
      "GSE22045\n",
      "GSE11292\n",
      "GSE25087\n",
      "GSE13234\n",
      "GSE15659\n",
      "GSE59786\n",
      "GSE65010\n",
      "GSE73927\n",
      "GSE14330\n",
      "GSE24634\n",
      "GSE13017\n",
      "GSE4571\n",
      "GSE4527\n",
      "14\n",
      "GSE26928\n",
      "GSE94510\n",
      "2\n",
      "GSE26928\n",
      "GSE94510\n",
      "2\n",
      "GSE73927\n",
      "1\n",
      "GSE50175\n",
      "1\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:11: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(890, 22899)\n",
      "(650, 16124)\n",
      "CD8.T.cells\n",
      "GSE49954\n",
      "GSE10178\n",
      "GSE109348\n",
      "GSE72642\n",
      "GSE26347\n",
      "GSE27928\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE14924\n",
      "GSE14926\n",
      "GSE42253\n",
      "GSE16130\n",
      "GSE16461\n",
      "GSE19069\n",
      "GSE31773\n",
      "GSE47920\n",
      "GSE49910\n",
      "GSE51288\n",
      "GSE67321\n",
      "GSE44126\n",
      "GSE84331\n",
      "GSE93776\n",
      "GSE98640\n",
      "GSE85074\n",
      "GSE7307\n",
      "GSE58173\n",
      "GSE62584\n",
      "GSE8835\n",
      "GSE6740\n",
      "GSE11188\n",
      "GSE1133\n",
      "GSE22886\n",
      "GSE17354\n",
      "GSE10631\n",
      "34\n",
      "GSE26347\n",
      "GSE26495\n",
      "GSE84331\n",
      "GSE85074\n",
      "GSE93683\n",
      "GSE98640\n",
      "6\n",
      "GSE85074\n",
      "GSE93683\n",
      "GSE98640\n",
      "3\n",
      "GSE93683\n",
      "1\n",
      "GSE93683\n",
      "GSE85074\n",
      "GSE98640\n",
      "3\n",
      "(423, 22899)\n",
      "(316, 17099)\n",
      "NK.cells\n",
      "GSE10358\n",
      "GSE10927\n",
      "GSE109348\n",
      "GSE11373\n",
      "GSE116487\n",
      "GSE116660\n",
      "GSE12198\n",
      "GSE72642\n",
      "GSE13501\n",
      "GSE13849\n",
      "GSE42253\n",
      "GSE19067\n",
      "GSE26050\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE47855\n",
      "GSE9891\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE7753\n",
      "GSE22886\n",
      "GSE4475\n",
      "GSE10957\n",
      "GSE39262\n",
      "GSE3982\n",
      "GSE57611\n",
      "26\n",
      "(197, 22899)\n",
      "(110, 17096)\n",
      "B.cells\n",
      "GSE51382\n",
      "GSE65856\n",
      "GSE10524\n",
      "GSE106382\n",
      "GSE10767\n",
      "GSE107683\n",
      "GSE10846\n",
      "GSE109348\n",
      "GSE110023\n",
      "GSE110064\n",
      "GSE110376\n",
      "GSE112953\n",
      "GSE24223\n",
      "GSE12195\n",
      "GSE72642\n",
      "GSE12734\n",
      "GSE13210\n",
      "GSE13300\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE42203\n",
      "GSE65135\n",
      "GSE17186\n",
      "GSE17269\n",
      "GSE19599\n",
      "GSE36907\n",
      "GSE56314\n",
      "GSE7307\n",
      "GSE49910\n",
      "GSE25638\n",
      "GSE4588\n",
      "GSE56313\n",
      "GSE29301\n",
      "GSE93984\n",
      "GSE45535\n",
      "GSE59412\n",
      "GSE83441\n",
      "GSE74266\n",
      "GSE64028\n",
      "GSE82159\n",
      "GSE67321\n",
      "GSE58173\n",
      "GSE94801\n",
      "GSE71730\n",
      "GSE13996\n",
      "GSE62584\n",
      "GSE10325\n",
      "GSE43677\n",
      "GSE44164\n",
      "GSE13850\n",
      "GSE12845\n",
      "GSE28079\n",
      "GSE12668\n",
      "GSE3982\n",
      "GSE57611\n",
      "GSE9378\n",
      "56\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE13300\n",
      "GSE13917\n",
      "GSE17186\n",
      "GSE17269\n",
      "GSE36907\n",
      "GSE25638\n",
      "GSE56314\n",
      "GSE64028\n",
      "GSE22886\n",
      "GSE43677\n",
      "GSE12845\n",
      "GSE13411\n",
      "14\n",
      "GSE10821\n",
      "GSE10831\n",
      "GSE110376\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE19246\n",
      "GSE42203\n",
      "GSE56313\n",
      "GSE93984\n",
      "GSE64555\n",
      "GSE74266\n",
      "GSE83441\n",
      "GSE4475\n",
      "GSE44164\n",
      "GSE10172\n",
      "GSE57611\n",
      "GSE43677\n",
      "GSE12845\n",
      "18\n",
      "GSE51382\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE17186\n",
      "GSE25638\n",
      "GSE56314\n",
      "GSE22886\n",
      "GSE13411\n",
      "8\n",
      "GSE19599\n",
      "1\n",
      "GSE42203\n",
      "GSE19246\n",
      "GSE64555\n",
      "GSE54017\n",
      "4\n",
      "GSE12366\n",
      "GSE13411\n",
      "2\n",
      "GSE13987\n",
      "GSE54017\n",
      "2\n",
      "GSE12845\n",
      "1\n",
      "(1424, 22899)\n",
      "(938, 16206)\n",
      "monocytic.lineage\n",
      "GSE100786\n",
      "GSE101587\n",
      "GSE103381\n",
      "GSE10595\n",
      "GSE22373\n",
      "GSE10896\n",
      "GSE109348\n",
      "GSE113645\n",
      "GSE114181\n",
      "GSE11430\n",
      "GSE23371\n",
      "GSE116487\n",
      "GSE11755\n",
      "GSE11864\n",
      "GSE24223\n",
      "GSE12108\n",
      "GSE12773\n",
      "GSE12838\n",
      "GSE16385\n",
      "GSE13762\n",
      "GSE13899\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE56767\n",
      "GSE17549\n",
      "GSE9801\n",
      "GSE49910\n",
      "GSE51997\n",
      "GSE72642\n",
      "GSE7307\n",
      "GSE41889\n",
      "GSE45535\n",
      "GSE67321\n",
      "GSE93776\n",
      "GSE79706\n",
      "GSE42733\n",
      "GSE58173\n",
      "GSE43346\n",
      "GSE71730\n",
      "GSE10612\n",
      "GSE62584\n",
      "GSE5580\n",
      "GSE11943\n",
      "GSE5099\n",
      "GSE1133\n",
      "GSE16972\n",
      "GSE7138\n",
      "GSE12803\n",
      "GSE46917\n",
      "GSE22886\n",
      "GSE8286\n",
      "GSE9378\n",
      "GSE46913\n",
      "GSE8921\n",
      "54\n",
      "GSE101855\n",
      "GSE10316\n",
      "GSE10463\n",
      "GSE112209\n",
      "GSE112210\n",
      "GSE11327\n",
      "GSE12806\n",
      "GSE13296\n",
      "GSE13762\n",
      "GSE14000\n",
      "GSE22886\n",
      "GSE9946\n",
      "GSE8921\n",
      "GSE3982\n",
      "14\n",
      "GSE10038\n",
      "GSE13670\n",
      "GSE22373\n",
      "GSE11430\n",
      "GSE11886\n",
      "GSE16385\n",
      "GSE13896\n",
      "GSE13931\n",
      "GSE9801\n",
      "GSE41889\n",
      "GSE94801\n",
      "GSE79706\n",
      "GSE43346\n",
      "GSE26294\n",
      "GSE5099\n",
      "GSE16972\n",
      "GSE7138\n",
      "GSE22886\n",
      "GSE3982\n",
      "GSE8286\n",
      "20\n",
      "GSE10316\n",
      "GSE13762\n",
      "GSE23371\n",
      "GSE3982\n",
      "GSE9946\n",
      "5\n",
      "GSE35340\n",
      "GSE109348\n",
      "GSE11151\n",
      "GSE114181\n",
      "GSE28490\n",
      "GSE46913\n",
      "6\n",
      "GSE5099\n",
      "1\n",
      "(1108, 22899)\n",
      "(949, 17198)\n",
      "neutrophils\n",
      "GSE109348\n",
      "GSE11083\n",
      "GSE72642\n",
      "GSE12662\n",
      "GSE12838\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE67596\n",
      "GSE49910\n",
      "GSE76492\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE12803\n",
      "GSE12484\n",
      "GSE22886\n",
      "GSE3982\n",
      "16\n",
      "GSE41831\n",
      "GSE45535\n",
      "2\n",
      "(192, 22899)\n",
      "(171, 17198)\n",
      "endothelial.cells\n",
      "GSE10479\n",
      "GSE105437\n",
      "GSE106532\n",
      "GSE108384\n",
      "GSE109178\n",
      "GSE109662\n",
      "GSE109836\n",
      "GSE110085\n",
      "GSE117676\n",
      "GSE12131\n",
      "GSE12293\n",
      "GSE12679\n",
      "GSE12846\n",
      "GSE12891\n",
      "GSE13054\n",
      "GSE13139\n",
      "GSE13637\n",
      "GSE13712\n",
      "GSE13736\n",
      "GSE13791\n",
      "GSE7307\n",
      "GSE59412\n",
      "GSE95298\n",
      "GSE47796\n",
      "GSE42672\n",
      "GSE9891\n",
      "GSE12246\n",
      "GSE10804\n",
      "GSE12907\n",
      "GSE11341\n",
      "GSE3920\n",
      "GSE1133\n",
      "GSE10778\n",
      "GSE14230\n",
      "GSE9378\n",
      "35\n",
      "(363, 22899)\n",
      "(284, 19228)\n",
      "fibroblasts\n",
      "GSE10046\n",
      "GSE10315\n",
      "GSE106151\n",
      "GSE106382\n",
      "GSE22167\n",
      "GSE10880\n",
      "GSE110085\n",
      "GSE11418\n",
      "GSE116912\n",
      "GSE11854\n",
      "GSE118958\n",
      "GSE11919\n",
      "GSE12583\n",
      "GSE13330\n",
      "GSE13606\n",
      "GSE13791\n",
      "GSE13828\n",
      "GSE13837\n",
      "GSE17549\n",
      "GSE95298\n",
      "GSE55673\n",
      "GSE33536\n",
      "GSE47796\n",
      "GSE82159\n",
      "GSE59051\n",
      "GSE42672\n",
      "GSE11767\n",
      "GSE26294\n",
      "GSE14256\n",
      "GSE10125\n",
      "GSE11450\n",
      "GSE3920\n",
      "GSE39262\n",
      "33\n",
      "(320, 22899)\n",
      "(285, 18471)\n"
     ]
    }
   ],
   "source": [
    "# format the first dataframe\n",
    "for cell in ctDict['Coarse'].keys():\n",
    "    print(cell)\n",
    "    gse_dfs=[]\n",
    "    for sub_ct in ctDict['Coarse'][cell]:\n",
    "        gse_list = sampleMeta[sampleMeta['cell.type']==sub_ct]['gse'].unique()\n",
    "        gse_dfs.extend(list(map(lambda x: format_df(x, sub_ct), gse_list)))\n",
    "        print(len(gse_list))\n",
    "    #     gse_dfs[:] = [g for g in gse_dfs if g != None]\n",
    "\n",
    "    ct_master = pd.concat(gse_dfs)\n",
    "    print(ct_master.shape)\n",
    "    \n",
    "    # drop bad samples... \n",
    "    gsm_nas = ct_master.apply(lambda x: sum(np.isnan(x)),1)\n",
    "    gsm_idx = gsm_nas[gsm_nas < 5000].index\n",
    "    # drop the bad genes\n",
    "    GS_nas = ct_master.loc[gsm_idx].apply(lambda x: sum(np.isnan(x)))\n",
    "    GS_idx = GS_nas[GS_nas == 0].index\n",
    "\n",
    "    ct_filtered = ct_master.loc[gsm_idx, GS_idx]\n",
    "    print(ct_filtered.shape)\n",
    "    \n",
    "    filepath = ntpath.join('picklejar',cell+'.pckl')\n",
    "    with open(filepath, 'wb') as f:\n",
    "        pickle.dump(ct_filtered, f)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM253929        3.903142  4.433145  3.256784  4.259730  3.197122  5.544762   \n",
       " GSM253930        3.778336  4.315451  3.347470  4.347301  3.266853  5.629200   \n",
       " GSM253931        3.921799  4.612568  3.509739  4.568272  3.368932  5.721997   \n",
       " GSM253932        3.857394  4.060022  3.507301  4.654821  3.052071  7.164087   \n",
       " GSM253933        3.749314  3.742328  3.230485  4.301812  2.869478  6.984066   \n",
       " GSM253934        4.011793  4.098360  3.212182  4.320827  2.872746  6.755912   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM253929        3.616787  7.039044  4.753434  5.337283  ...  6.806036   \n",
       " GSM253930        3.860964  7.023954  4.681677  5.269308  ...  6.608633   \n",
       " GSM253931        4.035671  7.056585  4.909189  5.637331  ...  6.863645   \n",
       " GSM253932        4.194234  7.109719  4.836110  5.824080  ...  7.501342   \n",
       " GSM253933        3.798506  7.076272  4.580599  5.378697  ...  7.292929   \n",
       " GSM253934        3.926874  6.914391  4.723711  5.351512  ...  7.032515   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM253929        4.590806  4.932780  5.891560  4.584579  6.759235  7.351205   \n",
       " GSM253930        4.832765  5.155579  6.128058  4.955600  6.512541  7.387762   \n",
       " GSM253931        4.605344  5.198019  5.815030  5.049368  6.234872  7.460859   \n",
       " GSM253932        4.665013  5.524819  5.920308  5.041664  6.442296  7.612078   \n",
       " GSM253933        4.806935  5.319700  6.180013  4.669292  6.976984  7.490245   \n",
       " GSM253934        4.590806  4.991948  5.844428  5.067704  6.760124  7.543675   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM253929        5.720198  7.563352  3.136274  \n",
       " GSM253930        5.999456  7.698567  3.100024  \n",
       " GSM253931        5.890555  7.353964  3.223108  \n",
       " GSM253932        6.097124  7.222270  3.055547  \n",
       " GSM253933        6.056153  7.591597  2.915601  \n",
       " GSM253934        5.979775  7.383300  2.930434  \n",
       " \n",
       " [6 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM260657        6.133399  5.108524  5.232569  5.668119  2.608809  4.259779   \n",
       " GSM260658        6.011227  6.704595  5.370312  5.464877  5.825277  3.968731   \n",
       " GSM260659        5.827819  3.201634  4.743719  5.304737  5.436295  4.558640   \n",
       " GSM260660        4.240314  4.892391  4.897452  5.001330  5.419539  6.050489   \n",
       " GSM260661        5.902074  5.749534  4.572074  5.917820  4.439623  4.384621   \n",
       " GSM260662        4.872829  4.104337  3.972371  5.529608  1.000000  3.520391   \n",
       " GSM260663        5.911692  6.083213  5.350025  4.719105  4.008989  3.988239   \n",
       " GSM260664        4.842979  3.292782  4.796322  6.193713  4.350497  4.645908   \n",
       " GSM260665        6.525129  5.909293  5.249828  5.861506  4.584963  4.984708   \n",
       " GSM260666        5.963474  6.171927  4.572686  6.421175  4.666757  4.189776   \n",
       " GSM260667        5.746850  2.744161  4.831379  5.684390  4.832890  4.277390   \n",
       " GSM260668        6.619120  5.022368  4.643810  5.643983  4.749534  5.519610   \n",
       " GSM260669        6.011227  4.439623  5.285170  6.039804  5.371559  4.078320   \n",
       " GSM260670        6.221104  2.459432  2.347997  6.217627  5.022368  1.903677   \n",
       " GSM260671        6.711495  6.520422  3.905736  6.522325  5.336283  3.732987   \n",
       " GSM260672        5.311067  5.730640  5.592713  6.373589  5.336283  4.687172   \n",
       " GSM260673        5.439623  3.232661  3.166175  6.268914  5.087463  3.464007   \n",
       " GSM260674        6.053111  5.459432  3.384092  8.264034  4.392317  4.927434   \n",
       " GSM260675        6.053111  3.307429  2.919728  5.478674  4.498251  3.004382   \n",
       " GSM260676        6.985273  3.307429  4.870008  5.488819  5.137504  3.109197   \n",
       " GSM260677        4.649615  3.017922  5.184962  5.324448  3.678072  4.140478   \n",
       " GSM260678        7.192786  3.689299  3.248327  7.403933  4.104337  4.458029   \n",
       " GSM260679        7.348728  3.104337  4.086289  8.151328  5.112700  4.835417   \n",
       " GSM260680        4.307429  5.611763  3.867355  7.261480  4.240314  1.998194   \n",
       " GSM260681        6.081084  3.087463  3.731353  7.895408  5.303781  4.577029   \n",
       " GSM260682        6.311067  5.368070  3.912002  8.205027  5.832890  4.415812   \n",
       " GSM260683        5.928370  2.786596  3.764129  7.305356  3.548437  4.320303   \n",
       " GSM260684        5.830357  2.786596  4.454947  7.868260  4.765535  2.258953   \n",
       " GSM260685        6.399171  5.478972  4.351156  7.688060  6.806066  4.837908   \n",
       " GSM260686        5.318317  2.536053  4.829962  6.638906  5.548437  3.522634   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM260657        3.765535  5.638074  5.727920  6.424586  ...  8.450386   \n",
       " GSM260658        1.378512  5.638074  5.091700  5.937815  ...  7.918863   \n",
       " GSM260659        5.087463  3.744161  5.149747  4.307429  ...  7.994353   \n",
       " GSM260660        2.944858  5.944858  5.263034  5.807355  ...  8.853310   \n",
       " GSM260661        3.169925  5.442943  5.429616  5.224966  ...  9.190072   \n",
       " GSM260662        3.392317  5.560715  4.906891  5.439623  ...  7.806711   \n",
       " GSM260663        2.981853  5.488644  3.925999  6.526695  ...  7.738092   \n",
       " GSM260664        2.378512  6.143638  5.786596  4.602884  ...  7.487840   \n",
       " GSM260665        4.649615  6.191800  5.057450  5.523562  ...  6.842979   \n",
       " GSM260666        3.472488  5.557655  2.432959  6.091700  ...  7.192786   \n",
       " GSM260667        4.217231  6.272397  4.852998  5.378512  ...  7.208478   \n",
       " GSM260668        3.472488  4.754888  4.643856  5.984134  ...  7.313790   \n",
       " GSM260669        2.722466  4.847997  4.247928  6.632268  ...  7.021258   \n",
       " GSM260670        1.584963  5.666757  5.247928  5.201634  ...  6.886306   \n",
       " GSM260671        5.495056  6.320124  3.523562  6.534497  ...  7.479780   \n",
       " GSM260672        5.951868  4.832890  5.247928  6.816344  ...  7.126188   \n",
       " GSM260673        1.678072  4.738768  4.321928  5.259272  ...  6.877744   \n",
       " GSM260674        5.070389  3.643856  4.542258  5.995485  ...  7.414474   \n",
       " GSM260675        3.981853  5.970394  4.321928  5.083213  ...  6.478972   \n",
       " GSM260676        3.916477  5.566815  4.008989  4.995485  ...  6.992089   \n",
       " GSM260677        3.263034  6.859224  4.121015  5.611763  ...  6.385431   \n",
       " GSM260678        4.523562  5.984134  3.419539  5.048759  ...  4.822730   \n",
       " GSM260679        2.700440  5.529821  5.193772  5.716991  ...  5.419539   \n",
       " GSM260680        1.536053  5.902074  2.201634  5.091700  ...  6.622052   \n",
       " GSM260681        4.177918  5.655352  4.247928  4.981853  ...  5.602884   \n",
       " GSM260682        4.596935  5.741467  5.504620  4.949535  ...  6.266787   \n",
       " GSM260683        1.925999  5.675251  2.321928  5.469235  ...  5.970394   \n",
       " GSM260684        3.247928  3.817623  2.906891  5.683696  ...  6.361066   \n",
       " GSM260685        4.232661  5.797013  5.321928  5.336283  ...  7.383704   \n",
       " GSM260686        5.083213  4.892391  4.201634  5.193772  ...  6.046578   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM260657        5.649615  5.140275  4.973513  3.981853  8.914305  10.101268   \n",
       " GSM260658        6.209453  6.278567  5.499623  4.321928  8.824914   8.759702   \n",
       " GSM260659        6.636625  5.177039  5.222257  2.321928  8.994110   8.482820   \n",
       " GSM260660        6.238405  5.797402  6.041351  3.786596  8.728841  10.428593   \n",
       " GSM260661        6.568336  3.731905  5.658431  1.432959  8.862316   9.248160   \n",
       " GSM260662        5.730640  4.903531  5.674302  2.070389  8.812090   9.939276   \n",
       " GSM260663        5.882643  6.241305  6.000908  2.867896  8.472535   8.823235   \n",
       " GSM260664        6.253611  4.759195  5.921665  2.847997  8.740388   9.101141   \n",
       " GSM260665        5.482203  5.654178  6.134398  4.446256  7.013305   9.098680   \n",
       " GSM260666        5.485427  6.132820  6.190781  3.722466  8.271633   8.693770   \n",
       " GSM260667        6.672425  4.980645  6.754124  3.722466  8.957080   8.776685   \n",
       " GSM260668        5.277985  5.717383  6.071777  3.017922  7.763477   8.891246   \n",
       " GSM260669        6.376777  5.301881  6.062566  2.786596  8.582004   9.049315   \n",
       " GSM260670        6.414474  4.347047  5.720911  1.887525  9.103722   8.503749   \n",
       " GSM260671        5.997744  5.687966  5.853122  2.485427  7.500638   9.192915   \n",
       " GSM260672        6.844235  5.150301  6.017372  4.185867  8.416135   8.591161   \n",
       " GSM260673        6.394034  4.433745  6.092480  1.321928  9.048169   8.508935   \n",
       " GSM260674        6.213347  4.377068  5.940844  3.827819  7.753485   8.777465   \n",
       " GSM260675        5.436295  3.333628  6.339832  1.378512  8.986200   8.711719   \n",
       " GSM260676        5.640968  5.054528  6.512911  3.153805  8.508907   8.359632   \n",
       " GSM260677        5.981853  5.188107  5.815095  3.459432  8.682056   9.262705   \n",
       " GSM260678        5.061776  4.067220  7.048478  2.584963  8.916857   9.201878   \n",
       " GSM260679        5.221104  4.014505  6.960100  2.459432  8.482056   8.695215   \n",
       " GSM260680        4.925999  4.714960  6.998555  1.000000  8.875129   8.907257   \n",
       " GSM260681        5.741467  4.755433  6.425593  2.584963  8.982493   8.538463   \n",
       " GSM260682        5.520422  4.969708  6.770647  2.321928  8.690957   8.374634   \n",
       " GSM260683        4.432959  3.706901  7.148605  1.807355  8.896124   8.751226   \n",
       " GSM260684        5.993221  4.333578  5.682569  1.632268  9.102571   9.072296   \n",
       " GSM260685        6.155830  4.859074  6.002105  2.035624  8.686657   8.493909   \n",
       " GSM260686        5.643856  6.052393  6.785250  3.277985  8.720976   7.871407   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM260657        4.953081  7.537221  5.733354  \n",
       " GSM260658        4.929363  7.792064  2.035624  \n",
       " GSM260659        4.536287  8.048368  1.632268  \n",
       " GSM260660        5.487515  8.001254  3.017922  \n",
       " GSM260661        4.908294  8.076230  2.000000  \n",
       " GSM260662        4.440496  7.909810  1.378512  \n",
       " GSM260663        5.562956  7.267139  1.722466  \n",
       " GSM260664        4.842485  7.111328  1.963474  \n",
       " GSM260665        5.764149  7.941183  2.744161  \n",
       " GSM260666        5.260595  7.835362  2.867896  \n",
       " GSM260667        5.678973  7.902770  2.263034  \n",
       " GSM260668        5.315244  6.422394  2.744161  \n",
       " GSM260669        6.023443  8.108486  2.321928  \n",
       " GSM260670        5.098881  8.109295  3.847997  \n",
       " GSM260671        5.043584  8.024829  3.584963  \n",
       " GSM260672        5.557075  8.183635  3.185867  \n",
       " GSM260673        4.439945  8.478648  1.263034  \n",
       " GSM260674        4.951582  7.783675  1.765535  \n",
       " GSM260675        4.967414  7.924408  2.000000  \n",
       " GSM260676        5.356606  7.537174  2.510962  \n",
       " GSM260677        5.410631  7.458225  2.786596  \n",
       " GSM260678        5.322278  7.921303  1.925999  \n",
       " GSM260679        4.507694  8.093594  0.925999  \n",
       " GSM260680        4.819359  7.977406  1.070389  \n",
       " GSM260681        4.815353  8.268763  1.378512  \n",
       " GSM260682        5.190116  8.244000  0.263034  \n",
       " GSM260683        5.702804  7.942674  1.678072  \n",
       " GSM260684        5.507709  7.989926  1.432959  \n",
       " GSM260685        4.156876  8.164977  1.584963  \n",
       " GSM260686        5.512607  8.276333  1.485427  \n",
       " \n",
       " [30 rows x 22898 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM2830518       8.177570  7.091112  4.753245  6.077951  5.086299  4.189478   \n",
       " GSM2830519       8.464326  7.286401  4.640745  6.230225  4.722610  4.420124   \n",
       " GSM2830520       8.567309  7.242341  4.734696  6.062693  4.678589  4.317246   \n",
       " GSM2830521       8.448587  7.336288  4.835442  6.071541  4.983665  4.096820   \n",
       " GSM2830522       8.439838  7.293728  4.764320  6.135620  4.807549  4.207679   \n",
       " GSM2830523       8.435544  7.367309  4.761672  5.848035  4.874789  4.034593   \n",
       " GSM2830524       8.357304  7.460432  4.561536  6.012957  4.764147  4.028060   \n",
       " GSM2830525       8.378831  7.300759  4.575292  5.888565  4.727878  4.074871   \n",
       " GSM2830526       8.367803  7.372180  4.763278  6.098263  4.723926  4.269331   \n",
       " GSM2830527       8.385181  7.141169  4.564357  6.116412  4.922879  4.212852   \n",
       " GSM2830528       8.292854  7.474466  4.698805  5.921939  4.516957  4.368913   \n",
       " GSM2830529       8.462315  7.529405  4.641525  6.077049  4.461614  4.256698   \n",
       " GSM2830530       8.501904  7.479128  4.782058  5.924923  4.676828  4.293817   \n",
       " GSM2830531       8.425714  7.406057  4.558828  5.790550  4.163671  4.151047   \n",
       " GSM2830532       8.384226  7.265516  4.696559  6.106419  4.818824  4.214750   \n",
       " GSM2830533       8.529440  7.323625  4.646197  6.176560  4.680415  4.269129   \n",
       " GSM2830534       8.459010  7.492488  4.636208  6.044867  4.760195  4.324524   \n",
       " GSM2830535       8.531530  7.297305  4.655899  6.133736  4.958443  4.046575   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...    ZWILCH  \\\n",
       " GSM2830518       5.077702  7.049808  6.338155  5.994071  ...  8.611195   \n",
       " GSM2830519       5.595008  7.102118  6.621345  5.986378  ...  8.896452   \n",
       " GSM2830520       5.105388  6.889804  6.474510  6.111591  ...  8.289515   \n",
       " GSM2830521       5.164496  6.876843  6.457002  6.173649  ...  8.078372   \n",
       " GSM2830522       5.436234  7.049939  6.723237  6.131691  ...  8.255503   \n",
       " GSM2830523       5.009375  7.031249  6.425596  6.094199  ...  9.677669   \n",
       " GSM2830524       5.205722  7.102193  6.394206  6.014461  ...  9.655030   \n",
       " GSM2830525       4.894600  7.067356  6.276730  6.073547  ...  9.696034   \n",
       " GSM2830526       5.071095  7.339566  6.577949  6.230133  ...  9.444833   \n",
       " GSM2830527       5.124328  7.081478  6.312428  6.173107  ...  9.432434   \n",
       " GSM2830528       4.840292  7.463007  6.521173  6.328655  ...  9.385161   \n",
       " GSM2830529       5.259883  7.056863  6.421576  6.023691  ...  9.647477   \n",
       " GSM2830530       5.443165  7.035286  6.589768  6.076231  ...  9.685497   \n",
       " GSM2830531       5.097055  6.920762  6.358335  5.998225  ...  9.550790   \n",
       " GSM2830532       5.093696  7.137411  6.561170  6.367783  ...  9.560253   \n",
       " GSM2830533       5.289937  7.186762  6.618784  6.250823  ...  9.385443   \n",
       " GSM2830534       4.983588  6.928559  6.612906  5.982860  ...  9.597350   \n",
       " GSM2830535       4.693237  7.160466  6.462997  5.896145  ...  9.202850   \n",
       " \n",
       " gene_symbol_sql      ZWINT      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B  \\\n",
       " GSM2830518        9.089413  7.465193  5.671706  6.852398  6.132009  9.386336   \n",
       " GSM2830519        9.596611  7.634785  5.798633  6.847215  6.451099  9.471822   \n",
       " GSM2830520        8.810135  7.691128  5.818277  6.819358  6.130951  9.408700   \n",
       " GSM2830521        8.951677  7.457200  5.545371  6.621451  6.112861  9.165547   \n",
       " GSM2830522        8.508791  7.496362  5.858392  6.787637  6.768579  9.403830   \n",
       " GSM2830523       10.515049  7.374466  5.675318  6.546174  6.533061  9.239360   \n",
       " GSM2830524       10.425674  7.367143  5.552888  6.565407  6.311067  9.424404   \n",
       " GSM2830525       10.506117  7.361694  5.558037  6.646901  6.297958  9.277119   \n",
       " GSM2830526       10.449058  7.398649  5.743879  6.635299  6.402373  9.321049   \n",
       " GSM2830527       10.404768  7.618828  5.861101  6.634953  6.551624  9.182944   \n",
       " GSM2830528       10.968133  6.739119  5.592553  6.317832  6.361648  9.087610   \n",
       " GSM2830529       10.913422  7.509853  5.721948  6.608597  6.579873  9.287299   \n",
       " GSM2830530       10.927092  7.227631  5.675054  6.522851  6.443318  9.159206   \n",
       " GSM2830531       10.774538  7.115291  5.510259  6.438678  6.229133  9.185749   \n",
       " GSM2830532       10.597708  7.517680  5.944866  6.716031  6.369601  9.330734   \n",
       " GSM2830533       10.655653  7.455786  5.829460  6.632918  6.489231  9.256060   \n",
       " GSM2830534       10.652160  7.110078  5.629134  6.429971  6.218302  9.345575   \n",
       " GSM2830535       10.357094  7.485896  5.637184  6.462988  6.264875  9.140645   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM2830518       8.140386  6.873020  9.087040  \n",
       " GSM2830519       8.260517  7.052330  9.241907  \n",
       " GSM2830520       7.930133  6.909751  9.009561  \n",
       " GSM2830521       7.866139  6.649220  9.092785  \n",
       " GSM2830522       8.132239  6.877018  9.139109  \n",
       " GSM2830523       8.118820  6.770484  8.989578  \n",
       " GSM2830524       8.098545  6.765778  8.951377  \n",
       " GSM2830525       8.103787  6.853741  8.951727  \n",
       " GSM2830526       8.116063  7.016828  9.044877  \n",
       " GSM2830527       8.160192  6.959850  9.019790  \n",
       " GSM2830528       7.906216  6.742924  8.758153  \n",
       " GSM2830529       8.197071  6.846311  9.154549  \n",
       " GSM2830530       8.118402  6.903445  9.008111  \n",
       " GSM2830531       7.941305  6.630567  8.960217  \n",
       " GSM2830532       8.128579  6.697720  9.151306  \n",
       " GSM2830533       8.174986  6.924472  9.094409  \n",
       " GSM2830534       8.049675  6.775464  8.904366  \n",
       " GSM2830535       8.080899  6.750199  8.855129  \n",
       " \n",
       " [18 rows x 21483 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM2836934       8.210199  6.454714  3.338840  7.173086  9.411777  3.777057   \n",
       " GSM2836935       9.014417  6.881364  5.452166  7.921764  7.899495  3.404676   \n",
       " GSM2836936       8.495095  6.776433  6.253685  8.239209  9.236093  6.169794   \n",
       " GSM2836938       8.145334  6.519830  5.413755  7.454654  7.859810  4.476158   \n",
       " GSM2836939       8.669766  5.624292  5.559355  6.471462  7.141491  5.098091   \n",
       " GSM2836940       8.325538  6.456978  5.178985  7.484246  8.222622  1.764705   \n",
       " GSM2836941       8.221685  6.370401  5.042248  7.284484  7.606524  3.632202   \n",
       " GSM2836948       8.161892  6.940030  3.854313  7.515630  7.450436  3.334898   \n",
       " GSM2836949       8.467593  6.041203  5.063404  7.176839  7.259505  3.628763   \n",
       " GSM2836950       7.805771  6.388079  3.866033  7.742524  8.189579  4.463808   \n",
       " GSM2836951       8.520099  6.988213  5.011780  6.069318  7.163989  3.652151   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM2836934       5.141108  3.679079  6.221183  1.783897  ...  11.163536   \n",
       " GSM2836935       5.409493  3.470071  3.600566  5.973234  ...   8.322992   \n",
       " GSM2836936       6.582071  3.826415  6.806630  5.690725  ...  11.177269   \n",
       " GSM2836938       4.750714  3.546843  5.479314  5.088162  ...  11.333121   \n",
       " GSM2836939       5.076453  4.087449  6.205099  5.490831  ...   8.674576   \n",
       " GSM2836940       4.735294  4.391327  3.344174  5.187882  ...  11.371426   \n",
       " GSM2836941       4.155398  3.618683  5.816922  5.391727  ...  10.681525   \n",
       " GSM2836948       4.202227  3.789436  6.311950  6.183212  ...  11.194724   \n",
       " GSM2836949       4.821832  4.444163  6.042751  4.757656  ...   9.218681   \n",
       " GSM2836950       5.332987  4.285365  2.428019  5.749844  ...  11.631270   \n",
       " GSM2836951       5.790502  3.078857  5.559629  5.565608  ...  10.555430   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B       ZYX  \\\n",
       " GSM2836934       8.211642  6.086051  8.049840  2.797846  10.811019  8.409478   \n",
       " GSM2836935       8.804394  6.973876  8.072845  2.913945  11.149321  8.999605   \n",
       " GSM2836936       7.947193  6.355152  8.186366  5.403404  10.685020  9.401726   \n",
       " GSM2836938       8.822856  6.798655  7.881991  2.371549  10.585248  8.472153   \n",
       " GSM2836939       8.857226  7.456447  8.156519  2.360499  10.949115  9.041222   \n",
       " GSM2836940       9.062387  6.883989  7.965198  3.640845   9.996349  9.168243   \n",
       " GSM2836941       8.895273  6.200228  7.903234  2.089423  10.318959  7.941039   \n",
       " GSM2836948       9.022554  6.482343  7.579369  4.627395  10.575726  8.491163   \n",
       " GSM2836949       8.974873  6.984480  8.021382  1.592603  10.942031  9.252326   \n",
       " GSM2836950       8.886972  6.670650  7.406678  3.410463   9.776008  9.343952   \n",
       " GSM2836951       8.534194  6.540098  7.835503  4.721346  10.083898  7.907124   \n",
       " \n",
       " gene_symbol_sql     ZZEF1       ZZZ3   mir-223  \n",
       " GSM2836934       5.712231  10.085787  0.800640  \n",
       " GSM2836935       7.871329   9.641209  2.035549  \n",
       " GSM2836936       7.200633   9.914944  2.083050  \n",
       " GSM2836938       5.914508  10.086303  1.553812  \n",
       " GSM2836939       6.179325   9.898124  1.510462  \n",
       " GSM2836940       5.652762  10.205646  1.846763  \n",
       " GSM2836941       5.844121  10.063972  1.630021  \n",
       " GSM2836948       5.704492  10.121514  1.814010  \n",
       " GSM2836949       5.728699   9.906214  1.753403  \n",
       " GSM2836950       5.631916  10.017171  1.516912  \n",
       " GSM2836951       5.384207   9.905525  2.060138  \n",
       " \n",
       " [11 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM551186        8.047450  6.671684  3.832538  4.330894  5.202427  4.006129   \n",
       " GSM551187        8.047322  6.138304  3.955388  4.508551  5.410347  3.924826   \n",
       " GSM551192        7.272463  5.511286  4.178391  4.300308  5.559290  4.152289   \n",
       " GSM551193        7.143422  5.561881  4.194275  4.431086  5.340700  4.287818   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM551186        3.929283  6.972073  4.457048  4.536106  ...   9.417217   \n",
       " GSM551187        4.148605  6.701010  4.630217  4.594723  ...   9.497592   \n",
       " GSM551192        4.172948  6.256079  4.500472  4.343254  ...  10.998691   \n",
       " GSM551193        4.187333  6.251027  4.647953  4.271382  ...  10.964725   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM551186        3.554460  4.808512  6.642124  5.928449  9.313159  8.740796   \n",
       " GSM551187        4.038438  4.746059  6.686788  6.025313  9.217313  8.833199   \n",
       " GSM551192        4.822597  4.726917  7.095917  6.772369  8.790063  7.416220   \n",
       " GSM551193        5.118114  4.948529  7.064135  6.716083  8.709897  7.470824   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM551186        6.148234  8.287331  3.480199  \n",
       " GSM551187        6.120372  8.334986  3.448786  \n",
       " GSM551192        5.741103  8.956456  3.348629  \n",
       " GSM551193        5.773388  8.897665  3.385994  \n",
       " \n",
       " [4 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM275687        6.825829  6.237424  3.753455  6.437529  3.789862  3.827654   \n",
       " GSM275688        6.825829  6.473445  3.834148  7.667915  3.856532  3.827654   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM275687        4.102894  7.031915  4.735015  4.625509  ...  10.030819   \n",
       " GSM275688        4.107933  6.920175  4.792686  4.625509  ...   9.936189   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM275687        5.508026  5.356369  5.094717  5.075199  8.746344  11.620775   \n",
       " GSM275688        5.508026  5.426420  5.098515  4.881126  8.479951  11.411115   \n",
       " \n",
       " gene_symbol_sql     ZZEF1     ZZZ3   mir-223  \n",
       " GSM275687        6.548394  7.97231  3.941771  \n",
       " GSM275688        6.548394  7.88711  3.941771  \n",
       " \n",
       " [2 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM2977388       6.123087  3.321928  3.879545  3.865422  3.548437  4.144779   \n",
       " GSM2977389       6.652486  5.412782  4.207343  4.367863  3.733354  4.621468   \n",
       " GSM2977390       6.572890  2.887525  2.467730  6.008949  3.485427  2.987379   \n",
       " GSM2977391       6.636625  5.990955  2.856710  2.660964  3.632268  3.737056   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM2977388       4.548437  2.378512  2.104337  4.694880  ...  10.627534   \n",
       " GSM2977389       2.405992  3.446256  5.765535  4.727920  ...  10.726133   \n",
       " GSM2977390       3.201634  3.292782  5.719731  5.412782  ...   8.996615   \n",
       " GSM2977391       4.000000  4.925999  3.498251  4.510962  ...  11.421434   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM2977388       6.515700  4.212229  5.134108  1.765535  8.994259  9.790188   \n",
       " GSM2977389       6.586465  3.949383  5.051617  2.104337  9.075792  9.227422   \n",
       " GSM2977390       6.345183  4.722064  5.535821  4.459432  9.340067  9.298062   \n",
       " GSM2977391       6.247928  4.626743  5.544922  5.000000  8.808620  9.285490   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM2977388       4.683516  7.871739  0.000000  \n",
       " GSM2977389       4.433814  8.157849  1.722466  \n",
       " GSM2977390       4.837288  7.951409  4.121015  \n",
       " GSM2977391       4.614654  8.107018  0.678072  \n",
       " \n",
       " [4 rows x 22899 columns],\n",
       " gene_symbol_sql       A1BG  A1BG-AS1      A1CF        A2M   A2M-AS1     A2ML1  \\\n",
       " GSM288223        10.150030  9.377831  8.627260  10.269436  7.932338  9.228567   \n",
       " GSM288224        10.648002  7.876055  7.552663   9.571469  8.992790  9.129850   \n",
       " GSM288225        11.359993  9.450395  7.186413  10.309189  9.017676  9.407024   \n",
       " GSM288226        10.512291  9.034328  8.356531   9.938592  8.122496  9.154165   \n",
       " GSM288227        11.011102  9.237611  7.823928   9.575718  9.277114  9.346398   \n",
       " GSM288228        10.387156  8.920353  7.293257   9.772172  9.275193  9.002628   \n",
       " GSM288229        10.527049  6.976116  8.872892  10.102778  9.499155  9.163608   \n",
       " GSM288230        10.443913  9.445203  8.306220  10.132274  8.952451  9.133483   \n",
       " \n",
       " gene_symbol_sql      A2MP1    A4GALT      A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM288223        10.185528  9.679936   9.989050  8.932136  ...  11.570693   \n",
       " GSM288224         9.737660  9.949245   9.168324  8.864702  ...  11.503797   \n",
       " GSM288225        10.127630  9.268104  10.057278  9.626658  ...  11.781190   \n",
       " GSM288226        10.434387  9.648134  10.400864  8.753327  ...  11.725496   \n",
       " GSM288227        10.423842  6.677455   9.106396  9.126973  ...  11.669244   \n",
       " GSM288228         9.917372  9.089583   7.584963  6.125155  ...  12.535081   \n",
       " GSM288229        10.553474  9.255457   9.942707  8.120669  ...  12.396688   \n",
       " GSM288230         9.449344  8.199180   9.484510  6.553874  ...  12.504471   \n",
       " \n",
       " gene_symbol_sql       ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B  \\\n",
       " GSM288223         9.721588  8.333214  9.012632  5.980647  12.317978   \n",
       " GSM288224        10.198982  7.392557  9.342997  5.661669  12.358746   \n",
       " GSM288225         9.930480  8.505575  8.708674  6.100407  12.300777   \n",
       " GSM288226         9.990867  8.439630  8.767596  5.714690  12.178337   \n",
       " GSM288227        10.382049  8.269545  8.284097  6.381889  12.137689   \n",
       " GSM288228         9.683521  7.960491  8.844331  7.243174  11.896760   \n",
       " GSM288229         9.784926  8.300240  8.792513  6.775691  12.158284   \n",
       " GSM288230        10.128883  8.175407  8.927758  7.013611  11.932054   \n",
       " \n",
       " gene_symbol_sql        ZYX     ZZEF1       ZZZ3   mir-223  \n",
       " GSM288223        11.491500  8.642101  12.290387  4.435297  \n",
       " GSM288224        11.768987  8.285372  12.190551  4.921755  \n",
       " GSM288225        12.305642  8.626004  12.171990  5.069365  \n",
       " GSM288226        12.550386  8.866017  12.099320  4.481818  \n",
       " GSM288227        12.251123  8.536996  12.101481  4.382431  \n",
       " GSM288228        13.522913  8.663736  11.995106  5.031219  \n",
       " GSM288229        13.495809  7.879528  12.080630  3.998933  \n",
       " GSM288230        13.589901  8.782242  12.056935  3.713815  \n",
       " \n",
       " [8 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M  A2M-AS1     A2ML1  \\\n",
       " GSM3263619       0.054891 -0.493339 -0.045866 -0.120401  0.74617 -0.169341   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT    A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM3263619      -0.156727  0.200416 -0.00586 -0.043986  ...  0.148186   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A   ZYG11B       ZYX  \\\n",
       " GSM3263619       0.035999  0.123881  0.319875  1.999415 -0.03105 -3.178675   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM3263619      -0.329631  0.267111 -0.263494  \n",
       " \n",
       " [1 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM299423        6.838338  2.236135  3.245361  4.842796  3.468652  2.508451   \n",
       " GSM299424        6.781289  5.799308  1.817999  2.863495  3.136961  3.295153   \n",
       " GSM299425        6.918374  5.089829  2.321025  3.906328  3.304902  2.133653   \n",
       " GSM299426        6.842904  4.580863  3.346949  4.768230  1.539561  3.427083   \n",
       " GSM299427        6.223255  5.865597  4.115855  3.054875  3.290950  2.782847   \n",
       " GSM299428        6.928983  5.405668  3.808918  4.451037  2.367193  3.189910   \n",
       " GSM299429        6.668454  5.802992  1.581057  4.576439  0.551579  0.431823   \n",
       " GSM299430        6.646739  5.705207  1.551417  4.850824  2.590235  3.727880   \n",
       " GSM299431        6.498859  5.662995  3.088029  2.856271  2.812901  2.834917   \n",
       " GSM299432        6.740516  3.428062  3.947073  4.569347  3.315497  1.879135   \n",
       " GSM299433        7.012886  4.948387  2.805769  4.460618  3.338892  3.266385   \n",
       " GSM299434        6.608600  6.209017  2.288278  3.492757  4.757292  3.465104   \n",
       " GSM299435        6.528414  4.797431  1.316285  4.864564  4.919002  4.163371   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM299423        1.697535  4.313055  4.437980  4.511965  ...   9.175178   \n",
       " GSM299424        1.448692  4.134005  3.057076  4.294105  ...  10.087887   \n",
       " GSM299425        4.469888  3.912067  5.011115  4.599895  ...   9.321091   \n",
       " GSM299426        3.455407  2.567240  3.603205  4.796970  ...   8.815397   \n",
       " GSM299427        3.313053  5.017849  4.624489  3.632490  ...   7.320909   \n",
       " GSM299428        2.552703  5.018254  3.279870  2.933879  ...   8.060152   \n",
       " GSM299429        1.280698  4.191136  4.558445  2.485156  ...   7.204564   \n",
       " GSM299430       -0.667133  5.188011  2.696017  0.678441  ...   7.001390   \n",
       " GSM299431        3.955966  4.911458  3.295049  4.513592  ...   7.868194   \n",
       " GSM299432        4.022599  5.028045  4.657654  4.831770  ...   6.958525   \n",
       " GSM299433        3.359797  5.025988  3.431471  4.514034  ...   7.451262   \n",
       " GSM299434        3.087027  5.572116  4.250638  4.987696  ...   7.356881   \n",
       " GSM299435        2.090406  5.552252  0.418281  3.984243  ...   7.764202   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM299423        5.535496  4.237848  5.439794  0.988753  8.568487  11.118227   \n",
       " GSM299424        6.033341  3.893911  4.511708  4.521288  8.654955  10.948124   \n",
       " GSM299425        5.914492  3.332292  5.287978  1.729114  8.574303  11.073805   \n",
       " GSM299426        5.765259  3.040139  5.171334  4.130537  8.651292  11.210865   \n",
       " GSM299427        5.719716  3.278764  5.018670  2.951265  8.634199  11.192622   \n",
       " GSM299428        5.810011  4.094640  5.144245  4.391881  8.833963  11.511361   \n",
       " GSM299429        5.365064  4.134322  4.819703  4.037710  8.658109  11.263932   \n",
       " GSM299430        6.059070  4.520929  4.689353  2.692520  8.442343  11.161470   \n",
       " GSM299431        5.413490  4.764378  4.663424  4.071760  8.756931  11.224601   \n",
       " GSM299432        6.055131  4.135958  5.133749  0.213725  8.048381  11.012595   \n",
       " GSM299433        6.000385  5.122815  4.499816  0.663792  8.663744  10.755839   \n",
       " GSM299434        5.710524  4.261524  5.449293  1.880364  9.264923  10.573776   \n",
       " GSM299435        6.120443  5.299280  5.347142  3.777402  9.298870  10.262043   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM299423        4.338228  7.941507  0.525293  \n",
       " GSM299424        3.673317  7.767114  0.560802  \n",
       " GSM299425        4.985075  8.037140  2.962017  \n",
       " GSM299426        4.240046  8.099794  2.293276  \n",
       " GSM299427        3.784086  7.901960  1.409187  \n",
       " GSM299428        4.159739  7.532248  1.148192  \n",
       " GSM299429        3.731056  7.802669  1.240262  \n",
       " GSM299430        4.376127  7.864444 -0.136960  \n",
       " GSM299431        4.699593  7.454248  1.894454  \n",
       " GSM299432        3.516279  7.985406 -0.010691  \n",
       " GSM299433        4.199977  7.997725  2.489880  \n",
       " GSM299434        4.300355  7.799962  1.313281  \n",
       " GSM299435        4.581632  7.782137  4.180072  \n",
       " \n",
       " [13 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM3354765       7.897845  5.906891  4.561129  6.054524  6.754888  5.027729   \n",
       " GSM3354766       7.507002  5.738768  2.796976  6.130004  6.066089  2.547962   \n",
       " GSM3354767       6.004501  5.694880  4.844999  7.813467  5.620586  4.775450   \n",
       " GSM3354768       6.388878  5.794416  3.227252  7.254396  7.142618  5.805476   \n",
       " GSM3354769       6.598425  5.904484  3.394278  8.273364  8.041112  5.078686   \n",
       " GSM3354770       7.086402  3.378512  3.962109  8.315606  7.811214  4.449527   \n",
       " GSM3354771       7.042207  4.906891  4.760064  6.580314  7.375908  3.611615   \n",
       " GSM3354772       6.456149  2.700440  4.021158  8.336168  7.189825  5.300649   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM3354765       1.847997  6.862947  4.285402  2.584963  ...  8.617651   \n",
       " GSM3354766       2.536053  7.516488  4.000000  3.153805  ...  8.264443   \n",
       " GSM3354767       2.137504  6.375039  4.459432  4.887525  ...  8.209453   \n",
       " GSM3354768       3.364572  6.125155  5.799605  5.201634  ...  8.246028   \n",
       " GSM3354769       4.517276  5.815063  6.723832  5.125155  ...  7.880196   \n",
       " GSM3354770       3.017922  5.405992  5.904484  6.346957  ...  7.867279   \n",
       " GSM3354771       5.296457  7.358431  6.483816  5.746850  ...  9.154818   \n",
       " GSM3354772       1.485427  7.113742  5.689299  6.161888  ...  9.024586   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B       ZYX  \\\n",
       " GSM3354765       8.430871  7.030500  7.361167  1.765535  10.774640  8.373457   \n",
       " GSM3354766       8.532551  6.503297  7.258434  1.925999  10.807626  8.857879   \n",
       " GSM3354767       8.422485  6.445763  7.231380  1.584963  10.701076  8.951293   \n",
       " GSM3354768       8.246978  6.314781  7.036254  1.678072  10.578361  9.283310   \n",
       " GSM3354769       8.775116  6.202637  7.346054  2.765535  10.511434  8.786059   \n",
       " GSM3354770       8.397033  5.392098  7.161420  1.584963  10.704626  8.697180   \n",
       " GSM3354771       8.415319  7.134441  7.254362  2.378512  11.000578  9.008211   \n",
       " GSM3354772       8.515306  6.463613  7.211002  2.321928  10.873980  8.833220   \n",
       " \n",
       " gene_symbol_sql     ZZEF1       ZZZ3   mir-223  \n",
       " GSM3354765       5.704869  10.181718  0.765535  \n",
       " GSM3354766       6.158123  10.194391  1.201634  \n",
       " GSM3354767       4.580787   9.826209  1.887525  \n",
       " GSM3354768       5.955043   9.783015  1.000000  \n",
       " GSM3354769       6.311933   9.709169  1.678072  \n",
       " GSM3354770       7.331656   9.817984  2.700440  \n",
       " GSM3354771       6.214085   9.969412  2.722466  \n",
       " GSM3354772       5.580642   9.965503  2.035624  \n",
       " \n",
       " [8 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM301253        7.927262  5.149259  2.722627  8.411072  5.433160  2.728649   \n",
       " GSM301254        7.676627  4.460546  3.526755  8.116142  5.785681  1.532958   \n",
       " GSM301256        7.924842  5.159347  2.585581  8.016383  5.770882  1.804411   \n",
       " GSM301258        8.034694  3.922674  3.392884  8.433110  5.585293  1.931885   \n",
       " GSM301262        7.894884  4.459038  2.517893  8.384721  5.786832  3.344851   \n",
       " GSM301263        7.846932  4.822322  3.363785  8.061768  6.028370  3.479986   \n",
       " GSM301264        8.075864  4.172087  2.792437  8.236491  5.445561  3.493689   \n",
       " GSM301265        7.830509  4.172247  3.847398  8.018814  5.305095  2.705533   \n",
       " GSM301266        7.797486  4.137913  3.883853  8.202426  5.677142  3.486938   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM301253        5.193062  7.393837  4.069788  5.559920  ...  9.229571   \n",
       " GSM301254        5.392730  7.488893  3.527571  4.926996  ...  9.234027   \n",
       " GSM301256        5.361102  7.715324  4.157691  5.570463  ...  9.242324   \n",
       " GSM301258        5.768396  7.045476  4.676775  5.707442  ...  9.468881   \n",
       " GSM301262        5.644462  7.607131  4.610109  5.168000  ...  9.410373   \n",
       " GSM301263        5.806221  6.958518  4.188480  5.744592  ...  9.374087   \n",
       " GSM301264        5.775393  7.072234  5.379760  6.039336  ...  8.132654   \n",
       " GSM301265        6.079122  7.167348  4.627548  5.984749  ...  8.226354   \n",
       " GSM301266        5.447811  7.493839  5.024187  5.329554  ...  7.935395   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM301253        6.032498  5.299391  5.504700  3.581472  8.193077  10.747193   \n",
       " GSM301254        6.209180  5.378448  5.311045  5.057060  8.402376  10.590067   \n",
       " GSM301256        5.861012  5.556555  5.553463  5.105133  8.193435  10.842077   \n",
       " GSM301258        5.898499  5.322565  5.234764  5.137340  8.136778  10.591351   \n",
       " GSM301262        6.031329  5.398106  5.735689  5.081127  8.359199  10.652690   \n",
       " GSM301263        5.954755  5.346041  5.560008  4.815012  8.337474  10.622401   \n",
       " GSM301264        6.036767  5.264510  5.604743  3.792751  8.281979  10.632140   \n",
       " GSM301265        5.975034  5.626466  5.643352  4.296457  8.252234  10.643011   \n",
       " GSM301266        6.251227  5.305169  5.629091  3.589524  8.273530  10.617063   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM301253        4.890904  7.876167  3.961160  \n",
       " GSM301254        4.878867  7.909357  3.383220  \n",
       " GSM301256        4.832105  7.778914  1.856787  \n",
       " GSM301258        5.455253  7.773312  0.503858  \n",
       " GSM301262        5.240882  7.949021  2.454176  \n",
       " GSM301263        5.224591  7.877767  2.425728  \n",
       " GSM301264        5.482065  7.815674  1.620117  \n",
       " GSM301265        5.237207  7.866900  1.590722  \n",
       " GSM301266        4.991446  7.785407  0.637842  \n",
       " \n",
       " [9 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM315617        4.369528  5.519704  2.364718  6.038004  5.785100  2.328224   \n",
       " GSM315618        2.849503  4.460716  2.444140  5.480659  3.214956  2.380913   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM315617        2.537011  2.606964  2.240889  2.252975  ...   9.233199   \n",
       " GSM315618        3.310740  2.512432  2.240545  2.252094  ...  10.941088   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM315617        7.118665  3.465333  5.325474  4.738000  8.362745  8.262741   \n",
       " GSM315618        6.495013  3.232012  5.080548  7.223156  8.121034  8.624954   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM315617        4.059796  7.895471  2.257196  \n",
       " GSM315618        4.113458  8.550473  2.256760  \n",
       " \n",
       " [2 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM336347        5.827311  3.803227  4.941289  6.088292  2.634593  2.539484   \n",
       " GSM336348        6.269220  3.915521  3.842063  5.285795  3.276497  3.081266   \n",
       " GSM336350        6.340384  4.152183  4.742944  5.683216  2.570463  3.707084   \n",
       " GSM336379        6.295907  3.382667  3.846243  5.443421  1.014355  2.604120   \n",
       " GSM336385        6.616916  3.798051  4.003669  6.801963  4.370164  2.108076   \n",
       " GSM336399        6.590961  3.859970  3.637510  6.514392  3.590961  3.238191   \n",
       " GSM336400        6.039138  2.715893  4.239298  7.200781  3.684819  2.712723   \n",
       " GSM336401        6.053763  2.702658  4.594840  7.389531  2.925999  2.321608   \n",
       " GSM336402        5.819157  3.666757  3.481523  7.218789  3.529821  2.924417   \n",
       " GSM336573        6.023255  3.428946  3.671082  7.308672  3.687061  2.771309   \n",
       " GSM336574        7.498809  3.561937  4.104374  5.702821  3.214125  3.333304   \n",
       " GSM336575        7.517118  3.639232  4.164222  6.212980  3.401903  2.194198   \n",
       " GSM336576        6.970969  2.698218  4.592585  6.430962  3.220330  2.863018   \n",
       " GSM336577        6.845490  1.922198  3.856284  6.414984  2.548437  3.524140   \n",
       " GSM336579        6.672142  2.608809  3.288503  5.492809  2.400538  2.884165   \n",
       " GSM336628        6.724923  3.524816  3.049391  5.407590  3.130931  3.101347   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM336347        6.045923  5.690696  4.374344  5.631977  ...  7.392918   \n",
       " GSM336348        5.788947  3.864929  4.018812  5.403268  ...  7.207307   \n",
       " GSM336350        5.601399  2.292782  4.638074  5.572283  ...  7.336819   \n",
       " GSM336379        5.536675  3.543496  4.334139  5.626147  ...  7.138221   \n",
       " GSM336385        5.160678  3.929791  4.549053  4.994127  ...  8.174926   \n",
       " GSM336399        5.273516  3.340562  4.496974  5.260026  ...  7.940343   \n",
       " GSM336400        5.516330  3.687061  4.676380  4.778734  ...  8.328720   \n",
       " GSM336401        5.368768  4.060047  4.318317  5.467279  ...  8.081031   \n",
       " GSM336402        4.864929  2.861955  4.489928  4.991408  ...  8.121119   \n",
       " GSM336573        5.068671  4.855990  4.450881  4.853497  ...  8.011675   \n",
       " GSM336574        5.349436  6.073392  4.582556  4.897724  ...  6.816856   \n",
       " GSM336575        5.424922  5.266787  4.537296  5.279471  ...  6.847746   \n",
       " GSM336576        5.456806  5.142005  4.516646  5.662205  ...  6.352617   \n",
       " GSM336577        5.609991  5.198887  4.158660  5.674404  ...  6.067165   \n",
       " GSM336579        5.391974  5.367721  4.341274  5.113117  ...  7.634448   \n",
       " GSM336628        5.359310  4.862451  4.375735  4.906891  ...  7.515384   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM336347        4.645586  4.320039  4.737642  2.313246  7.802003   9.186759   \n",
       " GSM336348        5.125155  4.575979  4.612808  0.000000  7.669059   9.348287   \n",
       " GSM336350        5.286142  4.369203  4.642586  1.356144  7.621304   9.153728   \n",
       " GSM336379        4.988230  4.229330  4.816837  1.275007  7.615887   9.244947   \n",
       " GSM336385        5.487036  4.647306  4.595475  1.650765  8.146086  10.622400   \n",
       " GSM336399        5.402586  4.757505  4.380982  3.354734  8.023133  10.654277   \n",
       " GSM336400        5.430954  4.559741  4.815317  2.946731  8.281060   9.474453   \n",
       " GSM336401        5.648465  4.908852  4.892739  3.395063  8.178327   9.331972   \n",
       " GSM336402        5.836429  4.728457  4.844747  3.647315  7.987929   9.526020   \n",
       " GSM336573        5.798569  4.587086  4.865758  3.267536  7.774765   9.202342   \n",
       " GSM336574        5.208673  4.685756  5.128817  3.040892  8.053447  10.486307   \n",
       " GSM336575        5.415488  4.724444  5.139201  2.625270  7.900320  10.325946   \n",
       " GSM336576        5.494736  4.346327  4.895418  2.198494  7.690514   9.913729   \n",
       " GSM336577        5.005849  4.645461  4.799713  2.301588  7.809870   9.670291   \n",
       " GSM336579        5.404971  4.361598  4.638766  2.650765  7.782677   9.109751   \n",
       " GSM336628        5.109778  4.596362  5.043939  1.981853  7.919867   9.172277   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM336347        4.933205  7.158077  3.182692  \n",
       " GSM336348        4.348022  6.971932  1.310340  \n",
       " GSM336350        4.060455  6.960688  1.996389  \n",
       " GSM336379        4.609876  7.081069  1.790772  \n",
       " GSM336385        4.545607  6.773441  0.367371  \n",
       " GSM336399        4.973143  6.712003  2.726831  \n",
       " GSM336400        4.272269  7.300000  1.361768  \n",
       " GSM336401        4.776554  7.369834  2.247928  \n",
       " GSM336402        4.840723  7.537127  2.301588  \n",
       " GSM336573        4.253134  7.402123  3.344828  \n",
       " GSM336574        4.648460  6.924923  1.232661  \n",
       " GSM336575        4.774664  7.151349  2.100978  \n",
       " GSM336576        4.904815  7.281753  1.992768  \n",
       " GSM336577        4.936173  7.087737  2.976364  \n",
       " GSM336579        4.456261  7.303269  0.150560  \n",
       " GSM336628        4.309261  7.275346  0.000000  \n",
       " \n",
       " [16 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM342998        8.405771  7.575312  4.288743  6.831752  4.634657  3.895365   \n",
       " GSM342998        8.405771  7.575312  4.288743  6.831752  4.634657  3.895365   \n",
       " GSM343003        8.272275  7.385043  4.647496  6.262278  5.366584  5.063033   \n",
       " GSM343003        8.272275  7.385043  4.647496  6.262278  5.366584  5.063033   \n",
       " GSM343006        8.554485  6.734696  4.722068  6.037815  4.949955  4.197231   \n",
       " GSM343006        8.554485  6.734696  4.722068  6.037815  4.949955  4.197231   \n",
       " GSM343007        8.429398  7.658347  3.149388  5.224308  5.390932  4.356054   \n",
       " GSM343007        8.429398  7.658347  3.149388  5.224308  5.390932  4.356054   \n",
       " GSM343008        8.943699  7.421997  3.942364  5.011040  5.369027  5.353596   \n",
       " GSM343008        8.943699  7.421997  3.942364  5.011040  5.369027  5.353596   \n",
       " GSM343009        8.253881  7.299703  3.762628  5.248795  1.857217  4.172950   \n",
       " GSM343009        8.253881  7.299703  3.762628  5.248795  1.857217  4.172950   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM342998        5.269796  5.874824  6.373422  7.158711  ...  12.839989   \n",
       " GSM342998        5.269796  5.874824  6.373422  7.158711  ...  12.839989   \n",
       " GSM343003        5.254764  6.728818  5.616928  7.317060  ...  12.889875   \n",
       " GSM343003        5.254764  6.728818  5.616928  7.317060  ...  12.889875   \n",
       " GSM343006        5.244073  7.520627  6.025605  6.560241  ...  12.707240   \n",
       " GSM343006        5.244073  7.520627  6.025605  6.560241  ...  12.707240   \n",
       " GSM343007        5.106633  6.334838  4.851009  6.248129  ...  12.678424   \n",
       " GSM343007        5.106633  6.334838  4.851009  6.248129  ...  12.678424   \n",
       " GSM343008        5.446494  7.529212  6.393816  6.615041  ...  12.407228   \n",
       " GSM343008        5.446494  7.529212  6.393816  6.615041  ...  12.407228   \n",
       " GSM343009        5.158680  6.614097  4.531812  6.913021  ...  12.521897   \n",
       " GSM343009        5.158680  6.614097  4.531812  6.913021  ...  12.521897   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B        ZYX  \\\n",
       " GSM342998        7.468111  6.176946  7.400465  7.256482  10.620043  12.040924   \n",
       " GSM342998        7.468111  6.176946  7.400465  7.256482  10.620043  12.040924   \n",
       " GSM343003        7.700273  5.508898  7.119615  7.166976  10.573997  11.514674   \n",
       " GSM343003        7.700273  5.508898  7.119615  7.166976  10.573997  11.514674   \n",
       " GSM343006        8.222674  5.656299  7.373061  7.186193  10.484103  11.142892   \n",
       " GSM343006        8.222674  5.656299  7.373061  7.186193  10.484103  11.142892   \n",
       " GSM343007        7.716223  6.873812  7.476964  7.149219  10.835412  12.068040   \n",
       " GSM343007        7.716223  6.873812  7.476964  7.149219  10.835412  12.068040   \n",
       " GSM343008        7.574866  6.737111  7.678534  6.856948  10.434771  11.928980   \n",
       " GSM343008        7.574866  6.737111  7.678534  6.856948  10.434771  11.928980   \n",
       " GSM343009        7.142475  6.576325  7.250333  6.768462  10.457394  13.197292   \n",
       " GSM343009        7.142475  6.576325  7.250333  6.768462  10.457394  13.197292   \n",
       " \n",
       " gene_symbol_sql     ZZEF1       ZZZ3   mir-223  \n",
       " GSM342998        5.762635  10.154676  3.054552  \n",
       " GSM342998        5.762635  10.154676  3.054552  \n",
       " GSM343003        6.090724   9.985281  2.779445  \n",
       " GSM343003        6.090724   9.985281  2.779445  \n",
       " GSM343006        6.034253  10.229445  4.761243  \n",
       " GSM343006        6.034253  10.229445  4.761243  \n",
       " GSM343007        6.145404  10.085518  2.470703  \n",
       " GSM343007        6.145404  10.085518  2.470703  \n",
       " GSM343008        5.819965   9.946488  3.464995  \n",
       " GSM343008        5.819965   9.946488  3.464995  \n",
       " GSM343009        5.941198   9.842118  2.900616  \n",
       " GSM343009        5.941198   9.842118  2.900616  \n",
       " \n",
       " [12 rows x 22899 columns],\n",
       " gene_symbol_sql       A2M      AAAS      AACS     AAED1     AAGAB      AAK1  \\\n",
       " GSM346941        6.760022  4.818097  6.275851  7.755485  6.667386  6.476498   \n",
       " GSM346942        6.850983  4.685280  6.240398  7.556991  6.445718  6.094876   \n",
       " GSM346943        8.396949  6.110971  7.940539  7.528553  7.639767  7.962319   \n",
       " GSM346944        8.231347  6.267566  7.451433  7.487354  7.588401  7.625999   \n",
       " GSM346945        6.296331  6.335292  6.908444  7.625532  6.794523  7.279582   \n",
       " GSM346946        6.818733  5.992053  6.995392  8.008161  7.326375  7.592318   \n",
       " GSM346947        6.280072  6.520278  7.647831  7.254876  7.432732  8.004492   \n",
       " GSM346948        7.054261  6.666521  7.682579  7.521319  7.744956  8.231428   \n",
       " \n",
       " gene_symbol_sql     AAMDC      AAMP      AAR2      AARS  ...      ZW10  \\\n",
       " GSM346941        6.735897  7.162015  6.476646  8.482697  ...  5.779432   \n",
       " GSM346942        6.547340  7.003699  6.505784  8.148094  ...  5.669353   \n",
       " GSM346943        7.947877  7.790267  7.441547  9.857189  ...  6.895447   \n",
       " GSM346944        7.825072  7.736363  7.341665  9.735587  ...  6.894340   \n",
       " GSM346945        7.226915  6.931403  6.459348  8.973789  ...  6.301033   \n",
       " GSM346946        7.446742  6.976853  6.449835  9.182444  ...  6.736389   \n",
       " GSM346947        7.411531  7.732699  7.059021  9.279116  ...  7.111164   \n",
       " GSM346948        7.616850  7.690283  7.344907  9.665260  ...  7.397021   \n",
       " \n",
       " gene_symbol_sql    ZWILCH      ZWINT      ZXDA      ZXDB      ZXDC    ZYG11B  \\\n",
       " GSM346941        6.794651   8.300852  3.984314  3.570298  4.761681  5.781015   \n",
       " GSM346942        6.565496   8.144727  3.998151  3.512696  5.049999  5.544275   \n",
       " GSM346943        8.051911   9.394342  4.939773  4.572394  5.073305  7.673463   \n",
       " GSM346944        7.868135   9.589244  4.962057  4.750676  5.379697  7.556527   \n",
       " GSM346945        8.125105   9.671003  4.122708  3.686653  4.212021  5.834940   \n",
       " GSM346946        8.277661   9.789566  4.333784  3.382003  4.329087  6.417214   \n",
       " GSM346947        8.548141  10.154597  4.487679  4.058040  5.060826  7.122487   \n",
       " GSM346948        8.805512  10.407030  4.761842  4.650557  5.592871  7.594185   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM346941        8.828239  5.591778  6.098757  \n",
       " GSM346942        8.284988  5.046423  6.048583  \n",
       " GSM346943        9.435886  6.817643  6.703635  \n",
       " GSM346944        9.104123  6.725927  6.629899  \n",
       " GSM346945        9.090943  6.127622  6.274143  \n",
       " GSM346946        9.130527  6.713163  6.177125  \n",
       " GSM346947        9.193206  6.718180  6.465732  \n",
       " GSM346948        9.414532  6.966451  6.919493  \n",
       " \n",
       " [8 rows x 11340 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM347919        7.494583  6.458395  4.132927  5.338078  4.749765  4.508927   \n",
       " GSM347920        6.875303  5.918945  4.158005  5.159791  4.562590  4.641670   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM347919        5.302146  6.347552  5.204717  5.411072  ...  10.489427   \n",
       " GSM347920        5.083985  6.383922  5.258141  5.195844  ...   9.744260   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B        ZYX  \\\n",
       " GSM347919        7.495112  5.460695  6.425555  5.657998  9.043738   9.903453   \n",
       " GSM347920        7.745109  5.914776  6.720629  5.867947  9.503999  10.144837   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM347919        6.653885  9.388831  4.656051  \n",
       " GSM347920        7.068020  9.728573  4.403826  \n",
       " \n",
       " [2 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM347191        6.500655  4.473644  5.605824  8.248309  6.810726  6.150730   \n",
       " GSM347193        7.525356  3.458264  2.998145  7.572834  7.567910  4.243383   \n",
       " GSM347194        7.222418  4.248391  2.864612  8.135218  7.310076  2.722832   \n",
       " GSM347196        7.014400  4.140501  4.255500  7.504328  6.966061  5.365227   \n",
       " GSM347197        7.132052  5.156219  4.999280  7.130272  7.098053  5.446213   \n",
       " GSM347328        7.278384  4.402934  5.815472  7.501673  7.371297  6.317355   \n",
       " GSM347330        5.652458  5.690708  4.726262  7.695438  7.102396  3.732156   \n",
       " GSM347332        5.758242  3.715893  4.117916  6.253601  6.997044  3.190830   \n",
       " GSM347335        5.917761  4.484634  5.702981  8.083499  7.061225  5.871974   \n",
       " GSM347337        6.641459  4.315704  4.553256  7.559430  6.783784  5.017216   \n",
       " GSM347340        4.553698  2.879872  3.067813  6.558416  7.085839  4.843099   \n",
       " GSM347342        4.817460  3.857682  2.803906  5.621671  7.370077  4.169405   \n",
       " GSM347733        4.954638  2.285020  4.711495  6.700249  6.600265  4.219953   \n",
       " GSM347734        5.780399  2.749152  4.020961  6.234456  6.491029  2.592429   \n",
       " GSM347735        5.309020  3.152032  4.694767  6.948734  6.181923  5.226094   \n",
       " GSM347736        4.858642  4.152402  4.577876  6.905101  4.803878  4.122252   \n",
       " GSM347737        5.491160  3.359535  2.595400  6.597326  6.643670  2.786023   \n",
       " GSM347738        2.522909  3.138628  4.005609  6.167155  6.023080  4.147737   \n",
       " GSM347739        5.181695  3.512214  4.164814  6.266165  6.811201  3.994419   \n",
       " GSM347740        4.692890  2.702662  3.347213  6.376941  6.127196  2.452383   \n",
       " GSM347741        6.075764  2.731967  2.496300  7.964747  6.555015  2.144586   \n",
       " GSM347742        5.717970  4.056679  3.594455  8.277710  6.519218  4.111924   \n",
       " GSM347743        6.327280  3.304972  4.887219  7.830293  6.403534  3.682902   \n",
       " GSM347744        5.659359  3.385307  2.151031  8.163990  5.812375  2.261760   \n",
       " GSM347745        6.097284  3.269334  5.186207  8.620063  6.326345  5.417847   \n",
       " GSM347746        5.407088  3.728323  4.259418  8.124911  6.717306  3.126039   \n",
       " GSM347747        6.117676  3.013512  3.640289  8.794282  6.872989  3.971018   \n",
       " GSM347748        5.699116  2.643039  3.853084  8.552650  5.937448  3.758621   \n",
       " GSM347749        5.834180  3.250771  4.289219  6.783179  5.528212  5.337312   \n",
       " GSM347764        5.651471  3.592385  2.500642  8.234891  7.390788  5.179505   \n",
       " GSM347765        7.164977  3.964565  5.360092  8.454551  6.748676  4.758295   \n",
       " GSM347766        6.462479  3.301948  3.388393  8.577122  6.777262  4.518706   \n",
       " GSM347767        6.447579  4.056193  6.133132  8.767055  6.201761  5.038717   \n",
       " GSM347798        5.837083  3.849619  4.548368  8.624406  5.820853  5.104178   \n",
       " GSM347799        6.752655  3.678444  4.879551  8.399994  6.408603  4.210074   \n",
       " GSM347819        6.678001  3.775978  3.922355  7.925385  6.100278  1.540454   \n",
       " GSM347837        5.998236  3.691032  5.481228  8.281166  6.548415  2.810326   \n",
       " GSM347838        5.472803  3.750607  3.934988  8.026978  5.195749  4.580862   \n",
       " GSM347839        5.289720  3.120435  3.904541  8.222453  5.634021  3.796661   \n",
       " GSM347840        6.283788  3.735501  4.132909  8.643438  6.102200  3.334170   \n",
       " GSM347841        6.564299  5.897887  3.928214  8.491482  6.549377  4.563858   \n",
       " GSM347842        7.397144  4.864835  4.794221  8.333824  6.810854  4.860923   \n",
       " GSM347843        7.327678  3.379302  4.354279  7.868082  5.632012  3.539325   \n",
       " GSM347844        6.699232  3.290009  3.367799  8.421239  6.031704  5.334092   \n",
       " GSM347845        6.868069  6.127258  5.399872  8.476720  6.906830  3.335899   \n",
       " GSM348017        7.320792  4.031051  5.434564  8.395850  6.537941  4.656494   \n",
       " GSM348018        7.803453  5.319311  3.495737  8.744084  6.295897  2.950425   \n",
       " GSM348020        7.328154  6.007518  5.186831  8.388270  6.115897  4.838434   \n",
       " GSM348022        7.224812  4.428095  2.180838  7.713199  3.894071  4.604028   \n",
       " GSM348023        6.631036  3.346219  4.058580  8.267994  6.905267  5.721742   \n",
       " GSM348024        6.639685  5.344750  2.130696  8.410730  7.993584  3.666730   \n",
       " GSM348027        6.772875  4.117304  5.759590  8.692356  7.014266  3.085248   \n",
       " GSM348028        6.986968  3.682652  3.297890  8.537467  6.628447  2.502443   \n",
       " GSM348029        6.870661  5.892801  3.832828  8.047592  6.998715  4.458189   \n",
       " GSM348030        7.518394  4.504144  4.429549  8.291218  7.660360  3.094605   \n",
       " GSM348031        6.857881  2.867254  2.598815  8.584676  6.727485  3.507107   \n",
       " GSM348034        7.545806  4.204970  4.250959  8.362545  6.978745  2.678615   \n",
       " GSM348035        7.053144  5.715048  3.711308  8.571229  6.115277  2.954333   \n",
       " GSM348036        6.311487  3.516116  3.764455  8.499585  5.604798  5.048469   \n",
       " GSM348037        7.134416  3.640401  4.067343  8.823641  7.999008  2.671443   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM347191        2.191398  7.994687  5.858533  6.596527  ...  10.013811   \n",
       " GSM347193        5.889722  8.519487  6.052790  5.904961  ...   9.732795   \n",
       " GSM347194        6.317206  8.810575  5.806726  6.850549  ...   9.713452   \n",
       " GSM347196        0.721189  8.145688  3.451277  6.565826  ...  10.122776   \n",
       " GSM347197        3.211367  9.207031  6.495511  7.242441  ...   9.677561   \n",
       " GSM347328        4.062217  8.875408  3.026789  6.985182  ...   9.638876   \n",
       " GSM347330        6.329238  9.033778  3.388121  6.704734  ...   9.532284   \n",
       " GSM347332        5.838395  9.266585  5.846052  6.434128  ...   9.779225   \n",
       " GSM347335        3.344899  6.647717  3.890408  6.619445  ...   9.254150   \n",
       " GSM347337        3.967169  9.954182  5.809039  5.450865  ...   9.427845   \n",
       " GSM347340        5.033057  7.133923  5.955592  5.999098  ...   9.365211   \n",
       " GSM347342        1.422938  6.979271  5.707158  4.453037  ...   9.415323   \n",
       " GSM347733        1.523763  7.188816  5.855227  5.883442  ...   9.376796   \n",
       " GSM347734        2.754494  7.509189  5.906340  5.538315  ...   9.433811   \n",
       " GSM347735        5.122470  7.630289  5.182525  5.015569  ...   9.057762   \n",
       " GSM347736        3.386866  7.445801  1.923697  5.907770  ...   8.697916   \n",
       " GSM347737        1.678590  7.114606  6.208786  6.081734  ...   9.156121   \n",
       " GSM347738        4.806566  8.151301  5.305770  5.407781  ...   8.957804   \n",
       " GSM347739        4.340755  8.788102  3.653725  5.488933  ...   9.104137   \n",
       " GSM347740        2.568370  8.816309  5.106420  5.450159  ...   8.064037   \n",
       " GSM347741        4.345418  7.013786  6.037426  5.403741  ...  10.055079   \n",
       " GSM347742        5.706064  7.587643  5.380487  5.381183  ...   9.716641   \n",
       " GSM347743        4.704214  7.892749  6.153897  5.898661  ...   9.605468   \n",
       " GSM347744        4.475455  6.930867  4.431509  5.754497  ...   8.967333   \n",
       " GSM347745        3.512442  7.277092  6.376761  4.114242  ...  10.019535   \n",
       " GSM347746        5.284721  7.297989  6.327867  4.968897  ...   9.734874   \n",
       " GSM347747        4.633658  7.958495  4.504093  6.483387  ...   9.615902   \n",
       " GSM347748        3.305825  8.672167  6.106015  6.436767  ...   9.549534   \n",
       " GSM347749        2.396151  9.063328  5.540393  3.891186  ...   9.641136   \n",
       " GSM347764        1.826774  7.794279  5.850367  6.365166  ...   9.664054   \n",
       " GSM347765        4.102818  7.505415  6.707180  5.943518  ...   9.570292   \n",
       " GSM347766        4.839315  8.314388  4.678556  6.146885  ...   9.196115   \n",
       " GSM347767        2.419498  8.396511  5.874598  7.128830  ...   9.072350   \n",
       " GSM347798        1.694871  8.997267  6.387213  6.250056  ...   8.757947   \n",
       " GSM347799        2.758593  7.641358  5.328546  5.427063  ...   9.478042   \n",
       " GSM347819        5.568747  7.639312  5.286323  5.187411  ...   9.455165   \n",
       " GSM347837        1.774793  8.396001  6.040290  6.013755  ...   9.233531   \n",
       " GSM347838        3.005116  8.181316  5.943992  5.148422  ...   8.984011   \n",
       " GSM347839        2.362007  7.230145  6.104731  5.338710  ...   8.608831   \n",
       " GSM347840        4.812293  7.498554  6.112225  6.121830  ...   9.766817   \n",
       " GSM347841        1.564525  7.902598  5.931586  3.605992  ...  10.346724   \n",
       " GSM347842        4.544133  7.403200  5.674215  5.984042  ...   9.648447   \n",
       " GSM347843        4.566298  8.422590  4.904214  5.533046  ...  10.076709   \n",
       " GSM347844        2.584847  6.976112  6.064715  5.843750  ...   9.306627   \n",
       " GSM347845        1.435405  6.886306  5.722332  6.229771  ...  10.104533   \n",
       " GSM348017        2.000094  7.496047  5.092254  5.711696  ...   9.665766   \n",
       " GSM348018        2.937009  8.228213  5.696706  6.116544  ...   9.772900   \n",
       " GSM348020        5.863701  8.873404  6.410148  5.182351  ...   9.739114   \n",
       " GSM348022        5.191914  9.304700  4.283759  6.576389  ...   9.782369   \n",
       " GSM348023        3.973575  7.743098  6.400663  6.135752  ...   9.673463   \n",
       " GSM348024        5.054636  6.960604  5.484860  5.733926  ...   9.704901   \n",
       " GSM348027        4.846548  7.620946  3.991381  5.751397  ...   9.763913   \n",
       " GSM348028        5.523625  8.402232  5.632853  4.339330  ...   9.502398   \n",
       " GSM348029        2.179387  6.785211  5.076589  5.739613  ...   9.227419   \n",
       " GSM348030        5.513336  7.646127  6.138081  1.548185  ...   9.593496   \n",
       " GSM348031        4.170173  7.595817  5.727485  6.314947  ...   9.407720   \n",
       " GSM348034        3.227629  8.012781  5.053541  4.225414  ...   9.243740   \n",
       " GSM348035        4.780137  8.696689  5.207908  4.976400  ...   9.042409   \n",
       " GSM348036        4.562847  9.136535  4.244453  5.313532  ...   9.358656   \n",
       " GSM348037        4.524647  6.960917  5.465406  5.507715  ...   9.367961   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B        ZYX  \\\n",
       " GSM347191        7.313627  6.056594  7.318898  3.566876  10.224239  11.682643   \n",
       " GSM347193        8.020786  6.196418  6.755989  5.250056   9.904842  11.841639   \n",
       " GSM347194        7.404324  4.838725  7.291183  4.563469   9.843693  11.748416   \n",
       " GSM347196        8.300847  5.762926  6.589418  4.365895   9.659084  11.240935   \n",
       " GSM347197        7.349295  4.478784  7.144983  4.739189   9.977051  11.557302   \n",
       " GSM347328        8.273059  5.289096  6.733497  4.018785   9.588130  11.291565   \n",
       " GSM347330        7.882472  5.864445  7.441046  5.268835   9.761554  11.568724   \n",
       " GSM347332        8.065238  6.108667  7.200260  4.573635   9.746523  11.078146   \n",
       " GSM347335        7.765548  6.523822  7.487013  2.849141  10.066656  11.776179   \n",
       " GSM347337        8.136653  5.481091  6.879730  3.714059   9.448439  10.878095   \n",
       " GSM347340        8.123423  5.516953  7.197920  1.908033  10.336252  10.590375   \n",
       " GSM347342        7.806195  5.380689  7.152011  1.662506  10.284248  10.647208   \n",
       " GSM347733        7.603070  4.581505  6.638526  3.193106  10.518821  10.596148   \n",
       " GSM347734        7.519054  5.604228  7.254901  2.403489  10.269222  10.479414   \n",
       " GSM347735        8.154980  4.852288  7.288511  2.275108   9.912224  10.418051   \n",
       " GSM347736        7.604687  5.254158  7.475981  2.610015  10.075331  10.858768   \n",
       " GSM347737        7.975670  6.024360  7.198603  2.736901  10.414326  10.325157   \n",
       " GSM347738        8.593488  6.182250  7.003940  2.615251  10.367330  10.350366   \n",
       " GSM347739        8.419682  6.360714  7.169005  2.653567  10.100082  10.165150   \n",
       " GSM347740        7.491677  5.877031  7.794061  2.969488  10.323545   9.837501   \n",
       " GSM347741        7.842721  6.428873  7.150348  2.953705  11.056150  11.262453   \n",
       " GSM347742        8.083001  6.531687  7.435795  3.280879  10.841952  10.857730   \n",
       " GSM347743        8.328316  5.689551  7.673160  2.931233  10.681513  11.373402   \n",
       " GSM347744        8.143526  5.841528  7.713657  4.817245  10.882156  10.830733   \n",
       " GSM347745        8.193210  4.855251  7.044524  3.265414  11.211985  11.184999   \n",
       " GSM347746        8.202398  6.213453  7.576791  3.415691  10.992394  10.804126   \n",
       " GSM347747        8.309740  6.360804  6.966204  5.473326  10.827717  10.806361   \n",
       " GSM347748        8.472041  5.900206  7.163614  3.748708  10.752426  10.983317   \n",
       " GSM347749        8.130710  6.075683  7.460811  3.737189  10.798074  10.179241   \n",
       " GSM347764        7.918839  5.624433  7.232250  2.402321  11.002413  10.988565   \n",
       " GSM347765        7.954220  5.773669  7.360348  3.223110  10.719909  10.915128   \n",
       " GSM347766        7.956446  5.896190  7.351831  3.161173  10.623529  10.940063   \n",
       " GSM347767        8.368843  6.033508  7.475699  2.006057  10.363309  10.812019   \n",
       " GSM347798        8.220393  6.961238  7.801838  5.164963  10.363531   9.593171   \n",
       " GSM347799        8.112721  6.510377  7.172412  4.282595  10.744860  10.974621   \n",
       " GSM347819        8.111965  5.953792  7.351683  3.785603  10.849597  10.837634   \n",
       " GSM347837        8.512527  5.467457  6.457601  5.356750  10.563770  11.083833   \n",
       " GSM347838        8.618125  6.652617  6.870879  3.728454  10.394772  10.844969   \n",
       " GSM347839        7.679839  5.722784  7.895237  1.511873  10.826759  10.398132   \n",
       " GSM347840        8.063244  5.721659  7.339184  4.442214  10.826884  11.019444   \n",
       " GSM347841        7.837792  5.534040  6.742765  2.497367  11.118288  11.243503   \n",
       " GSM347842        8.259301  5.631975  7.402271  1.865293  10.804984  10.790550   \n",
       " GSM347843        8.311367  6.011021  7.183687  2.038384  10.712582  10.831976   \n",
       " GSM347844        8.443296  5.982493  7.606144  4.151218  10.783055  11.031133   \n",
       " GSM347845        8.092747  6.515753  7.684342  4.601863  10.896689  11.132455   \n",
       " GSM348017        8.350612  6.394008  7.065686  2.284085  10.938967  11.005670   \n",
       " GSM348018        8.470895  5.899825  6.658106  2.680500  10.811067  10.880979   \n",
       " GSM348020        8.599411  6.647292  6.918561  2.134398  10.346239  10.342074   \n",
       " GSM348022        8.368847  7.042434  7.460815  4.336291  10.643622   9.621337   \n",
       " GSM348023        8.400606  5.937392  7.161213  4.734108  10.924581  10.953602   \n",
       " GSM348024        8.120787  6.457075  7.358339  5.717846  11.172370  10.531720   \n",
       " GSM348027        8.102910  5.349157  7.658064  4.487666  11.187961  10.352044   \n",
       " GSM348028        7.960124  6.410307  7.204559  2.754079  10.793759  10.692136   \n",
       " GSM348029        8.006292  6.041057  7.648798  4.415448  10.940609  10.602746   \n",
       " GSM348030        8.146869  6.216989  7.127688  2.649737  10.985265  10.728119   \n",
       " GSM348031        8.306868  5.968085  7.331092  2.032443  11.110991  10.397309   \n",
       " GSM348034        8.679786  5.578657  6.932637  5.333288  10.976190  10.956171   \n",
       " GSM348035        8.237004  6.470723  6.710845  1.874463  10.683488  10.137722   \n",
       " GSM348036        8.148131  7.126990  7.412194  2.419267  10.625367   9.220836   \n",
       " GSM348037        7.975602  6.426368  7.545429  4.077978  11.093711  10.419500   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM347191        6.316109  9.527090  1.490524  \n",
       " GSM347193        6.429311  9.357135  2.301915  \n",
       " GSM347194        6.365826  9.676856  2.944122  \n",
       " GSM347196        5.917993  9.117841  2.826225  \n",
       " GSM347197        5.926055  9.230627  5.862794  \n",
       " GSM347328        7.476464  9.007759  3.516116  \n",
       " GSM347330        7.233029  9.555641  3.724377  \n",
       " GSM347332        5.860783  8.652148  1.425846  \n",
       " GSM347335        6.566394  9.507299  5.804776  \n",
       " GSM347337        7.375041  9.245962  2.597693  \n",
       " GSM347340        6.921448  9.513961  1.584453  \n",
       " GSM347342        7.094264  9.508500  1.463622  \n",
       " GSM347733        6.433716  9.745790  3.377318  \n",
       " GSM347734        6.769837  9.521952  1.367980  \n",
       " GSM347735        5.660547  9.431806  1.771281  \n",
       " GSM347736        6.812550  9.729620  0.891722  \n",
       " GSM347737        6.235042  9.647030  1.367086  \n",
       " GSM347738        7.233161  9.322365  1.878396  \n",
       " GSM347739        6.621863  9.453865  1.470542  \n",
       " GSM347740        6.023315  9.455179  1.583134  \n",
       " GSM347741        6.595002  9.790693  2.251789  \n",
       " GSM347742        6.342964  9.806611  1.630792  \n",
       " GSM347743        6.913821  9.548988  1.533598  \n",
       " GSM347744        6.215889  9.629554  1.671728  \n",
       " GSM347745        6.484150  9.636560  1.917585  \n",
       " GSM347746        5.673489  9.700288  2.326391  \n",
       " GSM347747        6.715982  9.432600  2.066819  \n",
       " GSM347748        6.437042  8.998128  1.783721  \n",
       " GSM347749        6.231303  9.415732  0.800952  \n",
       " GSM347764        6.704571  9.690063  3.334826  \n",
       " GSM347765        6.262103  9.652075  2.625058  \n",
       " GSM347766        6.915294  9.542574  1.967526  \n",
       " GSM347767        5.904669  9.662463  1.414817  \n",
       " GSM347798        6.562719  9.503751  1.298148  \n",
       " GSM347799        6.003651  9.835874  1.498103  \n",
       " GSM347819        6.304371  9.715022  2.061196  \n",
       " GSM347837        6.692202  9.303339  1.801664  \n",
       " GSM347838        7.303091  8.592964  2.254597  \n",
       " GSM347839        5.725917  9.710930  2.510002  \n",
       " GSM347840        5.796152  9.626264  3.277082  \n",
       " GSM347841        5.656969  9.758889  2.988837  \n",
       " GSM347842        6.958240  9.819255  2.627198  \n",
       " GSM347843        5.402067  9.607218  1.102846  \n",
       " GSM347844        6.327794  9.754319  1.755336  \n",
       " GSM347845        6.648919  9.889031  1.556150  \n",
       " GSM348017        6.570367  9.836845  1.372345  \n",
       " GSM348018        6.669906  9.467713  2.575709  \n",
       " GSM348020        7.149845  9.049889  1.261603  \n",
       " GSM348022        6.326672  9.534415  2.071660  \n",
       " GSM348023        6.120764  9.677157  1.572633  \n",
       " GSM348024        6.379439  9.656265  1.349807  \n",
       " GSM348027        6.530280  9.570464  1.015241  \n",
       " GSM348028        6.678943  9.466254  1.800471  \n",
       " GSM348029        5.569098  9.708121  2.713821  \n",
       " GSM348030        6.115254  9.776706  1.832250  \n",
       " GSM348031        5.502497  9.426018  1.839424  \n",
       " GSM348034        6.624473  9.184408  1.392186  \n",
       " GSM348035        6.640782  8.987193  1.500180  \n",
       " GSM348036        6.639867  9.450938  1.479329  \n",
       " GSM348037        5.810329  9.871557  3.027012  \n",
       " \n",
       " [60 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM437497        9.509530  7.919801  6.272163  5.643542  4.587800  4.261713   \n",
       " GSM437498        8.841789  6.778506  4.336394  5.376700  4.792615  4.826495   \n",
       " GSM437499        8.577323  8.013457  5.117394  6.172574  2.467535  5.407929   \n",
       " GSM437500        8.934966  7.479179  4.013222  6.409767  5.349339  5.487880   \n",
       " GSM437505        8.909306  7.193701  6.676346  7.072505  4.986106  5.238707   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM437497        4.484526  6.123732  6.955882  6.895190  ...  8.985134   \n",
       " GSM437498        3.492944  6.952810  6.556406  6.735041  ...  8.424020   \n",
       " GSM437499        2.221331  6.582855  6.702315  6.213456  ...  8.833436   \n",
       " GSM437500        3.407784  6.961701  5.855907  5.194310  ...  9.155182   \n",
       " GSM437505        3.625683  5.277593  5.911000  6.031786  ...  8.999475   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A     ZYG11B        ZYX  \\\n",
       " GSM437497        8.008377  5.923466  7.211588  2.737546  10.019002  11.752323   \n",
       " GSM437498        7.607496  5.930537  6.955358  3.591284  10.758192   8.884045   \n",
       " GSM437499        8.016164  4.636143  6.711084  2.164312  10.504001  11.295171   \n",
       " GSM437500        7.779597  5.472098  7.052723  2.140685   9.981278  10.213076   \n",
       " GSM437505        8.758062  6.477068  7.251737  3.921004  10.172665  10.628928   \n",
       " \n",
       " gene_symbol_sql     ZZEF1       ZZZ3   mir-223  \n",
       " GSM437497        6.757282   9.352768  2.349795  \n",
       " GSM437498        6.305815  10.392785  2.497879  \n",
       " GSM437499        6.302416  10.201615  3.527976  \n",
       " GSM437500        5.336961  10.171880  2.270757  \n",
       " GSM437505        6.461305  10.259153  2.212467  \n",
       " \n",
       " [5 rows x 22899 columns],\n",
       " gene_symbol_sql     A1BG  A1BG-AS1      A1CF       A2M  A2M-AS1     A2ML1  \\\n",
       " GSM2501496       7.68068   5.66804  3.986515  5.821320  4.02699  3.812240   \n",
       " GSM2501497       7.68470   6.13055  4.032985  6.097565  5.14810  3.775240   \n",
       " GSM2501498       7.70576   5.95725  4.058630  6.159765  4.95483  3.844475   \n",
       " GSM2501499       7.83548   5.50992  4.116405  5.985350  4.38769  3.856730   \n",
       " GSM2501500       7.85847   5.80005  4.040150  6.076410  4.18492  3.697635   \n",
       " GSM2501501       7.42353   5.88521  4.031390  8.111435  4.88979  3.866625   \n",
       " GSM2501502       7.52765   5.97610  4.254775  8.055510  5.20655  3.825150   \n",
       " GSM2501503       7.59949   5.87764  4.112810  7.498565  4.65512  3.840210   \n",
       " GSM2501504       7.57404   5.49199  4.099660  7.627040  4.62390  3.796700   \n",
       " GSM2501505       8.10834   6.68770  4.191200  5.259555  4.21684  3.909995   \n",
       " GSM2501506       8.15976   6.35554  4.250375  5.370990  4.00505  3.891195   \n",
       " GSM2501507       6.98981   5.84819  4.122160  5.618495  4.24128  3.906330   \n",
       " GSM2501508       7.35761   5.68558  3.918180  5.133085  4.26860  3.816515   \n",
       " GSM2501478       7.70333   6.25623  4.222840  5.735955  4.42678  3.857080   \n",
       " GSM2501479       8.22974   5.59436  4.040615  6.065780  4.00284  3.778610   \n",
       " GSM2501490       8.57643   6.12568  4.617400  7.288355  5.65851  4.457065   \n",
       " GSM2501491       8.01677   5.91053  4.238915  6.175300  4.26905  3.719885   \n",
       " GSM2501492       7.99193   5.87268  4.130625  6.103990  4.40980  3.744190   \n",
       " GSM2501493       7.64785   5.97003  4.234610  6.577115  5.58465  3.906535   \n",
       " GSM2501494       7.76039   5.67831  4.146500  6.337975  5.68038  3.915135   \n",
       " GSM2501495       7.72497   5.67896  4.167440  6.057875  4.44804  3.792940   \n",
       " GSM2501480       8.05126   5.64122  4.114350  5.748060  4.17551  3.804700   \n",
       " GSM2501481       7.94213   6.30015  4.034530  6.531360  5.23966  3.826555   \n",
       " GSM2501482       7.92719   6.34650  3.946045  6.471705  5.40108  3.799695   \n",
       " GSM2501483       6.82533   5.83034  4.106195  6.201710  4.09596  3.755245   \n",
       " GSM2501484       6.76029   5.54161  4.181875  5.929185  4.04867  3.573435   \n",
       " GSM2501485       7.57882   5.91450  4.328945  5.795575  4.64058  4.032770   \n",
       " GSM2501486       7.39456   6.08289  4.284745  5.787515  4.59242  3.787175   \n",
       " GSM2501487       7.37731   5.75112  4.263990  4.832190  4.44886  3.817515   \n",
       " GSM2501488       7.45418   5.82586  4.289835  4.936205  4.25676  3.855715   \n",
       " GSM2501489       8.13959   5.78440  4.210735  6.743675  5.56226  3.719175   \n",
       " GSM2501477       7.50383   6.18807  4.314045  5.695750  4.55881  3.927595   \n",
       " \n",
       " gene_symbol_sql    A2MP1   A4GALT    A4GNT     AA06  ...    ZWINT     ZXDA  \\\n",
       " GSM2501496       4.92102  6.67230  5.83174  5.70128  ...  7.90669  7.17334   \n",
       " GSM2501497       4.87385  6.25390  5.99527  5.49910  ...  7.07475  7.26679   \n",
       " GSM2501498       5.19964  6.17956  6.27970  5.68454  ...  6.93639  7.16691   \n",
       " GSM2501499       4.77620  6.59578  5.94943  5.43561  ...  7.79083  7.08855   \n",
       " GSM2501500       4.93601  6.51410  5.64960  5.53239  ...  7.64852  7.10288   \n",
       " GSM2501501       4.98129  6.44685  5.94125  5.21718  ...  7.04922  7.17330   \n",
       " GSM2501502       5.08177  6.15968  6.03933  5.42342  ...  7.12971  7.06856   \n",
       " GSM2501503       4.98903  6.50816  5.85521  5.15101  ...  8.67732  7.53606   \n",
       " GSM2501504       5.07092  6.43975  6.09880  5.62235  ...  8.55348  7.40199   \n",
       " GSM2501505       5.06856  6.60773  6.01533  5.51471  ...  7.45862  7.42429   \n",
       " GSM2501506       5.11166  6.71694  6.00817  5.50569  ...  7.63055  7.16007   \n",
       " GSM2501507       5.02132  6.26089  5.79910  5.66944  ...  8.79507  7.62957   \n",
       " GSM2501508       5.12335  6.62034  5.78891  5.70129  ...  8.88128  7.95402   \n",
       " GSM2501478       5.05051  6.57092  5.90983  5.56124  ...  7.41478  7.45621   \n",
       " GSM2501479       4.92855  6.53514  6.33951  5.89566  ...  7.27444  6.92702   \n",
       " GSM2501490       4.91306  6.78062  6.41482  5.93696  ...  7.82122  7.57819   \n",
       " GSM2501491       5.14511  6.70984  5.86330  5.14665  ...  8.81038  7.35940   \n",
       " GSM2501492       5.21645  6.42661  5.90976  5.20263  ...  8.65967  7.66839   \n",
       " GSM2501493       4.93042  6.61186  6.05568  5.30124  ...  7.65579  7.33771   \n",
       " GSM2501494       4.92500  6.71516  6.06866  5.82833  ...  7.49437  7.31350   \n",
       " GSM2501495       5.31288  6.66871  5.99097  5.69526  ...  8.03119  6.28814   \n",
       " GSM2501480       4.97994  6.76710  5.99575  5.54821  ...  7.32936  6.90909   \n",
       " GSM2501481       5.16944  6.71948  5.92040  5.63588  ...  8.59846  6.88714   \n",
       " GSM2501482       4.64919  6.60543  5.77355  5.26634  ...  8.52983  7.25023   \n",
       " GSM2501483       5.30305  6.52874  5.68830  5.46947  ...  8.69154  7.47928   \n",
       " GSM2501484       4.94311  6.73248  5.96113  5.54300  ...  8.57016  7.84936   \n",
       " GSM2501485       5.45630  6.88943  6.14027  5.83029  ...  7.36050  7.33305   \n",
       " GSM2501486       4.89105  6.92132  6.07756  5.70698  ...  7.39228  7.44645   \n",
       " GSM2501487       5.25074  7.04150  5.78911  5.52695  ...  9.42320  7.86690   \n",
       " GSM2501488       5.24461  6.94261  5.81831  5.72595  ...  9.00615  7.51170   \n",
       " GSM2501489       5.09161  6.60945  6.05271  5.61220  ...  7.63387  7.00221   \n",
       " GSM2501477       5.15205  6.70504  6.04717  5.86233  ...  7.33951  7.55444   \n",
       " \n",
       " gene_symbol_sql      ZXDB      ZXDC   ZYG11A     ZYG11B       ZYX     ZZEF1  \\\n",
       " GSM2501496       5.984220  7.729902  6.69746  10.017387  7.278525  6.330057   \n",
       " GSM2501497       5.518940  6.910037  5.29723   9.822927  7.146115  6.273380   \n",
       " GSM2501498       5.534045  6.788132  5.45774   9.811330  7.182875  6.316833   \n",
       " GSM2501499       5.637310  7.355333  5.82923  10.238010  7.034160  6.391373   \n",
       " GSM2501500       5.489342  7.227843  6.16331  10.178650  7.089080  6.281690   \n",
       " GSM2501501       5.580525  6.981965  5.03308   9.762497  7.224640  6.418243   \n",
       " GSM2501502       5.514658  6.934332  5.14585   9.706777  7.212340  6.408950   \n",
       " GSM2501503       5.653220  7.134025  5.60865   9.909703  6.783930  6.202377   \n",
       " GSM2501504       5.673845  7.025188  5.64888   9.871987  6.966850  6.233910   \n",
       " GSM2501505       5.667262  6.441570  5.06840   9.627590  7.125575  6.078550   \n",
       " GSM2501506       5.624783  6.583303  5.26290   9.550667  7.174940  6.302050   \n",
       " GSM2501507       6.436875  6.624503  6.06892   9.578250  6.923630  6.212397   \n",
       " GSM2501508       6.459505  7.069918  5.45618   9.918820  7.141500  6.221783   \n",
       " GSM2501478       5.573445  6.839000  5.55039   9.721047  6.754650  6.362620   \n",
       " GSM2501479       5.870850  7.530030  5.98723  10.148473  7.567835  6.389470   \n",
       " GSM2501490       5.506200  6.488870  6.03622   9.783060  7.376800  6.310267   \n",
       " GSM2501491       5.723978  6.531843  5.93920   9.419920  6.884080  5.978743   \n",
       " GSM2501492       5.972990  6.432245  6.32295   9.413970  6.783230  5.978920   \n",
       " GSM2501493       5.491450  6.662625  5.28700   9.872450  7.570575  6.425780   \n",
       " GSM2501494       5.507735  6.740830  5.16581   9.894827  7.472055  6.430713   \n",
       " GSM2501495       5.819070  7.614077  6.49359   9.694830  7.287610  6.398497   \n",
       " GSM2501480       5.809175  7.605517  6.31066  10.222330  7.554005  6.446503   \n",
       " GSM2501481       5.451065  6.496425  5.23966   9.295643  7.259710  6.028407   \n",
       " GSM2501482       5.441847  6.564853  5.12114   9.337053  7.349570  6.098647   \n",
       " GSM2501483       6.338150  6.883805  6.25503   9.253257  7.056355  6.134760   \n",
       " GSM2501484       6.277163  6.935808  6.03962   9.433847  6.914825  6.169333   \n",
       " GSM2501485       5.639868  6.669438  5.02099   9.514113  7.152320  6.344897   \n",
       " GSM2501486       5.557143  6.686858  5.34189   9.597067  7.205110  6.248030   \n",
       " GSM2501487       6.114997  7.088343  6.19129   9.661690  7.573020  6.086730   \n",
       " GSM2501488       6.093947  7.077550  5.98251   9.733127  7.277060  6.010357   \n",
       " GSM2501489       5.466817  6.377643  5.11814   9.659970  7.224895  6.157750   \n",
       " GSM2501477       5.682985  6.794918  5.67226   9.811450  6.792375  6.309787   \n",
       " \n",
       " gene_symbol_sql       ZZZ3  mir-223  \n",
       " GSM2501496        8.911835  4.89795  \n",
       " GSM2501497        9.528290  4.55629  \n",
       " GSM2501498        9.422840  4.70443  \n",
       " GSM2501499        9.759360  4.56043  \n",
       " GSM2501500        9.777805  4.64745  \n",
       " GSM2501501        9.359610  4.27569  \n",
       " GSM2501502        9.246480  4.60466  \n",
       " GSM2501503        9.887565  4.62773  \n",
       " GSM2501504        9.874800  4.56685  \n",
       " GSM2501505        9.618375  4.70304  \n",
       " GSM2501506        9.573795  4.81935  \n",
       " GSM2501507        9.864010  4.74783  \n",
       " GSM2501508        9.768990  4.56505  \n",
       " GSM2501478        9.512555  4.54577  \n",
       " GSM2501479        9.656430  4.70304  \n",
       " GSM2501490        9.581850  4.80698  \n",
       " GSM2501491        9.413605  4.90013  \n",
       " GSM2501492        9.422040  4.65424  \n",
       " GSM2501493        9.470815  4.72280  \n",
       " GSM2501494        9.462210  4.46161  \n",
       " GSM2501495        8.836830  4.73483  \n",
       " GSM2501480        9.715970  4.67868  \n",
       " GSM2501481        9.708545  4.58000  \n",
       " GSM2501482        9.743285  4.47930  \n",
       " GSM2501483       10.020510  4.71106  \n",
       " GSM2501484        9.969525  4.64323  \n",
       " GSM2501485        9.627730  4.99219  \n",
       " GSM2501486        9.613165  4.55140  \n",
       " GSM2501487        9.956090  4.85755  \n",
       " GSM2501488        9.966870  4.83487  \n",
       " GSM2501489        9.614935  4.48409  \n",
       " GSM2501477        9.486805  4.40959  \n",
       " \n",
       " [32 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM1341459       8.072369  6.890171  3.837661  4.442063  5.073472  3.819422   \n",
       " GSM1341460       7.910755  6.204118  3.914902  4.651479  5.303237  3.849092   \n",
       " GSM1341461       7.939848  6.568983  4.188444  8.981795  5.023594  3.895109   \n",
       " GSM1341462       7.845868  6.310837  4.084831  8.947458  4.867855  3.668693   \n",
       " GSM1341463       7.059638  5.363085  4.047479  4.697086  5.130633  3.975655   \n",
       " GSM1341464       7.053228  5.562353  4.140363  4.812391  5.080964  3.892402   \n",
       " GSM1341471       6.850599  5.650510  4.010205  4.301376  5.079896  4.738279   \n",
       " GSM1341472       6.921486  5.447529  4.200198  4.748936  4.828091  4.480314   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM1341459       3.892644  6.880820  4.705914  4.449593  ...   9.513997   \n",
       " GSM1341460       4.085885  6.659783  4.795204  4.435540  ...   9.542149   \n",
       " GSM1341461       4.044657  6.482842  5.119786  4.775548  ...   9.859337   \n",
       " GSM1341462       4.073140  6.571566  4.787731  4.833817  ...   9.891447   \n",
       " GSM1341463       3.884062  6.252199  4.788157  4.422546  ...  10.948030   \n",
       " GSM1341464       3.954390  6.236326  4.886975  4.200831  ...  10.948234   \n",
       " GSM1341471       3.840499  6.257966  4.683612  4.058137  ...  10.953061   \n",
       " GSM1341472       4.224002  6.276546  4.642301  4.379628  ...  11.012062   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM1341459       2.557027  4.858567  6.639362  5.929260  9.343552  8.951335   \n",
       " GSM1341460       2.724516  4.713329  6.707549  6.008604  9.248559  8.984049   \n",
       " GSM1341461       2.698994  5.025164  6.441848  5.447785  9.013940  9.499690   \n",
       " GSM1341462       2.770434  5.196032  6.452339  5.679015  9.003679  9.389930   \n",
       " GSM1341463       2.831510  5.063148  7.124290  6.594950  8.436795  7.625346   \n",
       " GSM1341464       2.792952  5.211241  7.122378  6.589319  8.413562  7.634054   \n",
       " GSM1341471       2.900462  5.150297  6.872273  7.205524  8.545254  7.854530   \n",
       " GSM1341472       2.727320  5.091449  6.810340  7.151144  8.509055  7.861389   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM1341459       6.269176  8.305600  3.424984  \n",
       " GSM1341460       6.183583  8.348071  3.401091  \n",
       " GSM1341461       6.491929  8.527556  3.366582  \n",
       " GSM1341462       6.318893  8.611661  3.466260  \n",
       " GSM1341463       5.806146  8.823030  3.317550  \n",
       " GSM1341464       5.799353  8.830029  3.334514  \n",
       " GSM1341471       5.785328  8.830079  3.478077  \n",
       " GSM1341472       5.812805  8.774528  3.460211  \n",
       " \n",
       " [8 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM829480        6.460859  3.596079  2.064032  2.039959  1.613203  2.236882   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...      ZWINT  \\\n",
       " GSM829480        1.615982  4.072155  1.780229  2.396128  ...  12.119001   \n",
       " \n",
       " gene_symbol_sql      ZXDA     ZXDB      ZXDC    ZYG11A    ZYG11B     ZYX  \\\n",
       " GSM829480        2.751472  3.49567  4.385061  2.795914  6.930505  5.4706   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM829480        3.709023  7.764043  2.021147  \n",
       " \n",
       " [1 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM1159729       1.573349  0.307025  0.192127 -1.033964  0.321707  0.104977   \n",
       " GSM1159730       0.743395  0.055620  0.104555  1.265995 -0.000187  0.047044   \n",
       " GSM1159731       1.356688  0.423817  0.165298 -0.165945  0.000000  0.112650   \n",
       " GSM1159732       1.224016  0.561399 -0.050161  0.721479 -0.083012 -0.057265   \n",
       " GSM1159733       1.284739  0.492218 -0.090795 -1.019220  0.002872 -0.106431   \n",
       " GSM1159765       0.581281  0.056198  0.192361  2.679726 -0.398259 -0.170109   \n",
       " GSM1159766       0.601573  0.302614 -0.090116 -0.965010 -0.245102 -0.174261   \n",
       " GSM1159767       0.330869 -0.276407  0.335232  1.123372 -1.001134 -0.142887   \n",
       " GSM1159768       0.211904  0.314269 -0.044615  2.073890 -0.073077 -0.100112   \n",
       " GSM1159769       0.583959  0.022679 -0.231417 -0.231424 -0.082861 -0.022928   \n",
       " GSM1159770       0.813584  0.221729 -0.071583  2.542400 -0.835883 -0.277729   \n",
       " GSM1159771       1.161192  0.172973 -0.170111 -1.693255 -0.771644 -0.284284   \n",
       " GSM1159772       0.958368  0.605826  0.056074 -2.557296 -0.493691 -0.124672   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...    ZWILCH  \\\n",
       " GSM1159729       0.194453 -0.325711  0.274387  0.178527  ... -0.816996   \n",
       " GSM1159730       0.027636  0.088265  0.245535  0.304712  ... -1.173556   \n",
       " GSM1159731      -0.115428 -0.007199  0.029064  0.102423  ... -0.545377   \n",
       " GSM1159732       0.028091 -0.052250  0.171794  0.252611  ... -1.120101   \n",
       " GSM1159733      -0.065732 -0.011472  0.064887  0.034987  ... -0.497506   \n",
       " GSM1159765       0.103031  0.114339  0.014778  0.341397  ... -1.733923   \n",
       " GSM1159766      -0.136006  0.068622 -0.051174  0.236575  ... -0.491775   \n",
       " GSM1159767      -0.160150 -0.068404  0.137142  0.306000  ... -1.198174   \n",
       " GSM1159768       0.054028  0.077365 -0.236028  0.181144  ...  1.287775   \n",
       " GSM1159769      -0.384485  0.324101  0.319045 -0.184169  ... -0.455040   \n",
       " GSM1159770      -0.205743  0.354974 -0.388146  0.085793  ... -1.931166   \n",
       " GSM1159771      -0.039762  0.072676  0.129694 -0.244566  ... -1.585692   \n",
       " GSM1159772      -0.038013  0.531881 -0.171852  0.105212  ... -0.149230   \n",
       " \n",
       " gene_symbol_sql     ZWINT      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B  \\\n",
       " GSM1159729      -1.930802 -1.359089 -0.095326 -0.303924 -0.617717  0.186145   \n",
       " GSM1159730      -2.986377  0.526247  0.163162 -0.066254 -0.721199  0.031122   \n",
       " GSM1159731      -0.876024 -1.082891  0.184468 -0.113625  0.085550 -0.380461   \n",
       " GSM1159732      -3.029459 -1.594985  0.150186 -0.090487 -0.413927 -0.105257   \n",
       " GSM1159733      -0.896302 -1.408682  0.209985 -0.124027  0.386749 -0.330142   \n",
       " GSM1159765      -3.483896 -0.620678 -0.514713 -0.279649 -0.133668  0.760476   \n",
       " GSM1159766      -0.778399  0.738272  0.056174 -0.387150  0.305682  0.420097   \n",
       " GSM1159767      -2.749197  0.268022  0.192824 -0.150021 -0.395432  0.486486   \n",
       " GSM1159768       0.831235  0.295076 -0.014504 -0.275839  0.786587  0.024499   \n",
       " GSM1159769      -1.114967  0.665074 -0.039594 -0.260537 -0.046523  0.663982   \n",
       " GSM1159770      -3.104749  1.035536  0.115000  0.134729 -0.865054  0.945054   \n",
       " GSM1159771      -3.169541  0.820101  0.308103  0.923308 -0.076711  0.881739   \n",
       " GSM1159772      -0.584216  0.386726 -0.022394 -0.203999 -0.249186  0.705548   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM1159729       2.472630  0.087775 -0.063193  \n",
       " GSM1159730       1.836432  0.052225  0.066342  \n",
       " GSM1159731       0.717133  0.028742 -0.333524  \n",
       " GSM1159732       1.188045  0.150699 -0.178862  \n",
       " GSM1159733       0.650307 -0.000185 -0.265851  \n",
       " GSM1159765       1.739825  0.044540 -0.475739  \n",
       " GSM1159766      -0.685991 -0.217220  0.042067  \n",
       " GSM1159767       1.802631  0.169572 -0.176982  \n",
       " GSM1159768       2.451052  0.211612 -0.009940  \n",
       " GSM1159769      -0.412348  0.034464  0.093935  \n",
       " GSM1159770      -0.708894 -0.017800  0.337427  \n",
       " GSM1159771      -0.642224 -0.048786  0.945359  \n",
       " GSM1159772      -0.602981  0.143003  0.106020  \n",
       " \n",
       " [13 rows x 19236 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM2185129       1.126946  1.358610 -0.994463  1.983724  0.219329 -0.736245   \n",
       " GSM2185130      -0.104885 -1.794352 -0.061043  0.075480 -0.144924  0.909440   \n",
       " GSM2185131      -0.189461  0.401208  0.592763 -1.519703  0.823903  1.012378   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM2185129      -0.913379  0.183287 -0.082204 -1.127490  ... -3.238321   \n",
       " GSM2185130      -0.003841 -0.736600  0.641501  0.904915  ... -0.116529   \n",
       " GSM2185131       0.700898  1.032552  0.633241  0.923555  ... -0.004576   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM2185129       0.824628 -0.111698  0.025049 -5.904938  1.196831 -0.236090   \n",
       " GSM2185130      -0.062430 -0.475934  0.028365  0.959862  0.277075 -0.500534   \n",
       " GSM2185131      -0.548265 -0.462800  0.096202  0.263001  0.486873 -0.319178   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM2185129      -0.315992  0.266770 -1.448330  \n",
       " GSM2185130      -1.219084  0.241493 -1.140964  \n",
       " GSM2185131      -0.209880  0.204159 -0.180012  \n",
       " \n",
       " [3 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM1425030       7.294449  5.735731  3.501159  6.453595  5.679464  3.275071   \n",
       " GSM1425031       7.618814  6.022397  3.450450  5.404776  5.493630  3.408397   \n",
       " GSM1425032       7.425886  5.880119  3.581446  5.439156  5.198630  3.436430   \n",
       " GSM1425029       6.929945  5.473971  3.345812  6.289693  6.330881  3.397105   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM1425030       3.559293  6.022206  5.422559  4.354274  ...  8.517884   \n",
       " GSM1425031       3.590772  5.865624  5.310318  4.381584  ...  8.195164   \n",
       " GSM1425032       3.382851  5.951146  5.477768  4.390286  ...  7.888685   \n",
       " GSM1425029       3.224939  5.552455  5.249955  3.851083  ...  8.902539   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX  \\\n",
       " GSM1425030       3.271991  4.422939  7.252252  4.912672  9.418416  5.859724   \n",
       " GSM1425031       3.045548  4.756873  7.437598  4.340603  9.511356  5.980924   \n",
       " GSM1425032       5.249400  5.084715  7.299560  4.202887  9.758034  5.919900   \n",
       " GSM1425029       3.020549  4.738161  7.261193  4.593856  9.631742  5.926453   \n",
       " \n",
       " gene_symbol_sql     ZZEF1      ZZZ3   mir-223  \n",
       " GSM1425030       6.635029  8.334010  3.048558  \n",
       " GSM1425031       6.718069  8.392501  2.934414  \n",
       " GSM1425032       6.756294  8.629649  3.340254  \n",
       " GSM1425029       6.446218  8.568736  2.644278  \n",
       " \n",
       " [4 rows x 22899 columns],\n",
       " gene_symbol_sql      A1BG  A1BG-AS1      A1CF       A2M   A2M-AS1     A2ML1  \\\n",
       " GSM1047773       5.590724   4.78722  3.070014  5.534397  2.778957  3.302858   \n",
       " \n",
       " gene_symbol_sql     A2MP1    A4GALT     A4GNT      AA06  ...     ZWINT  \\\n",
       " GSM1047773       3.086041  5.608562  4.513006  3.720167  ...  9.846012   \n",
       " \n",
       " gene_symbol_sql      ZXDA      ZXDB      ZXDC    ZYG11A    ZYG11B      ZYX  \\\n",
       " GSM1047773       5.622755  4.110455  5.815364  4.977733  7.968081  9.13172   \n",
       " \n",
       " gene_symbol_sql     ZZEF1     ZZZ3   mir-223  \n",
       " GSM1047773       5.535268  7.61695  2.485795  \n",
       " \n",
       " [1 rows x 22899 columns],\n",
       " gene_symbol_sql      A1CF       A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM298011       -1.280835  0.582077 -0.402420 -1.374544 -0.120057 -0.942479   \n",
       " GSM298010       -0.958783  2.296092  0.475701 -1.223990 -0.375271 -1.013296   \n",
       " GSM298008       -1.346684  2.372782  0.483578 -1.338894 -0.345152 -1.314742   \n",
       " GSM298009       -1.416373  1.573663  0.227092 -1.304178 -0.297424 -1.109826   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM298011       -0.459081 -0.235873 -0.216148 -0.106747  ... -0.110508   \n",
       " GSM298010       -0.228310 -0.121799 -0.355192  0.097756  ... -0.285477   \n",
       " GSM298008       -0.233305  0.169065 -0.359752  0.152448  ... -0.146287   \n",
       " GSM298009       -0.392102  0.038409 -0.336998 -0.057155  ... -0.054734   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH     ZWINT      ZXDB      ZXDC  \\\n",
       " GSM298011         -0.010126 -0.195954 -0.649167 -1.587831 -0.113803 -0.468947   \n",
       " GSM298010         -0.251623 -0.136964 -0.721481 -1.619217  0.282670 -0.272640   \n",
       " GSM298008         -0.714489 -0.321190 -0.997834 -1.642898 -0.078793 -0.604118   \n",
       " GSM298009         -0.022629 -0.427346 -0.418480 -0.735677 -0.375744 -0.236680   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM298011        1.035308 -0.524923  0.327069  \n",
       " GSM298010        0.692765 -0.453071  0.515670  \n",
       " GSM298008        0.708078 -0.205702  0.086053  \n",
       " GSM298009        0.656961 -0.477317  0.346318  \n",
       " \n",
       " [4 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF        A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM645644        6.472481  12.958161  5.759814  6.795208  7.662376  9.192394   \n",
       " GSM645646        5.965895  13.118589  4.828987  6.307214  7.210233  9.407580   \n",
       " GSM645647        6.377710  13.254410  6.496055  4.151997  7.143353  9.708251   \n",
       " GSM645645        6.670061  12.946301  5.098356  6.680226  7.725891  9.355684   \n",
       " GSM645651        6.776762  14.106113  6.197624  7.006612  7.367327  9.409957   \n",
       " GSM645648        6.611268  13.692223  4.001694  6.315503  8.274951  9.362029   \n",
       " GSM645650        3.921589  13.016707  5.077405  6.616352  7.468453  9.444472   \n",
       " GSM645649        7.167880  14.126156  5.174115  7.154261  7.355184  8.912958   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM645644        2.479130  8.328055  5.581704  6.860130  ...  6.805885   \n",
       " GSM645646        3.359170  8.323790  6.784234  7.802269  ...  6.458971   \n",
       " GSM645647        2.471598  7.834549  6.795283  8.020667  ...  5.845738   \n",
       " GSM645645        2.652044  8.894776  6.427602  6.700439  ...  5.311284   \n",
       " GSM645651        3.343294  7.385234  5.926594  7.887164  ...  7.063789   \n",
       " GSM645648        2.329017  8.286348  6.043164  6.631614  ...  6.011843   \n",
       " GSM645650        2.856990  7.789219  5.458517  7.329741  ...  5.082325   \n",
       " GSM645649        2.685828  8.402116  5.912454  8.299016  ...  4.643869   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH      ZWINT      ZXDB  \\\n",
       " GSM645644          5.962496  9.230294  7.931990   9.910883  3.840637   \n",
       " GSM645646          6.969001  9.882235  7.976644   9.886539  4.527652   \n",
       " GSM645647          7.068886  9.805549  8.552066   9.696609  4.860518   \n",
       " GSM645645          5.222608  9.532564  8.187382   9.645622  5.127282   \n",
       " GSM645651          5.339337  9.422429  8.237043  10.237903  5.426114   \n",
       " GSM645648          6.170798  9.818964  8.297444   9.936058  3.551328   \n",
       " GSM645650          7.073167  9.030013  8.128500  11.250121  4.892216   \n",
       " GSM645649          5.151839  9.917054  7.708608   9.964000  4.811188   \n",
       " \n",
       " gene_symbol_sql      ZXDC        ZYX     ZZEF1       ZZZ3  \n",
       " GSM645644        8.859861  10.893173  7.007374  10.858354  \n",
       " GSM645646        8.805244  10.102762  7.666576  10.760304  \n",
       " GSM645647        9.539141  11.119862  7.546433  10.305879  \n",
       " GSM645645        9.452153  11.318036  8.547185  10.946387  \n",
       " GSM645651        8.300476  10.079719  7.735014  11.922956  \n",
       " GSM645648        8.061134  10.788610  7.954869  11.697862  \n",
       " GSM645650        8.305756  10.832041  7.861274  10.981303  \n",
       " GSM645649        8.075367  10.350498  7.320623  12.060821  \n",
       " \n",
       " [8 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF       A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM356968        2.596632  3.058192  2.534525 -1.697289  3.766309  6.769811   \n",
       " GSM356966        3.091857 -2.554714  3.238377  2.789738  4.350257  6.729294   \n",
       " GSM356967        2.354430  2.223404  1.854562  2.495153  4.422327  7.127396   \n",
       " GSM356965        3.517238  1.677666  2.323952  1.621717  4.222550  7.654249   \n",
       " GSM356970        3.379718  2.062267  3.671950  2.560321  4.266697  6.840060   \n",
       " GSM356971        1.604422  3.369522  2.212000  1.025525  4.725839  7.136919   \n",
       " GSM356973        3.414826  0.737929  2.880029  3.317612  4.530708  6.904111   \n",
       " GSM356974        3.730618  1.227187  1.934800  3.267939  4.596709  7.676860   \n",
       " GSM356972        3.343848  2.990058  3.179269  2.015644  4.323954  7.829621   \n",
       " GSM356969        3.197455  1.655091  1.165725  1.498920  4.760205  6.808462   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM356968       -1.100682  4.401347  1.745688  4.978007  ...  2.683661   \n",
       " GSM356966        1.599113  5.288698  2.288504  4.945360  ...  3.361785   \n",
       " GSM356967       -1.196862  5.567152  1.946815  4.877987  ...  2.315896   \n",
       " GSM356965       -0.904494  5.370391  2.027492  4.136597  ...  2.196649   \n",
       " GSM356970        1.934962  4.100005  1.506034  4.327391  ...  2.500001   \n",
       " GSM356971       -0.701295  5.507140  1.386993  5.085971  ...  2.205325   \n",
       " GSM356973        0.442609  5.172066  2.535224  4.897836  ...  1.903091   \n",
       " GSM356974        0.920834  4.630338  1.729727  4.261033  ...  3.358283   \n",
       " GSM356972        2.480107  5.644354  2.517294  4.245248  ...  2.778797   \n",
       " GSM356969       -1.838858  3.941992  2.111861  4.153865  ...  2.949003   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH     ZWINT      ZXDB      ZXDC  \\\n",
       " GSM356968          2.468839  5.986693  5.067776  4.456655  1.258033  3.727594   \n",
       " GSM356966          1.524886  6.243717  5.209956  4.840604  0.915955  3.449178   \n",
       " GSM356967          3.219619  6.349530  6.980254  7.865665  1.800974  3.651109   \n",
       " GSM356965          2.971742  6.220640  6.284301  6.390098  1.252915  4.193149   \n",
       " GSM356970          0.670559  5.855900  4.768475  5.112354  1.780909  2.509394   \n",
       " GSM356971          2.403017  5.824844  4.639967  4.887491  2.388526  3.463701   \n",
       " GSM356973          3.472618  5.739381  4.773226  5.149511  2.109809  1.279994   \n",
       " GSM356974          0.507028  6.577429  6.385888  7.368899  2.055716  3.953209   \n",
       " GSM356972          2.043908  6.604718  7.159871  8.879954  1.727050  4.275655   \n",
       " GSM356969          0.267548  6.036530  5.234363  4.507909  3.093712  3.677204   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM356968        8.186518  2.573806  7.668367  \n",
       " GSM356966        8.227770  3.058030  7.607648  \n",
       " GSM356967        8.276556  2.853345  7.477904  \n",
       " GSM356965        8.520444  2.892278  7.560715  \n",
       " GSM356970        8.346284  2.610656  7.770360  \n",
       " GSM356971        8.358574  2.696342  7.710827  \n",
       " GSM356973        8.603646  3.894645  7.501980  \n",
       " GSM356974        8.895973  2.865752  7.569210  \n",
       " GSM356972        8.561755  2.628616  7.487551  \n",
       " GSM356969        8.041015  3.259323  7.586209  \n",
       " \n",
       " [10 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF       A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM255845        5.258400  8.445408  6.392799  3.711353  7.632066  8.309018   \n",
       " GSM255842        5.246098  8.393318  6.310312  3.889753  7.343475  8.245147   \n",
       " GSM255843        5.334581  8.515696  6.369100  4.018896  7.442434  8.422519   \n",
       " GSM255844        5.102829  8.291514  6.335798  3.730103  7.829182  8.356292   \n",
       " GSM255846        5.277678  8.603588  6.375723  3.704630  7.764029  8.301518   \n",
       " GSM255841        5.268867  9.076537  6.498850  4.081580  7.571869  8.622439   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM255845        3.869235  6.710664  5.020350  5.639232  ...  5.667646   \n",
       " GSM255842        3.954053  6.391701  4.944015  5.844507  ...  5.692757   \n",
       " GSM255843        4.003458  6.312584  4.940578  5.964217  ...  5.696075   \n",
       " GSM255844        3.937166  6.648905  5.060742  5.572447  ...  5.688546   \n",
       " GSM255846        3.722446  6.576038  4.958024  5.830541  ...  5.736052   \n",
       " GSM255841        4.205789  6.574883  5.100202  6.236709  ...  5.960410   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH     ZWINT      ZXDB      ZXDC  \\\n",
       " GSM255845          5.719804  8.106559  6.411427  6.792457  4.460252  6.878664   \n",
       " GSM255842          5.826539  8.119807  6.545577  6.299172  5.084311  7.035070   \n",
       " GSM255843          5.768360  8.130543  6.402062  6.570916  4.985050  6.993543   \n",
       " GSM255844          5.809502  7.941251  6.820882  6.701568  4.500475  6.797826   \n",
       " GSM255846          5.541592  8.162608  5.963824  6.198808  4.476584  6.854883   \n",
       " GSM255841          6.134205  8.326299  4.881732  6.074515  4.883153  7.463345   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM255845        9.654946  5.824981  8.819998  \n",
       " GSM255842        9.426108  5.717895  8.990154  \n",
       " GSM255843        9.574814  5.708670  9.068672  \n",
       " GSM255844        9.679788  5.736804  8.659692  \n",
       " GSM255846        9.532776  5.891889  8.757321  \n",
       " GSM255841        9.906390  5.946731  9.107105  \n",
       " \n",
       " [6 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF      A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM288880        4.425513  5.63953  4.208408  1.324309  5.792314  7.071441   \n",
       " GSM288878        5.201983  8.37797  4.031263  3.901099  5.852009  7.091308   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM288880        3.379663  5.424575  2.708572  4.553272  ...  4.025842   \n",
       " GSM288878        0.620436  4.598398  2.793934  4.130354  ...  3.661784   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH     ZWINT      ZXDB      ZXDC  \\\n",
       " GSM288880          4.189216  6.318588  6.007642  6.679733  2.631525  4.369606   \n",
       " GSM288878          4.788341  6.549275  5.125221  5.544903  2.467490  4.659873   \n",
       " \n",
       " gene_symbol_sql       ZYX     ZZEF1      ZZZ3  \n",
       " GSM288880        9.250244  3.150141  6.844235  \n",
       " GSM288878        8.879560  3.640747  6.553029  \n",
       " \n",
       " [2 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF       A2M    A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM89653         6.341307  3.979470  3.978688  2.788460  7.305061  8.173000   \n",
       " GSM89654         6.912836  4.432174  4.195665  2.909112  7.858615  6.968724   \n",
       " GSM89655         6.709305  4.829518  4.319417  3.330661  6.691548  7.695138   \n",
       " GSM89656         5.776579  3.404889  3.892898  2.629252  7.484437  8.898735   \n",
       " GSM89657         5.804163  3.869934  3.905375  2.626566  7.208326  8.898521   \n",
       " GSM89658         5.749384  4.027187  3.879037  2.601286  7.045078  8.738208   \n",
       " GSM89652         6.418760  4.105432  4.028216  2.701776  8.137205  7.932655   \n",
       " GSM89651         6.466957  4.179934  4.655307  2.822306  8.202361  6.936110   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1     AAMDC  ...    ZSWIM8  \\\n",
       " GSM89653         2.607225  5.878716  3.765023  8.151874  ...  5.454100   \n",
       " GSM89654         3.131252  5.217597  5.412802  9.044641  ...  6.024615   \n",
       " GSM89655         3.122941  4.681622  4.443866  8.272868  ...  6.612081   \n",
       " GSM89656         2.677498  5.894758  3.786850  7.589366  ...  5.658108   \n",
       " GSM89657         2.484699  5.627553  3.418469  7.360257  ...  5.423195   \n",
       " GSM89658         2.582771  5.747990  3.380776  7.448646  ...  5.305956   \n",
       " GSM89652         2.655236  5.744507  3.720459  8.428002  ...  5.454032   \n",
       " GSM89651         2.845519  5.343651  4.533858  9.060343  ...  5.442759   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH      ZWINT      ZXDB  \\\n",
       " GSM89653           3.157203  6.779642  8.007259  10.132721  3.021348   \n",
       " GSM89654           3.755683  1.960010  5.285554   5.753916  4.543028   \n",
       " GSM89655           3.788247  6.400456  4.895736   5.660916  4.438287   \n",
       " GSM89656           3.095871  7.034934  7.839357  10.290564  3.003992   \n",
       " GSM89657           3.705527  7.418474  8.107676   9.967060  3.124871   \n",
       " GSM89658           3.317770  7.147911  8.231276   9.980741  3.009796   \n",
       " GSM89652           3.391161  6.773161  7.572287   8.946971  3.786769   \n",
       " GSM89651           3.697262  5.417253  6.496835   7.615673  4.318896   \n",
       " \n",
       " gene_symbol_sql      ZXDC        ZYX     ZZEF1      ZZZ3  \n",
       " GSM89653         5.075180  10.210171  4.776074  7.408362  \n",
       " GSM89654         4.608647   9.682471  5.390067  3.706532  \n",
       " GSM89655         5.345214   8.758724  5.428663  5.267107  \n",
       " GSM89656         5.391069  11.059630  5.152459  7.984299  \n",
       " GSM89657         6.041628  11.070493  4.749739  7.847312  \n",
       " GSM89658         5.641084  10.671337  4.719887  7.435731  \n",
       " GSM89652         4.326583  11.358916  5.182842  6.730592  \n",
       " GSM89651         4.216464  10.716628  5.144206  5.376438  \n",
       " \n",
       " [8 rows x 13277 columns],\n",
       " gene_symbol_sql      A1CF       A2M   A4GALT     A4GNT      AAAS      AACS  \\\n",
       " GSM959042        7.150838  9.818474  7.31961  7.171042  9.032594  8.063631   \n",
       " \n",
       " gene_symbol_sql     AADAC     AAGAB      AAK1   AAMDC  ...    ZSWIM8  \\\n",
       " GSM959042        6.211536  8.015287  6.865059  7.0728  ...  7.464711   \n",
       " \n",
       " gene_symbol_sql  ZSWIM8-AS1      ZW10    ZWILCH     ZWINT      ZXDB      ZXDC  \\\n",
       " GSM959042          7.131519  8.612663  5.895113  6.567599  6.133235  7.297073   \n",
       " \n",
       " gene_symbol_sql        ZYX     ZZEF1      ZZZ3  \n",
       " GSM959042        11.237723  7.066841  9.029608  \n",
       " \n",
       " [1 rows x 13277 columns]]"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save all data frames as csv too\n",
    "with open('picklejar/fibroblasts.pckl', 'rb') as f:\n",
    "    in_f = pickle.load(f)\n",
    "    d_path = ntpath.join('mixed_dfs','fibroblasts'+'.csv')\n",
    "    in_f.T.to_csv(d_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
