{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DATA CONDENSTAION ##\n",
    "\n",
    "### OBJECTIVES ###\n",
    "\n",
    "1. What genes are most represented? Need to get above 5000 genes\n",
    "    - Do all GPLs hvae the same gene symbols?\n",
    "    - create a definitive gene symbol list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pickle\n",
    "import ntpath\n",
    "import mysql.connector #maybe unnecessary\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = create_engine('mysql+mysqlconnector://Simon:Bane@localhost/microarray_data_v2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#sample meta import\n",
    "sampleMeta = pd.read_sql_table('master', con=engine)\n",
    "gse_training = pd.read_excel('../Training/geo-expression-array-immune-cells-studies.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [],
   "source": [
    "sampleMeta['cell.type']=sampleMeta['cell.type'].apply(lambda x: x.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gse', 'cell.types', 'n.cell.types', 'title'], dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_training.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>platform_id</th>\n",
       "      <th>gse</th>\n",
       "      <th>file</th>\n",
       "      <th>id</th>\n",
       "      <th>col</th>\n",
       "      <th>pattern</th>\n",
       "      <th>hit</th>\n",
       "      <th>cell.type</th>\n",
       "      <th>val</th>\n",
       "      <th>technology</th>\n",
       "      <th>title</th>\n",
       "      <th>normalization</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10021</td>\n",
       "      <td>GSE10021_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253209</td>\n",
       "      <td>characteristics_ch1_1</td>\n",
       "      <td>\\bT.?cell(?!.*lymphoma)(?!.*leukemia)</td>\n",
       "      <td>from T cell luke</td>\n",
       "      <td>t_cells</td>\n",
       "      <td>human cultured cell line derived from T cell l...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253634</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253635</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253636</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE10038</td>\n",
       "      <td>GSE10038_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM253637</td>\n",
       "      <td>source_name_ch1</td>\n",
       "      <td>macrophage(?!.*progenitor)</td>\n",
       "      <td>olar macrophages obt</td>\n",
       "      <td>macrophage</td>\n",
       "      <td>alveolar macrophages obtained by bronchoscopy ...</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>MAS5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  platform_id       gse                                    file         id  \\\n",
       "0      GPL570  GSE10021  GSE10021_series_matrix.txt.gz.anno.tsv  GSM253209   \n",
       "1      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253634   \n",
       "2      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253635   \n",
       "3      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253636   \n",
       "4      GPL570  GSE10038  GSE10038_series_matrix.txt.gz.anno.tsv  GSM253637   \n",
       "\n",
       "                     col                                pattern  \\\n",
       "0  characteristics_ch1_1  \\bT.?cell(?!.*lymphoma)(?!.*leukemia)   \n",
       "1        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "2        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "3        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "4        source_name_ch1             macrophage(?!.*progenitor)   \n",
       "\n",
       "                    hit   cell.type  \\\n",
       "0      from T cell luke     t_cells   \n",
       "1  olar macrophages obt  macrophage   \n",
       "2  olar macrophages obt  macrophage   \n",
       "3  olar macrophages obt  macrophage   \n",
       "4  olar macrophages obt  macrophage   \n",
       "\n",
       "                                                 val               technology  \\\n",
       "0  human cultured cell line derived from T cell l...  in situ oligonucleotide   \n",
       "1  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "2  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "3  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "4  alveolar macrophages obtained by bronchoscopy ...  in situ oligonucleotide   \n",
       "\n",
       "                                               title normalization  \n",
       "0  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "1  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "2  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "3  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  \n",
       "4  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...          MAS5  "
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleMeta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['GPL570', 'GPL571', 'GPL6244', 'GPL96', 'GPL97'], dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sampleMeta['platform_id'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gse10021',\n",
       " 'gse10038',\n",
       " 'gse49954',\n",
       " 'gse10046',\n",
       " 'gse100786',\n",
       " 'gse10147',\n",
       " 'gse101587',\n",
       " 'gse10178',\n",
       " 'gse101855',\n",
       " 'gse13576',\n",
       " 'gse102459',\n",
       " 'gse51382',\n",
       " 'gse10315',\n",
       " 'gse10316',\n",
       " 'gse35340',\n",
       " 'gse103381',\n",
       " 'gse65856',\n",
       " 'gse13670',\n",
       " 'gse10358',\n",
       " 'gse10437',\n",
       " 'gse36767',\n",
       " 'gse10463',\n",
       " 'gse10479',\n",
       " 'gse10524',\n",
       " 'gse105437',\n",
       " 'gse10586',\n",
       " 'gse10595',\n",
       " 'gse106151',\n",
       " 'gse22045',\n",
       " 'gse106382',\n",
       " 'gse22167',\n",
       " 'gse106532',\n",
       " 'gse10767',\n",
       " 'gse22373',\n",
       " 'gse107683',\n",
       " 'gse107818',\n",
       " 'gse10821',\n",
       " 'gse10831',\n",
       " 'gse108384',\n",
       " 'gse10846',\n",
       " 'gse10880',\n",
       " 'gse10896',\n",
       " 'gse109178',\n",
       " 'gse10927',\n",
       " 'gse109348',\n",
       " 'gse109662',\n",
       " 'gse109836',\n",
       " 'gse110023',\n",
       " 'gse110064',\n",
       " 'gse110085',\n",
       " 'gse110376',\n",
       " 'gse110545',\n",
       " 'gse11057',\n",
       " 'gse11083',\n",
       " 'gse11118',\n",
       " 'gse11151',\n",
       " 'gse11190',\n",
       " 'gse112209',\n",
       " 'gse112210',\n",
       " 'gse112809',\n",
       " 'gse11281',\n",
       " 'gse11292',\n",
       " 'gse112953',\n",
       " 'gse11327',\n",
       " 'gse113645',\n",
       " 'gse11373',\n",
       " 'gse11418',\n",
       " 'gse114181',\n",
       " 'gse11430',\n",
       " 'gse23371',\n",
       " 'gse116487',\n",
       " 'gse116660',\n",
       " 'gse116912',\n",
       " 'gse117139',\n",
       " 'gse11755',\n",
       " 'gse11761',\n",
       " 'gse117676',\n",
       " 'gse11854',\n",
       " 'gse11864',\n",
       " 'gse11881',\n",
       " 'gse11886',\n",
       " 'gse118958',\n",
       " 'gse11919',\n",
       " 'gse24223',\n",
       " 'gse119416',\n",
       " 'gse12079',\n",
       " 'gse12108',\n",
       " 'gse12131',\n",
       " 'gse12187',\n",
       " 'gse12195',\n",
       " 'gse12198',\n",
       " 'gse12293',\n",
       " 'gse12366',\n",
       " 'gse12488',\n",
       " 'gse12507',\n",
       " 'gse12583',\n",
       " 'gse72642',\n",
       " 'gse25087',\n",
       " 'gse12662',\n",
       " 'gse12679',\n",
       " 'gse12734',\n",
       " 'gse12773',\n",
       " 'gse12806',\n",
       " 'gse12838',\n",
       " 'gse16385',\n",
       " 'gse12846',\n",
       " 'gse12875',\n",
       " 'gse12891',\n",
       " 'gse25638',\n",
       " 'gse12963',\n",
       " 'gse13054',\n",
       " 'gse13139',\n",
       " 'gse13210',\n",
       " 'gse13234',\n",
       " 'gse13296',\n",
       " 'gse13300',\n",
       " 'gse13330',\n",
       " 'gse26347',\n",
       " 'gse13501',\n",
       " 'gse26928',\n",
       " 'gse13606',\n",
       " 'gse13637',\n",
       " 'gse13712',\n",
       " 'gse13727',\n",
       " 'gse13732',\n",
       " 'gse13736',\n",
       " 'gse13738',\n",
       " 'gse13762',\n",
       " 'gse13791',\n",
       " 'gse13828',\n",
       " 'gse13837',\n",
       " 'gse13849',\n",
       " 'gse13887',\n",
       " 'gse13896',\n",
       " 'gse13899',\n",
       " 'gse13906',\n",
       " 'gse13909',\n",
       " 'gse13917',\n",
       " 'gse13931',\n",
       " 'gse13987',\n",
       " 'gse14000',\n",
       " 'gse14062',\n",
       " 'gse27928',\n",
       " 'gse28490',\n",
       " 'gse28491',\n",
       " 'gse14596',\n",
       " 'gse14924',\n",
       " 'gse14926',\n",
       " 'gse19246',\n",
       " 'gse15659',\n",
       " 'gse42203',\n",
       " 'gse56767',\n",
       " 'gse42253',\n",
       " 'gse16130',\n",
       " 'gse16461',\n",
       " 'gse65135',\n",
       " 'gse17186',\n",
       " 'gse17269',\n",
       " 'gse17549',\n",
       " 'gse19067',\n",
       " 'gse19069',\n",
       " 'gse31773',\n",
       " 'gse95298',\n",
       " 'gse47855',\n",
       " 'gse47920',\n",
       " 'gse9801',\n",
       " 'gse19599',\n",
       " 'gse49910',\n",
       " 'gse64555',\n",
       " 'gse51288',\n",
       " 'gse51997',\n",
       " 'gse67321',\n",
       " 'gse4588',\n",
       " 'gse67596',\n",
       " 'gse36907',\n",
       " 'gse56314',\n",
       " 'gse55673',\n",
       " 'gse33536',\n",
       " 'gse7307',\n",
       " 'gse26050',\n",
       " 'gse26156',\n",
       " 'gse26495',\n",
       " 'gse44126',\n",
       " 'gse59412',\n",
       " 'gse94510',\n",
       " 'gse28726',\n",
       " 'gse56313',\n",
       " 'gse41831',\n",
       " 'gse41889',\n",
       " 'gse29301',\n",
       " 'gse93984',\n",
       " 'gse60920',\n",
       " 'gse84331',\n",
       " 'gse85074',\n",
       " 'gse93683',\n",
       " 'gse42733',\n",
       " 'gse45535',\n",
       " 'gse76492',\n",
       " 'gse4527',\n",
       " 'gse59786',\n",
       " 'gse94801',\n",
       " 'gse61697',\n",
       " 'gse6743',\n",
       " 'gse83441',\n",
       " 'gse93776',\n",
       " 'gse74266',\n",
       " 'gse65010',\n",
       " 'gse47796',\n",
       " 'gse42672',\n",
       " 'gse71566',\n",
       " 'gse64028',\n",
       " 'gse98640',\n",
       " 'gse9891',\n",
       " 'gse50175',\n",
       " 'gse82159',\n",
       " 'gse58173',\n",
       " 'gse43769',\n",
       " 'gse79706',\n",
       " 'gse7753',\n",
       " 'gse59051',\n",
       " 'gse43346',\n",
       " 'gse73927',\n",
       " 'gse71730']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(map(lambda x: x.lower(), sampleMeta[sampleMeta['platform_id']=='GPL570']['gse'].unique()))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " ### GENERAL ATTRIBUTES ###\n",
    " \n",
    " __GPL570__\n",
    " - lots of GSEs\n",
    " - 22899 genes\n",
    " \n",
    " __GPL571__\n",
    " - fewer GSEs\n",
    " - 13277 genes\n",
    " \n",
    " __GPL6244__\n",
    " - 1 GSE\n",
    " \n",
    " __GPL96__\n",
    " - decent amount of GSEs\n",
    " - same gene # as GPL571, 13277\n",
    "     -- though some are in the 22000 range\n",
    " \n",
    " __GPL97__\n",
    " - handful of GSEs\n",
    " - also 13277\n",
    " \n",
    " ### COMBINATIONS ###\n",
    " \n",
    " - GPL97 and GPL96 *seem* to combine well\n",
    " - GPL96 and GPL571 *seem* to combine well\n",
    " - it appears that most--if not all--of the '13000' gene symbol sets are included in the 22000 gene symbol sets\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['gse10021',\n",
       " 'gse10038',\n",
       " 'gse10040',\n",
       " 'gse10046',\n",
       " 'gse100786',\n",
       " 'gse10125',\n",
       " 'gse10147',\n",
       " 'gse101587',\n",
       " 'gse10172',\n",
       " 'gse10178',\n",
       " 'gse101855',\n",
       " 'gse102459',\n",
       " 'gse10315',\n",
       " 'gse10316',\n",
       " 'gse10325',\n",
       " 'gse103381',\n",
       " 'gse10358',\n",
       " 'gse10437',\n",
       " 'gse10463',\n",
       " 'gse10479',\n",
       " 'gse10524',\n",
       " 'gse105437',\n",
       " 'gse10586',\n",
       " 'gse10595',\n",
       " 'gse10612',\n",
       " 'gse106151',\n",
       " 'gse10631',\n",
       " 'gse106382',\n",
       " 'gse106532',\n",
       " 'gse10767',\n",
       " 'gse107683',\n",
       " 'gse10778',\n",
       " 'gse107818',\n",
       " 'gse10804',\n",
       " 'gse10821',\n",
       " 'gse10831',\n",
       " 'gse108384',\n",
       " 'gse10846',\n",
       " 'gse10880',\n",
       " 'gse10896',\n",
       " 'gse109178',\n",
       " 'gse10927',\n",
       " 'gse109348',\n",
       " 'gse10957',\n",
       " 'gse109662',\n",
       " 'gse109836',\n",
       " 'gse110023',\n",
       " 'gse110064',\n",
       " 'gse110085',\n",
       " 'gse110376',\n",
       " 'gse110545',\n",
       " 'gse11057',\n",
       " 'gse11083',\n",
       " 'gse11118',\n",
       " 'gse11151',\n",
       " 'gse11188',\n",
       " 'gse11190',\n",
       " 'gse112209',\n",
       " 'gse112210',\n",
       " 'gse112809',\n",
       " 'gse11281',\n",
       " 'gse11292',\n",
       " 'gse112953',\n",
       " 'gse11327',\n",
       " 'gse1133',\n",
       " 'gse11341',\n",
       " 'gse11342',\n",
       " 'gse113645',\n",
       " 'gse11373',\n",
       " 'gse11418',\n",
       " 'gse114181',\n",
       " 'gse11430',\n",
       " 'gse11450',\n",
       " 'gse116487',\n",
       " 'gse116660',\n",
       " 'gse116912',\n",
       " 'gse117139',\n",
       " 'gse11755',\n",
       " 'gse11761',\n",
       " 'gse11767',\n",
       " 'gse117676',\n",
       " 'gse11854',\n",
       " 'gse11864',\n",
       " 'gse11881',\n",
       " 'gse11886',\n",
       " 'gse118958',\n",
       " 'gse11909',\n",
       " 'gse11919',\n",
       " 'gse119416',\n",
       " 'gse11943',\n",
       " 'gse12079',\n",
       " 'gse12108',\n",
       " 'gse12131',\n",
       " 'gse12187',\n",
       " 'gse12195',\n",
       " 'gse12198',\n",
       " 'gse12246',\n",
       " 'gse12293',\n",
       " 'gse12366',\n",
       " 'gse12484',\n",
       " 'gse12488',\n",
       " 'gse12507',\n",
       " 'gse12583',\n",
       " 'gse12585',\n",
       " 'gse12586',\n",
       " 'gse12662',\n",
       " 'gse12668',\n",
       " 'gse12679',\n",
       " 'gse12734',\n",
       " 'gse12773',\n",
       " 'gse12803',\n",
       " 'gse12806',\n",
       " 'gse12838',\n",
       " 'gse12839',\n",
       " 'gse12845',\n",
       " 'gse12846',\n",
       " 'gse12875',\n",
       " 'gse12891',\n",
       " 'gse12907',\n",
       " 'gse12963',\n",
       " 'gse13017',\n",
       " 'gse13054',\n",
       " 'gse13139',\n",
       " 'gse13210',\n",
       " 'gse13234',\n",
       " 'gse13296',\n",
       " 'gse13300',\n",
       " 'gse13330',\n",
       " 'gse13411',\n",
       " 'gse13496',\n",
       " 'gse13501',\n",
       " 'gse13576',\n",
       " 'gse13591',\n",
       " 'gse13606',\n",
       " 'gse13637',\n",
       " 'gse13670',\n",
       " 'gse13712',\n",
       " 'gse13727',\n",
       " 'gse13732',\n",
       " 'gse13736',\n",
       " 'gse13738',\n",
       " 'gse13762',\n",
       " 'gse13791',\n",
       " 'gse13828',\n",
       " 'gse13837',\n",
       " 'gse13849',\n",
       " 'gse13850',\n",
       " 'gse13887',\n",
       " 'gse13896',\n",
       " 'gse13899',\n",
       " 'gse13906',\n",
       " 'gse13909',\n",
       " 'gse13917',\n",
       " 'gse13931',\n",
       " 'gse13987',\n",
       " 'gse13996',\n",
       " 'gse14000',\n",
       " 'gse14062',\n",
       " 'gse14230',\n",
       " 'gse14256',\n",
       " 'gse14330',\n",
       " 'gse14334',\n",
       " 'gse14335',\n",
       " 'gse14340',\n",
       " 'gse14386',\n",
       " 'gse14390',\n",
       " 'gse14405',\n",
       " 'gse14419',\n",
       " 'gse14465',\n",
       " 'gse14479',\n",
       " 'gse14577',\n",
       " 'gse14596',\n",
       " 'gse14630',\n",
       " 'gse14642',\n",
       " 'gse14711',\n",
       " 'gse14801',\n",
       " 'gse14816',\n",
       " 'gse14827',\n",
       " 'gse14895',\n",
       " 'gse14897',\n",
       " 'gse14924',\n",
       " 'gse14926',\n",
       " 'gse15072',\n",
       " 'gse15174',\n",
       " 'gse15215',\n",
       " 'gse15245',\n",
       " 'gse15322',\n",
       " 'gse15389',\n",
       " 'gse15400',\n",
       " 'gse15464',\n",
       " 'gse15490',\n",
       " 'gse15495',\n",
       " 'gse15499',\n",
       " 'gse15537',\n",
       " 'gse15615',\n",
       " 'gse15645',\n",
       " 'gse15659',\n",
       " 'gse15743',\n",
       " 'gse15777',\n",
       " 'gse15928',\n",
       " 'gse15949',\n",
       " 'gse16032',\n",
       " 'gse16051',\n",
       " 'gse16093',\n",
       " 'gse16098',\n",
       " 'gse16111',\n",
       " 'gse16129',\n",
       " 'gse16130',\n",
       " 'gse16214',\n",
       " 'gse16226',\n",
       " 'gse16250',\n",
       " 'gse16354',\n",
       " 'gse16355',\n",
       " 'gse16356',\n",
       " 'gse16382',\n",
       " 'gse16385',\n",
       " 'gse16386',\n",
       " 'gse16416',\n",
       " 'gse1644',\n",
       " 'gse16455',\n",
       " 'gse16461',\n",
       " 'gse16524',\n",
       " 'gse16547',\n",
       " 'gse16654',\n",
       " 'gse16683',\n",
       " 'gse16712',\n",
       " 'gse16715',\n",
       " 'gse16728',\n",
       " 'gse16746',\n",
       " 'gse16755',\n",
       " 'gse16791',\n",
       " 'gse16836',\n",
       " 'gse16837',\n",
       " 'gse16879',\n",
       " 'gse16962',\n",
       " 'gse16972',\n",
       " 'gse17046',\n",
       " 'gse17061',\n",
       " 'gse17078',\n",
       " 'gse17114',\n",
       " 'gse17119',\n",
       " 'gse17186',\n",
       " 'gse17189',\n",
       " 'gse17269',\n",
       " 'gse17299',\n",
       " 'gse17301',\n",
       " 'gse17306',\n",
       " 'gse17354',\n",
       " 'gse17356',\n",
       " 'gse17372',\n",
       " 'gse17407',\n",
       " 'gse17459',\n",
       " 'gse17477',\n",
       " 'gse17549',\n",
       " 'gse17718',\n",
       " 'gse17777',\n",
       " 'gse17938',\n",
       " 'gse17939',\n",
       " 'gse17974',\n",
       " 'gse18026',\n",
       " 'gse18090',\n",
       " 'gse18113',\n",
       " 'gse18147',\n",
       " 'gse18198',\n",
       " 'gse18226',\n",
       " 'gse18291',\n",
       " 'gse18309',\n",
       " 'gse18313',\n",
       " 'gse18323',\n",
       " 'gse18373',\n",
       " 'gse18497',\n",
       " 'gse18565',\n",
       " 'gse18593',\n",
       " 'gse18723',\n",
       " 'gse18791',\n",
       " 'gse18810',\n",
       " 'gse18864',\n",
       " 'gse18877',\n",
       " 'gse18913',\n",
       " 'gse19067',\n",
       " 'gse19069',\n",
       " 'gse19098',\n",
       " 'gse19128',\n",
       " 'gse19143',\n",
       " 'gse19147',\n",
       " 'gse19240',\n",
       " 'gse19243',\n",
       " 'gse19246',\n",
       " 'gse19301',\n",
       " 'gse19315',\n",
       " 'gse19475',\n",
       " 'gse19510',\n",
       " 'gse19556',\n",
       " 'gse19599',\n",
       " 'gse19615',\n",
       " 'gse19625',\n",
       " 'gse19627',\n",
       " 'gse19735',\n",
       " 'gse19784',\n",
       " 'gse19816',\n",
       " 'gse19882',\n",
       " 'gse19888',\n",
       " 'gse20086',\n",
       " 'gse20124',\n",
       " 'gse20125',\n",
       " 'gse20151',\n",
       " 'gse20198',\n",
       " 'gse20307',\n",
       " 'gse20484',\n",
       " 'gse20499',\n",
       " 'gse20504',\n",
       " 'gse20508',\n",
       " 'gse20538',\n",
       " 'gse20641',\n",
       " 'gse20666',\n",
       " 'gse20685',\n",
       " 'gse20745',\n",
       " 'gse20847',\n",
       " 'gse20978',\n",
       " 'gse20986',\n",
       " 'gse21039',\n",
       " 'gse21050',\n",
       " 'gse21132',\n",
       " 'gse21243',\n",
       " 'gse2125',\n",
       " 'gse21251',\n",
       " 'gse21252',\n",
       " 'gse21351',\n",
       " 'gse21521',\n",
       " 'gse21545',\n",
       " 'gse21547',\n",
       " 'gse21648',\n",
       " 'gse21723',\n",
       " 'gse21774',\n",
       " 'gse21837',\n",
       " 'gse21891',\n",
       " 'gse21899',\n",
       " 'gse21943',\n",
       " 'gse21989',\n",
       " 'gse22025',\n",
       " 'gse22045',\n",
       " 'gse22103',\n",
       " 'gse22166',\n",
       " 'gse22167',\n",
       " 'gse22183',\n",
       " 'gse22224',\n",
       " 'gse22246',\n",
       " 'gse22250',\n",
       " 'gse22255',\n",
       " 'gse22282',\n",
       " 'gse22325',\n",
       " 'gse22373',\n",
       " 'gse22392',\n",
       " 'gse22469',\n",
       " 'gse22470',\n",
       " 'gse22501',\n",
       " 'gse22522',\n",
       " 'gse22528',\n",
       " 'gse22529',\n",
       " 'gse22552',\n",
       " 'gse22589',\n",
       " 'gse22632',\n",
       " 'gse22667',\n",
       " 'gse22668',\n",
       " 'gse22688',\n",
       " 'gse22695',\n",
       " 'gse22759',\n",
       " 'gse22886',\n",
       " 'gse23371',\n",
       " 'gse24223',\n",
       " 'gse24634',\n",
       " 'gse25087',\n",
       " 'gse25638',\n",
       " 'gse26050',\n",
       " 'gse26156',\n",
       " 'gse26294',\n",
       " 'gse26347',\n",
       " 'gse26495',\n",
       " 'gse26928',\n",
       " 'gse27928',\n",
       " 'gse28079',\n",
       " 'gse28490',\n",
       " 'gse28491',\n",
       " 'gse28726',\n",
       " 'gse29301',\n",
       " 'gse31773',\n",
       " 'gse33536',\n",
       " 'gse35340',\n",
       " 'gse36767',\n",
       " 'gse36907',\n",
       " 'gse3920',\n",
       " 'gse39262',\n",
       " 'gse3982',\n",
       " 'gse41831',\n",
       " 'gse41889',\n",
       " 'gse42203',\n",
       " 'gse42253',\n",
       " 'gse4251',\n",
       " 'gse42672',\n",
       " 'gse42733',\n",
       " 'gse43346',\n",
       " 'gse43677',\n",
       " 'gse43769',\n",
       " 'gse44126',\n",
       " 'gse44164',\n",
       " 'gse4475',\n",
       " 'gse4527',\n",
       " 'gse45535',\n",
       " 'gse4571',\n",
       " 'gse4588',\n",
       " 'gse46913',\n",
       " 'gse46917',\n",
       " 'gse47796',\n",
       " 'gse47855',\n",
       " 'gse47920',\n",
       " 'gse49910',\n",
       " 'gse49954',\n",
       " 'gse50175',\n",
       " 'gse5099',\n",
       " 'gse51288',\n",
       " 'gse51382',\n",
       " 'gse51997',\n",
       " 'gse54017',\n",
       " 'gse55673',\n",
       " 'gse5580',\n",
       " 'gse56313',\n",
       " 'gse56314',\n",
       " 'gse56767',\n",
       " 'gse57611',\n",
       " 'gse58173',\n",
       " 'gse59051',\n",
       " 'gse59412',\n",
       " 'gse59786',\n",
       " 'gse6053',\n",
       " 'gse60920',\n",
       " 'gse61697',\n",
       " 'gse62584',\n",
       " 'gse64028',\n",
       " 'gse64555',\n",
       " 'gse65010',\n",
       " 'gse65135',\n",
       " 'gse65856',\n",
       " 'gse67321',\n",
       " 'gse6740',\n",
       " 'gse6743',\n",
       " 'gse67596',\n",
       " 'gse7138',\n",
       " 'gse71566',\n",
       " 'gse71730',\n",
       " 'gse72642',\n",
       " 'gse7307',\n",
       " 'gse73927',\n",
       " 'gse74266',\n",
       " 'gse76492',\n",
       " 'gse7753',\n",
       " 'gse79706',\n",
       " 'gse82159',\n",
       " 'gse8286',\n",
       " 'gse83441',\n",
       " 'gse84331',\n",
       " 'gse85074',\n",
       " 'gse8835',\n",
       " 'gse8921',\n",
       " 'gse93683',\n",
       " 'gse93776',\n",
       " 'gse9378',\n",
       " 'gse93984',\n",
       " 'gse94510',\n",
       " 'gse94801',\n",
       " 'gse95298',\n",
       " 'gse9801',\n",
       " 'gse98640',\n",
       " 'gse9891',\n",
       " 'gse9946',\n",
       " 'master',\n",
       " 'master_test']"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "engine.table_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22899"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# len(gse_df['gene_symbol_sql'])\n",
    "gse_df = pd.read_sql_table('gse10021', con=engine)\n",
    "geneSymbolList = set(gse_df['gene_symbol_sql'])\n",
    "len(geneSymbolList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gse10021\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10038\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10040\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10046\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse100786\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse101587\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10172\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10178\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse101855\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse102459\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10315\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10316\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10325\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse103381\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10358\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10437\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10463\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10479\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10524\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse105437\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10586\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10595\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10612\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10631\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse106532\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse107683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10778\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse107818\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10804\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10821\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10831\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse108384\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10846\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10880\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10896\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109178\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10927\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109348\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse10957\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109662\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse109836\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110023\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110064\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110085\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110376\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse110545\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11057\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11083\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11118\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11188\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11190\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112209\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112210\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112809\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11281\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11292\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse112953\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11327\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse1133\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11341\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11342\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse113645\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11418\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse114181\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11430\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11450\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116487\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116660\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse116912\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse117139\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11755\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11761\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse117676\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11854\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11864\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11881\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11886\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse118958\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11909\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11919\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse119416\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse11943\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12079\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12108\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12131\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12187\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12195\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12293\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12366\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12484\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12488\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12507\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12583\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12585\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12586\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12662\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12668\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12679\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12734\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12773\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12803\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12806\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12838\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12839\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12845\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12846\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12875\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12907\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse12963\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13017\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13054\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13139\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13210\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13234\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13296\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13300\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13330\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13411\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13496\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13501\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13576\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13591\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13606\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13637\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13670\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13712\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13727\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13732\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13736\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13738\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13762\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13828\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13849\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13850\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13887\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13896\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13899\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13906\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13909\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13917\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13931\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13987\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse13996\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14000\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14062\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14230\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14256\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14330\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14334\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14335\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14340\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14386\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14390\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14405\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14419\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14465\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14479\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14577\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14596\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14630\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14642\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14711\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14816\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14827\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14895\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14897\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14924\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse14926\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15072\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15174\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15215\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15245\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15322\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15389\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15400\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15464\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15490\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15495\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15499\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15537\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15615\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15645\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15659\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15743\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15777\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse15949\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16032\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16051\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16093\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16098\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16111\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16129\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16130\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16214\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16226\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16250\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16354\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16355\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16356\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16385\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16386\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16416\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse1644\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16455\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16461\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16524\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16547\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16654\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16712\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16715\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16728\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16746\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16755\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16836\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16879\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16962\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse16972\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17046\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17061\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17078\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17114\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17119\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17186\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17189\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17269\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17299\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17301\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17306\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17354\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17356\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17372\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17407\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17459\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17477\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17549\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17718\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17777\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17938\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17939\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "22899\n",
      "gse17974\n",
      "not in genesymbollist:\n",
      "3707\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18026\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18090\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18113\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18226\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18291\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18309\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18313\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18323\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18497\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18565\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18593\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18723\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18791\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18810\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18864\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18877\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse18913\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19067\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19069\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19098\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19128\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19143\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19147\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19240\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19243\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19301\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19315\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19475\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19510\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19556\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19599\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19615\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19625\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19627\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19735\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19784\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19816\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19882\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse19888\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20086\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20124\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20151\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20198\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20307\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20484\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20499\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20504\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20508\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20538\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20641\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20666\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20685\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20745\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20847\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20978\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse20986\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21039\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21050\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21132\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21243\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse2125\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21251\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21252\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21351\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21521\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21545\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21547\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21648\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21723\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21774\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21837\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21899\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21943\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse21989\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22025\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22045\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22103\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22166\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22167\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22183\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22224\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22246\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22250\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22255\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22282\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22325\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22373\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22392\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22469\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22470\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22501\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22522\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22528\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22529\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22552\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22589\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22632\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22667\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22668\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22688\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22695\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22759\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse22886\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse23371\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse24223\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse24634\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse25087\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse25638\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26050\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26156\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26294\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26347\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26495\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse26928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse27928\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28079\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28490\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28491\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse28726\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse29301\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse31773\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse33536\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse35340\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse36767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse36907\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse3920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse39262\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse3982\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse41831\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse41889\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42203\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42253\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4251\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42672\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse42733\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43346\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43677\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse43769\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse44126\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse44164\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4475\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4527\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse45535\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4571\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse4588\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse46913\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse46917\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47796\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47855\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse47920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse49910\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse49954\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse50175\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse5099\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51288\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51382\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse51997\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse54017\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse55673\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse5580\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56313\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56314\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse56767\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse57611\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse58173\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59051\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59412\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse59786\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6053\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse60920\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse61697\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse62584\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse64028\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse64555\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65010\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65135\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse65856\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse67321\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6740\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse6743\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse67596\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7138\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse71566\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse71730\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse72642\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7307\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse73927\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse74266\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse76492\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse7753\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse79706\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse82159\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8286\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse83441\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse84331\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse85074\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8835\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse8921\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93683\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93776\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9378\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse93984\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse94510\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse94801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse95298\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9801\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse98640\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9891\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n",
      "gse9946\n",
      "not in genesymbollist:\n",
      "0\n",
      "new geneSymbolList length:\n",
      "26606\n"
     ]
    }
   ],
   "source": [
    "gselist = [x for x in engine.table_names() if 'gse' in x]\n",
    "# geneSymbolList was initialized above\n",
    "for gse in gselist:\n",
    "    print(gse)\n",
    "    gse_df = pd.read_sql_table(gse, con=engine)\n",
    "    gse_gs = set(gse_df['gene_symbol_sql'])\n",
    "    # how many gene symbols are not in geneSymbolList?\n",
    "    new_geneSymbols = gse_gs.difference(geneSymbolList)\n",
    "    print(\"not in genesymbollist:\")\n",
    "    print(len(new_geneSymbols))\n",
    "    # how much bigger does geneSymbolList get after union?\n",
    "    geneSymbolList = geneSymbolList.union(gse_gs)\n",
    "    print(\"new geneSymbolList length:\")\n",
    "    print(len(geneSymbolList))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/allGeneSymbols.pckl', 'wb') as f:\n",
    "    pickle.dump(geneSymbolList, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. LOAD THE GENE SYMBOL PICKLE FILE\n",
    "    - sampleMeta\n",
    "        - dammit, cell type has weird capitalization\n",
    "    - gse_training\n",
    "\n",
    "2. MAKE A CELL TYPE DF\n",
    "    - try with 'neutrophils'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/allGeneSymbols.pckl', 'rb') as f:\n",
    "    geneSymbolList = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('picklejar/ctDict.pckl', 'rb') as f:\n",
    "    ctDict = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'CD4.T.cells': ['memory_cd4_t_cells',\n",
       "  'th17_cells',\n",
       "  'naive_cd4_t_cells',\n",
       "  'th1_cells',\n",
       "  'th2_cells',\n",
       "  'cd4_t_cells',\n",
       "  'tregs',\n",
       "  'central_memory_cd4_t_cells',\n",
       "  'effector_memory_cd4_t_cells',\n",
       "  'follicular_helper_t_cells',\n",
       "  'th1th17_cells',\n",
       "  'effector_cd4_t_cells'],\n",
       " 'CD8.T.cells': ['cd8_t_cells',\n",
       "  'naive_cd8_t_cells',\n",
       "  'effector_memory_cd8_t_cells',\n",
       "  'memory_cd8_t_cells',\n",
       "  'central_memory_cd8_t_cells'],\n",
       " 'NK.cells': ['nk_cells'],\n",
       " 'B.cells': ['b_cells',\n",
       "  'naive_b_cells',\n",
       "  'gc_b_cells',\n",
       "  'memory_b_cells',\n",
       "  'immature_b_cells',\n",
       "  'activated_b_cells',\n",
       "  'plasma_b_cells',\n",
       "  'resting_b_cells',\n",
       "  'pre-gc_b_cells'],\n",
       " 'monocytic.lineage': ['monocytes',\n",
       "  'dendritic_cells',\n",
       "  'macrophage',\n",
       "  'immature_dendritic_cells',\n",
       "  'myeloid_dendritic_cells',\n",
       "  'activated_macrophages'],\n",
       " 'neutrophils': ['neutrophils', 'granulocytes'],\n",
       " 'endothelial.cells': ['endothelial'],\n",
       " 'fibroblasts': ['fibroblast']}"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ctDict['Coarse']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>platform_id</th>\n",
       "      <th>gse</th>\n",
       "      <th>file</th>\n",
       "      <th>id</th>\n",
       "      <th>col</th>\n",
       "      <th>pattern</th>\n",
       "      <th>hit</th>\n",
       "      <th>cell.type</th>\n",
       "      <th>val</th>\n",
       "      <th>technology</th>\n",
       "      <th>title</th>\n",
       "      <th>normalization</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3529</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025152</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3530</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025153</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3531</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025154</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3532</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025155</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3533</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025156</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3534</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025157</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3535</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025158</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3536</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025159</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3537</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025160</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3538</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025161</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025162</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3540</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025163</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3791</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE45535</td>\n",
       "      <td>GSE45535_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1108140</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3792</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE45535</td>\n",
       "      <td>GSE45535_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1108146</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4437</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025135</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4438</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025142</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4439</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025151</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4488</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025134</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4489</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025136</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4490</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025137</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4563</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025138</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4564</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025139</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4565</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025146</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4611</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025140</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4612</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025143</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4613</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025144</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4614</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025145</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4615</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025147</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4616</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025148</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4667</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025129</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4668</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025130</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4669</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025133</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4670</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025141</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4748</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025131</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4749</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025132</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4753</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025150</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4833</th>\n",
       "      <td>GPL570</td>\n",
       "      <td>GSE41831</td>\n",
       "      <td>GSE41831_series_matrix.txt.gz.anno.tsv</td>\n",
       "      <td>GSM1025149</td>\n",
       "      <td>characteristics_ch1_cell.type</td>\n",
       "      <td>granulocyte(?!.*progenitor)</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>Granulocytes</td>\n",
       "      <td>granulocytes</td>\n",
       "      <td>in situ oligonucleotide</td>\n",
       "      <td>[HG-U133_Plus_2] Affymetrix Human Genome U133 ...</td>\n",
       "      <td>RMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     platform_id       gse                                    file  \\\n",
       "3529      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3530      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3531      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3532      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3533      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3534      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3535      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3536      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3537      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3538      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3539      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3540      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "3791      GPL570  GSE45535  GSE45535_series_matrix.txt.gz.anno.tsv   \n",
       "3792      GPL570  GSE45535  GSE45535_series_matrix.txt.gz.anno.tsv   \n",
       "4437      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4438      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4439      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4488      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4489      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4490      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4563      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4564      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4565      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4611      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4612      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4613      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4614      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4615      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4616      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4667      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4668      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4669      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4670      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4748      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4749      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4753      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "4833      GPL570  GSE41831  GSE41831_series_matrix.txt.gz.anno.tsv   \n",
       "\n",
       "              id                            col                      pattern  \\\n",
       "3529  GSM1025152  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3530  GSM1025153  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3531  GSM1025154  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3532  GSM1025155  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3533  GSM1025156  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3534  GSM1025157  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3535  GSM1025158  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3536  GSM1025159  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3537  GSM1025160  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3538  GSM1025161  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3539  GSM1025162  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3540  GSM1025163  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3791  GSM1108140  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "3792  GSM1108146  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4437  GSM1025135  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4438  GSM1025142  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4439  GSM1025151  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4488  GSM1025134  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4489  GSM1025136  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4490  GSM1025137  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4563  GSM1025138  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4564  GSM1025139  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4565  GSM1025146  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4611  GSM1025140  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4612  GSM1025143  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4613  GSM1025144  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4614  GSM1025145  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4615  GSM1025147  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4616  GSM1025148  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4667  GSM1025129  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4668  GSM1025130  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4669  GSM1025133  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4670  GSM1025141  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4748  GSM1025131  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4749  GSM1025132  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4753  GSM1025150  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "4833  GSM1025149  characteristics_ch1_cell.type  granulocyte(?!.*progenitor)   \n",
       "\n",
       "               hit     cell.type           val               technology  \\\n",
       "3529  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3530  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3531  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3532  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3533  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3534  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3535  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3536  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3537  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3538  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3539  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3540  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "3791  Granulocytes  Granulocytes  Granulocytes  in situ oligonucleotide   \n",
       "3792  Granulocytes  Granulocytes  Granulocytes  in situ oligonucleotide   \n",
       "4437  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4438  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4439  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4488  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4489  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4490  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4563  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4564  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4565  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4611  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4612  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4613  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4614  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4615  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4616  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4667  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4668  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4669  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4670  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4748  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4749  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4753  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "4833  granulocytes  Granulocytes  granulocytes  in situ oligonucleotide   \n",
       "\n",
       "                                                  title normalization  \n",
       "3529  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3530  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3531  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3532  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3533  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3534  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3535  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3536  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3537  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3538  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3539  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3540  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "3791  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...       unknown  \n",
       "3792  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...       unknown  \n",
       "4437  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4438  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4439  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4488  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4489  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4490  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4563  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4564  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4565  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4611  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4612  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4613  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4614  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4615  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4616  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4667  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4668  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4669  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4670  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4748  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4749  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4753  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  \n",
       "4833  [HG-U133_Plus_2] Affymetrix Human Genome U133 ...           RMA  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get neutrophil GSEs\n",
    "sampleMeta[sampleMeta['cell.type']=='Granulocytes']\n",
    "# GSE11083\n",
    "# - get relevant GSMs from GSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_df(gse, sub_ct):\n",
    "    print(gse)\n",
    "    df = pd.read_sql_table(gse.lower(), con=engine)\n",
    "    df.set_index('gene_symbol_sql', inplace=True)\n",
    "    gsm_idx = sampleMeta[sampleMeta['cell.type']==sub_ct][sampleMeta['gse']==gse]['id']\n",
    "    #apparently GSMs in sampleMeta are not present in some GSE dfs\n",
    "    gsm_idx = [g for g in gsm_idx if g in df.columns]\n",
    "    df = df[gsm_idx].T\n",
    "    return df\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CD4.T.cells\n",
      "GSE36767\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  \"\"\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GSE26928\n",
      "GSE15659\n",
      "GSE44126\n",
      "GSE60920\n",
      "GSE22886\n",
      "6\n",
      "GSE110545\n",
      "1\n",
      "GSE22045\n",
      "GSE109348\n",
      "GSE25087\n",
      "GSE26928\n",
      "GSE13732\n",
      "GSE94510\n",
      "GSE71566\n",
      "GSE93776\n",
      "GSE13017\n",
      "9\n",
      "GSE50175\n",
      "GSE43769\n",
      "GSE22886\n",
      "GSE3982\n",
      "4\n",
      "GSE109348\n",
      "GSE43769\n",
      "GSE22886\n",
      "GSE3982\n",
      "4\n",
      "GSE10437\n",
      "GSE36767\n",
      "GSE11292\n",
      "GSE117139\n",
      "GSE119416\n",
      "GSE12079\n",
      "GSE12488\n",
      "GSE72642\n",
      "GSE12875\n",
      "GSE12963\n",
      "GSE27928\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE14924\n",
      "GSE14926\n",
      "GSE15659\n",
      "GSE42253\n",
      "GSE16130\n",
      "GSE16461\n",
      "GSE19069\n",
      "GSE31773\n",
      "GSE47920\n",
      "GSE49954\n",
      "GSE51288\n",
      "GSE67321\n",
      "GSE4588\n",
      "GSE7307\n",
      "GSE49910\n",
      "GSE26156\n",
      "GSE28726\n",
      "GSE60920\n",
      "GSE4527\n",
      "GSE59786\n",
      "GSE71566\n",
      "GSE51997\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE73927\n",
      "GSE14330\n",
      "GSE62584\n",
      "GSE22886\n",
      "GSE8835\n",
      "GSE10325\n",
      "GSE1133\n",
      "GSE4571\n",
      "GSE6740\n",
      "GSE17354\n",
      "GSE46917\n",
      "48\n",
      "GSE10586\n",
      "GSE22045\n",
      "GSE11292\n",
      "GSE25087\n",
      "GSE13234\n",
      "GSE15659\n",
      "GSE59786\n",
      "GSE65010\n",
      "GSE73927\n",
      "GSE14330\n",
      "GSE24634\n",
      "GSE13017\n",
      "GSE4571\n",
      "GSE4527\n",
      "14\n",
      "GSE26928\n",
      "GSE94510\n",
      "2\n",
      "GSE26928\n",
      "GSE94510\n",
      "2\n",
      "GSE73927\n",
      "1\n",
      "GSE50175\n",
      "1\n",
      "0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:11: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(890, 22899)\n",
      "(650, 16124)\n",
      "CD8.T.cells\n",
      "GSE49954\n",
      "GSE10178\n",
      "GSE109348\n",
      "GSE72642\n",
      "GSE26347\n",
      "GSE27928\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE14924\n",
      "GSE14926\n",
      "GSE42253\n",
      "GSE16130\n",
      "GSE16461\n",
      "GSE19069\n",
      "GSE31773\n",
      "GSE47920\n",
      "GSE49910\n",
      "GSE51288\n",
      "GSE67321\n",
      "GSE44126\n",
      "GSE84331\n",
      "GSE93776\n",
      "GSE98640\n",
      "GSE85074\n",
      "GSE7307\n",
      "GSE58173\n",
      "GSE62584\n",
      "GSE8835\n",
      "GSE6740\n",
      "GSE11188\n",
      "GSE1133\n",
      "GSE22886\n",
      "GSE17354\n",
      "GSE10631\n",
      "34\n",
      "GSE26347\n",
      "GSE26495\n",
      "GSE84331\n",
      "GSE85074\n",
      "GSE93683\n",
      "GSE98640\n",
      "6\n",
      "GSE85074\n",
      "GSE93683\n",
      "GSE98640\n",
      "3\n",
      "GSE93683\n",
      "1\n",
      "GSE93683\n",
      "GSE85074\n",
      "GSE98640\n",
      "3\n",
      "(423, 22899)\n",
      "(316, 17099)\n",
      "NK.cells\n",
      "GSE10358\n",
      "GSE10927\n",
      "GSE109348\n",
      "GSE11373\n",
      "GSE116487\n",
      "GSE116660\n",
      "GSE12198\n",
      "GSE72642\n",
      "GSE13501\n",
      "GSE13849\n",
      "GSE42253\n",
      "GSE19067\n",
      "GSE26050\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE47855\n",
      "GSE9891\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE7753\n",
      "GSE22886\n",
      "GSE4475\n",
      "GSE10957\n",
      "GSE39262\n",
      "GSE3982\n",
      "GSE57611\n",
      "26\n",
      "(197, 22899)\n",
      "(110, 17096)\n",
      "B.cells\n",
      "GSE51382\n",
      "GSE65856\n",
      "GSE10524\n",
      "GSE106382\n",
      "GSE10767\n",
      "GSE107683\n",
      "GSE10846\n",
      "GSE109348\n",
      "GSE110023\n",
      "GSE110064\n",
      "GSE110376\n",
      "GSE112953\n",
      "GSE24223\n",
      "GSE12195\n",
      "GSE72642\n",
      "GSE12734\n",
      "GSE13210\n",
      "GSE13300\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE42203\n",
      "GSE65135\n",
      "GSE17186\n",
      "GSE17269\n",
      "GSE19599\n",
      "GSE36907\n",
      "GSE56314\n",
      "GSE7307\n",
      "GSE49910\n",
      "GSE25638\n",
      "GSE4588\n",
      "GSE56313\n",
      "GSE29301\n",
      "GSE93984\n",
      "GSE45535\n",
      "GSE59412\n",
      "GSE83441\n",
      "GSE74266\n",
      "GSE64028\n",
      "GSE82159\n",
      "GSE67321\n",
      "GSE58173\n",
      "GSE94801\n",
      "GSE71730\n",
      "GSE13996\n",
      "GSE62584\n",
      "GSE10325\n",
      "GSE43677\n",
      "GSE44164\n",
      "GSE13850\n",
      "GSE12845\n",
      "GSE28079\n",
      "GSE12668\n",
      "GSE3982\n",
      "GSE57611\n",
      "GSE9378\n",
      "56\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE13300\n",
      "GSE13917\n",
      "GSE17186\n",
      "GSE17269\n",
      "GSE36907\n",
      "GSE25638\n",
      "GSE56314\n",
      "GSE64028\n",
      "GSE22886\n",
      "GSE43677\n",
      "GSE12845\n",
      "GSE13411\n",
      "14\n",
      "GSE10821\n",
      "GSE10831\n",
      "GSE110376\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE19246\n",
      "GSE42203\n",
      "GSE56313\n",
      "GSE93984\n",
      "GSE64555\n",
      "GSE74266\n",
      "GSE83441\n",
      "GSE4475\n",
      "GSE44164\n",
      "GSE10172\n",
      "GSE57611\n",
      "GSE43677\n",
      "GSE12845\n",
      "18\n",
      "GSE51382\n",
      "GSE12195\n",
      "GSE12366\n",
      "GSE17186\n",
      "GSE25638\n",
      "GSE56314\n",
      "GSE22886\n",
      "GSE13411\n",
      "8\n",
      "GSE19599\n",
      "1\n",
      "GSE42203\n",
      "GSE19246\n",
      "GSE64555\n",
      "GSE54017\n",
      "4\n",
      "GSE12366\n",
      "GSE13411\n",
      "2\n",
      "GSE13987\n",
      "GSE54017\n",
      "2\n",
      "GSE12845\n",
      "1\n",
      "(1424, 22899)\n",
      "(938, 16206)\n",
      "monocytic.lineage\n",
      "GSE100786\n",
      "GSE101587\n",
      "GSE103381\n",
      "GSE10595\n",
      "GSE22373\n",
      "GSE10896\n",
      "GSE109348\n",
      "GSE113645\n",
      "GSE114181\n",
      "GSE11430\n",
      "GSE23371\n",
      "GSE116487\n",
      "GSE11755\n",
      "GSE11864\n",
      "GSE24223\n",
      "GSE12108\n",
      "GSE12773\n",
      "GSE12838\n",
      "GSE16385\n",
      "GSE13762\n",
      "GSE13899\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE56767\n",
      "GSE17549\n",
      "GSE9801\n",
      "GSE49910\n",
      "GSE51997\n",
      "GSE72642\n",
      "GSE7307\n",
      "GSE41889\n",
      "GSE45535\n",
      "GSE67321\n",
      "GSE93776\n",
      "GSE79706\n",
      "GSE42733\n",
      "GSE58173\n",
      "GSE43346\n",
      "GSE71730\n",
      "GSE10612\n",
      "GSE62584\n",
      "GSE5580\n",
      "GSE11943\n",
      "GSE5099\n",
      "GSE1133\n",
      "GSE16972\n",
      "GSE7138\n",
      "GSE12803\n",
      "GSE46917\n",
      "GSE22886\n",
      "GSE8286\n",
      "GSE9378\n",
      "GSE46913\n",
      "GSE8921\n",
      "54\n",
      "GSE101855\n",
      "GSE10316\n",
      "GSE10463\n",
      "GSE112209\n",
      "GSE112210\n",
      "GSE11327\n",
      "GSE12806\n",
      "GSE13296\n",
      "GSE13762\n",
      "GSE14000\n",
      "GSE22886\n",
      "GSE9946\n",
      "GSE8921\n",
      "GSE3982\n",
      "14\n",
      "GSE10038\n",
      "GSE13670\n",
      "GSE22373\n",
      "GSE11430\n",
      "GSE11886\n",
      "GSE16385\n",
      "GSE13896\n",
      "GSE13931\n",
      "GSE9801\n",
      "GSE41889\n",
      "GSE94801\n",
      "GSE79706\n",
      "GSE43346\n",
      "GSE26294\n",
      "GSE5099\n",
      "GSE16972\n",
      "GSE7138\n",
      "GSE22886\n",
      "GSE3982\n",
      "GSE8286\n",
      "20\n",
      "GSE10316\n",
      "GSE13762\n",
      "GSE23371\n",
      "GSE3982\n",
      "GSE9946\n",
      "5\n",
      "GSE35340\n",
      "GSE109348\n",
      "GSE11151\n",
      "GSE114181\n",
      "GSE28490\n",
      "GSE46913\n",
      "6\n",
      "GSE5099\n",
      "1\n",
      "(1108, 22899)\n",
      "(949, 17198)\n",
      "neutrophils\n",
      "GSE109348\n",
      "GSE11083\n",
      "GSE72642\n",
      "GSE12662\n",
      "GSE12838\n",
      "GSE28490\n",
      "GSE28491\n",
      "GSE67596\n",
      "GSE49910\n",
      "GSE76492\n",
      "GSE58173\n",
      "GSE93776\n",
      "GSE12803\n",
      "GSE12484\n",
      "GSE22886\n",
      "GSE3982\n",
      "16\n",
      "GSE41831\n",
      "GSE45535\n",
      "2\n",
      "(192, 22899)\n",
      "(171, 17198)\n",
      "endothelial.cells\n",
      "GSE10479\n",
      "GSE105437\n",
      "GSE106532\n",
      "GSE108384\n",
      "GSE109178\n",
      "GSE109662\n",
      "GSE109836\n",
      "GSE110085\n",
      "GSE117676\n",
      "GSE12131\n",
      "GSE12293\n",
      "GSE12679\n",
      "GSE12846\n",
      "GSE12891\n",
      "GSE13054\n",
      "GSE13139\n",
      "GSE13637\n",
      "GSE13712\n",
      "GSE13736\n",
      "GSE13791\n",
      "GSE7307\n",
      "GSE59412\n",
      "GSE95298\n",
      "GSE47796\n",
      "GSE42672\n",
      "GSE9891\n",
      "GSE12246\n",
      "GSE10804\n",
      "GSE12907\n",
      "GSE11341\n",
      "GSE3920\n",
      "GSE1133\n",
      "GSE10778\n",
      "GSE14230\n",
      "GSE9378\n",
      "35\n",
      "(363, 22899)\n",
      "(284, 19228)\n",
      "fibroblasts\n",
      "GSE10046\n",
      "GSE10315\n",
      "GSE106151\n",
      "GSE106382\n",
      "GSE22167\n",
      "GSE10880\n",
      "GSE110085\n",
      "GSE11418\n",
      "GSE116912\n",
      "GSE11854\n",
      "GSE118958\n",
      "GSE11919\n",
      "GSE12583\n",
      "GSE13330\n",
      "GSE13606\n",
      "GSE13791\n",
      "GSE13828\n",
      "GSE13837\n",
      "GSE17549\n",
      "GSE95298\n",
      "GSE55673\n",
      "GSE33536\n",
      "GSE47796\n",
      "GSE82159\n",
      "GSE59051\n",
      "GSE42672\n",
      "GSE11767\n",
      "GSE26294\n",
      "GSE14256\n",
      "GSE10125\n",
      "GSE11450\n",
      "GSE3920\n",
      "GSE39262\n",
      "33\n",
      "(320, 22899)\n",
      "(285, 18471)\n"
     ]
    }
   ],
   "source": [
    "# format the first dataframe\n",
    "for cell in ctDict['Coarse'].keys():\n",
    "    print(cell)\n",
    "    gse_dfs=[]\n",
    "    for sub_ct in ctDict['Coarse'][cell]:\n",
    "        gse_list = sampleMeta[sampleMeta['cell.type']==sub_ct]['gse'].unique()\n",
    "        gse_dfs.extend(list(map(lambda x: format_df(x, sub_ct), gse_list)))\n",
    "        print(len(gse_list))\n",
    "    #     gse_dfs[:] = [g for g in gse_dfs if g != None]\n",
    "\n",
    "    ct_master = pd.concat(gse_dfs)\n",
    "    print(ct_master.shape)\n",
    "    \n",
    "    # drop bad samples... \n",
    "    gsm_nas = ct_master.apply(lambda x: sum(np.isnan(x)),1)\n",
    "    gsm_idx = gsm_nas[gsm_nas < 5000].index\n",
    "    # drop the bad genes\n",
    "    GS_nas = ct_master.loc[gsm_idx].apply(lambda x: sum(np.isnan(x)))\n",
    "    GS_idx = GS_nas[GS_nas == 0].index\n",
    "\n",
    "    ct_filtered = ct_master.loc[gsm_idx, GS_idx]\n",
    "    print(ct_filtered.shape)\n",
    "    \n",
    "    filepath = ntpath.join('picklejar',cell+'.pckl')\n",
    "    with open(filepath, 'wb') as f:\n",
    "        pickle.dump(ct_filtered, f)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(285, 18471)"
      ]
     },
     "execution_count": 265,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ct_filtered.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
